<div class="container">

<table style="width: 100%;"><tr>
<td>human</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Amino acid compositions of human proteins</h2>

<h3>Description</h3>

<p>Amino acid compositions of human proteins derived from UniProt.
</p>


<h3>Format</h3>

<p><code>human.aa</code> is a data frame with 25 columns in the format used for amino acid compositions in <span class="pkg">CHNOSZ</span> (see <code>thermo</code>):
</p>

<table>
<tr>
<td style="text-align: left;">
  <code>protein</code> </td>
<td style="text-align: left;"> character </td>
<td style="text-align: left;"> Identification of protein</td>
</tr>
<tr>
<td style="text-align: left;">
  <code>organism</code> </td>
<td style="text-align: left;"> character </td>
<td style="text-align: left;"> Identification of organism</td>
</tr>
<tr>
<td style="text-align: left;">
  <code>ref</code> </td>
<td style="text-align: left;"> character </td>
<td style="text-align: left;"> Reference key for source of sequence data</td>
</tr>
<tr>
<td style="text-align: left;">
  <code>abbrv</code> </td>
<td style="text-align: left;"> character </td>
<td style="text-align: left;"> Abbreviation or other ID for protein (e.g. gene name)</td>
</tr>
<tr>
<td style="text-align: left;">
  <code>chains</code> </td>
<td style="text-align: left;"> numeric </td>
<td style="text-align: left;"> Number of polypeptide chains in the protein</td>
</tr>
<tr>
<td style="text-align: left;">
  <code>Ala</code>...<code>Tyr</code> </td>
<td style="text-align: left;"> numeric </td>
<td style="text-align: left;"> Number of each amino acid in the protein
</td>
</tr>
</table>
<p>The <code>protein</code> column contains UniProt IDs in the format <code>database|accession-isoform</code>, where <code>database</code> is most often ‘<span class="samp">⁠sp⁠</span>’ (Swiss-Prot) or ‘<span class="samp">⁠tr⁠</span>’ (TrEMBL), and <code>isoform</code> is an optional suffix indicating the isoform of the protein (particularly in the <code>human.additional</code> file).
</p>


<h3>Details</h3>

<p>The amino acid compositions of human proteins are stored in three files under <code>extdata/protein</code>.
</p>

<ul>
<li> <p><code>human.base.rds</code> contains amino acid compositions of canonical isoforms of manually reviewed proteins in the <a href="https://www.uniprot.org/">UniProt</a> reference human proteome (computed from sequences in <code>UP000005640_9606.fasta.gz</code>, dated 2016-04-03).
</p>
</li>
<li> <p><code>human.additional.rds</code> contains amino acid compositions of additional proteins (<code>UP000005640_9606_additional.fasta.gz</code>) including isoforms and unreviewed sequences. In version 0.1.5, this file was trimmed to include only those proteins that are used in any of the datasets in the package.
</p>
</li>
<li> <p><code>human.extra.csv</code> contains amino acid compositions of other (“extra”) proteins used in a dataset but not listed in one of the files above. These proteins may include obsolete, unreviewed, or newer additions to the UniProt database. Most, but not all, sequences here are HUMAN (see the <code>organism</code> column and the <code>ref</code> column for the reference keys).
</p>
</li>
</ul>
<p>On loading the package, the individual data files are read and combined, and the result is assigned to the <code>human.aa</code> object in the <code>canprot</code> environment.
</p>


<h3>See Also</h3>

<p><code>human_aa</code> gets amino acid compositions for human proteins specified by their UniProt IDs.
</p>


<h3>Examples</h3>

<pre><code class="language-R"># The number of proteins
nrow(get("human.aa", canprot))
</code></pre>


</div>