<div class="container">

<table style="width: 100%;"><tr>
<td>metrics</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Calculate chemical metrics for proteins</h2>

<h3>Description</h3>

<p>Calculate chemical metrics for proteins from their amino acid compositions.
</p>


<h3>Usage</h3>

<pre><code class="language-R">  Zc(AAcomp, ...)
  nO2(AAcomp, basis = "QEC", ...)
  nH2O(AAcomp, basis = "QEC", terminal_H2O = 0)
  GRAVY(AAcomp, ...)
  pI(AAcomp, terminal_H2O = 1, ...)
  MW(AAcomp, terminal_H2O = 0, ...)
  pMW(AAcomp, terminal_H2O = 1, ...)
  V0(AAcomp, terminal_H2O = 0, ...)
  pV0(AAcomp, terminal_H2O = 1, ...)
  V0g(AAcomp, ...)
  Density(AAcomp, ...)
  S0(AAcomp, terminal_H2O = 0, ...)
  pS0(AAcomp, terminal_H2O = 1, ...)
  S0g(AAcomp, ...)
  SV(AAcomp, ...)
  Zcg(AAcomp, ...)
  nH2Og(AAcomp, ...)
  nO2g(AAcomp, ...)
  HC(AAcomp, ...)
  NC(AAcomp, ...)
  OC(AAcomp, ...)
  SC(AAcomp, ...)
  nC(AAcomp, ...)
  pnC(AAcomp, ...)
  plength(AAcomp, ...)
  Cost(AAcomp, ...)
  RespiratoryCost(AAcomp, ...)
  FermentativeCost(AAcomp, ...)
  B20Cost(AAcomp, ...)
  Y20Cost(AAcomp, ...)
  H11Cost(AAcomp, ...)
  cplab
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>AAcomp</code></td>
<td>
<p>data frame, amino acid compositions</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>ignored additional arguments</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>basis</code></td>
<td>
<p>character, set of basis species</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>terminal_H2O</code></td>
<td>
<p>numeric, number of pairs of terminal groups</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Columns in <code>AAcomp</code> should be named with the three-letter abbreviations for the amino acids.
Case-insensitive matching matching of the abbreviations is used; e.g., ‘<span class="samp">⁠Ala⁠</span>’, ‘<span class="samp">⁠ALA⁠</span>’, ‘<span class="samp">⁠ala⁠</span>’ all refer to alanine.
</p>
<p>Metrics are normalized per amino acid residue except for <code>Zc</code>, <code>pI</code>, <code>Density</code>, <code>plength</code>, and other functions starting with <code>p</code> (for protein).
The contribution of protein terminal groups (-H and -OH) to residue-normalized metrics is turned off by default.
Set <code>terminal_H2O</code> to 1 (or to the number of polypeptide chains, if greater than one) to include their contribution.
</p>
<p>The metrics are described below:
</p>

<dl>
<dt><code>Zc</code></dt>
<dd>
<p>Average oxidation state of carbon (<i>Z</i><sub>C</sub>) (Dick, 2014).
This metric is independent of the choice of basis species.
Note that <i>Z</i><sub>C</sub> is normalized by number of carbon atoms, not by number of residues.
</p>
</dd>
<dt><code>nO2</code></dt>
<dd>
<p>Stoichiometric oxidation state (<i>n</i><sub>O<sub>2</sub></sub> per residue).
The available <code>basis</code> species are:
</p>

<ul>
<li>
<p>‘<span class="samp">⁠QEC⁠</span>’ - glutamine, glutamic acid, cysteine, H<sub>2</sub>O, O<sub>2</sub> (Dick et al., 2020)
</p>
</li>
<li>
<p>‘<span class="samp">⁠QCa⁠</span>’ - glutamine, cysteine, acetic acid, H<sub>2</sub>O, O<sub>2</sub></p>
</li>
</ul>
</dd>
<dt><code>nH2O</code></dt>
<dd>
<p>Stoichiometric hydration state (<i>n</i><sub>H<sub>2</sub>O</sub> per residue).
The basis species also affect this calculation.
</p>
</dd>
<dt><code>GRAVY</code></dt>
<dd>
<p>Grand average of hydropathy.
Values of the hydropathy index for individual amino acids are from Kyte and Doolittle (1982).
</p>
</dd>
<dt><code>pI</code></dt>
<dd>
<p>Isoelectric point.
The net charge for each ionizable group was pre-calculated from pH 0 to 14 at intervals of 0.01.
The isoelectric point is found as the pH where the sum of charges of all groups in the protein is closest to zero.
The p<i>K</i> values for the terminal groups and sidechains are taken from Bjellqvist et al. (1993) and Bjellqvist et al. (1994); note that the calculation does not implement position-specific adjustments described in the latter paper.
The number of N- and C-terminal groups is taken from <code>terminal_H2O</code>.
</p>
</dd>
<dt><code>MW</code></dt>
<dd>
<p>Molecular weight.</p>
</dd>
<dt><code>pMW</code></dt>
<dd>
<p>Molecular weight per protein.</p>
</dd>
<dt><code>V0</code></dt>
<dd>
<p>Standard molal volume.
The values are derived from group contributions of amino acid sidechains and protein backbones (Dick et al., 2006).
</p>
</dd>
<dt><code>pV0</code></dt>
<dd>
<p>Standard molal volume per protein.</p>
</dd>
<dt><code>V0g</code></dt>
<dd>
<p>Specific volume (reciprocal density).</p>
</dd>
<dt><code>Density</code></dt>
<dd>
<p>Density (MW / V0).</p>
</dd>
<dt><code>S0</code></dt>
<dd>
<p>Standard molal entropy.
The values are derived from group contributions of amino acid sidechains and protein backbones (Dick et al., 2006).
</p>
</dd>
<dt><code>pS0</code></dt>
<dd>
<p>Standard molal entropy per protein.</p>
</dd>
<dt><code>S0g</code></dt>
<dd>
<p>Specific entropy.</p>
</dd>
<dt><code>SV</code></dt>
<dd>
<p>Entropy density.</p>
</dd>
<dt><code>Zcg</code></dt>
<dd>
<p>Carbon oxidation state per gram.</p>
</dd>
<dt><code>nO2g</code></dt>
<dd>
<p>Stoichiometric oxidation state per gram.</p>
</dd>
<dt><code>nH2Og</code></dt>
<dd>
<p>Stoichiometric hydration state per gram.</p>
</dd>
<dt><code>HC</code></dt>
<dd>
<p>H/C ratio (not counting terminal -H and -OH groups).</p>
</dd>
<dt><code>NC</code></dt>
<dd>
<p>N/C ratio.</p>
</dd>
<dt><code>OC</code></dt>
<dd>
<p>O/C ratio (not counting terminal -H and -OH groups).</p>
</dd>
<dt><code>SC</code></dt>
<dd>
<p>S/C ratio.</p>
</dd>
<dt><code>nC</code></dt>
<dd>
<p>Number of carbon atoms per residue.</p>
</dd>
<dt><code>pnC</code></dt>
<dd>
<p>Number of carbon atoms per protein.</p>
</dd>
<dt><code>plength</code></dt>
<dd>
<p>Protein length (number of amino acid residues).</p>
</dd>
<dt><code>Cost</code></dt>
<dd>
<p>Metabolic cost (Akashi and Gojobori, 2002).</p>
</dd>
<dt><code>RespiratoryCost</code></dt>
<dd>
<p>Respiratory cost (Wagner, 2005).</p>
</dd>
<dt><code>FermentativeCost</code></dt>
<dd>
<p>Fermentative cost (Wagner, 2005).</p>
</dd>
<dt><code>B20Cost</code></dt>
<dd>
<p>Biosynthetic cost in bacteria (Zhang et al., 2018).</p>
</dd>
<dt><code>Y20Cost</code></dt>
<dd>
<p>Biosynthetic cost in yeast (Zhang et al., 2018).</p>
</dd>
<dt><code>H11Cost</code></dt>
<dd>
<p>Biosynthetic cost in humans (Zhang et al., 2018).</p>
</dd>
</dl>
<p><code>...</code> is provided to permit <code>get</code> or <code>do.call</code> constructions with the same arguments for all metrics.
For instance, a <code>terminal_H2O</code> argument can be suppled to either <code>Zc</code> or <code>nH2O</code>, but it only has an effect on the latter.
</p>
<p><code>cplab</code> is a list of formatted labels for each of the chemical metrics listed here.
A check in the code ensures that the names of the functions for calculating metrics and the names for labels listed <code>cplab</code> are identical.
</p>


<h3>References</h3>

<p>Akashi H, Gojobori T. 2002. Metabolic efficiency and amino acid composition in the proteomes of <em>Escherichia coli</em> and <em>Bacillus subtilis</em>. <em>Proceedings of the National Academy of Sciences</em> <b>99</b>(6): 3695–3700. <a href="https://doi.org/10.1073/pnas.062526999">doi:10.1073/pnas.062526999</a>
</p>
<p>Bjellqvist B, Hughes GJ, Pasquali C, Paquet N, Ravier F, Sanchez J-C, Frutiger S, Hochstrasser D. 1993. The focusing positions of polypeptides in immobilized pH gradients can be predicted from their amino acid sequences. <em>Electrophoresis</em> <b>14</b>: 1023–1031. <a href="https://doi.org/10.1002/elps.11501401163">doi:10.1002/elps.11501401163</a>
</p>
<p>Bjellqvist B, Basse B, Olsen E, Celis JE. 1994. Reference points for comparisons of two-dimensional maps of proteins from different human cell types defined in a pH scale where isoelectric points correlate with polypeptide compositions. <em>Electrophoresis</em> <b>15</b>: 529–539. <a href="https://doi.org/10.1002/elps.1150150171">doi:10.1002/elps.1150150171</a>
</p>
<p>Dick JM, LaRowe DE, Helgeson HC. 2006. Temperature, pressure, and electrochemical constraints on protein speciation: Group additivity calculation of the standard molal thermodynamic properties of ionized unfolded proteins. <em>Biogeosciences</em> <b>3</b>(3): 311–336. <a href="https://doi.org/10.5194/bg-3-311-2006">doi:10.5194/bg-3-311-2006</a>
</p>
<p>Dick JM. 2014. Average oxidation state of carbon in proteins. <em>J. R. Soc. Interface</em> <b>11</b>: 20131095. <a href="https://doi.org/10.1098/rsif.2013.1095">doi:10.1098/rsif.2013.1095</a>
</p>
<p>Dick JM, Yu M, Tan J. 2020. Uncovering chemical signatures of salinity gradients through compositional analysis of protein sequences. <em>Biogeosciences</em> <b>17</b>: 6145–6162. <a href="https://doi.org/10.5194/bg-17-6145-2020">doi:10.5194/bg-17-6145-2020</a>
</p>
<p>Kyte J, Doolittle RF. 1982. A simple method for displaying the hydropathic character of a protein. <em>J. Mol. Biol.</em> <b>157</b>: 105–132. <a href="https://doi.org/10.1016/0022-2836%2882%2990515-0">doi:10.1016/0022-2836(82)90515-0</a>
</p>
<p>Wagner A. 2005. Energy constraints on the evolution of gene expression. <em>Molecular Biology and Evolution</em> <b>22</b>(6): 1365–1374. <a href="https://doi.org/10.1093/molbev/msi126">doi:10.1093/molbev/msi126</a>
</p>
<p>Zhang H, Wang Y, Li J, Chen H, He X, Zhang H, Liang H, Lu J. 2018. Biosynthetic energy cost for amino acids decreases in cancer evolution. <em>Nature Communications</em> <b>9</b>(1): 4124. <a href="https://doi.org/10.1038/s41467-018-06461-1">doi:10.1038/s41467-018-06461-1</a>
</p>


<h3>See Also</h3>

<p>For calculation of <i>Z</i><sub>C</sub> from an elemental formula (instead of amino acid composition), see the <code>ZC</code> function in <span class="pkg">CHNOSZ</span>.
<code>calc_metrics</code> is a wrapper to calculate one or more metrics specified in an argument.
</p>


<h3>Examples</h3>

<pre><code class="language-R"># Amino acid composition of a tripeptide (Gly-Ala-Gly)
aa &lt;- data.frame(Ala = 1, Gly = 2)
# Calculate Zc, nH2O, and length
Zc(aa)
nH2O(aa)
plength(aa)

# Make a plot with formatted labels
plot(Zc(aa), nH2O(aa), xlab = cplab$Zc, ylab = cplab$nH2O)
</code></pre>


</div>