<div class="container">

<table style="width: 100%;"><tr>
<td>plot_variant_influences</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Plot cape coefficients</h2>

<h3>Description</h3>

<p>This function plots the the cape coefficients between
pairs of markers as a heat map.
The interactions are shown in the main part of the heatmap
while the main effects are shown on the right hand side.
Directed interactions are read from the y axis to the x axis.
For example an interaction from marker1 to marker2 will be shown
in the row corresponding to marker1 and the column corresponding
to marker2. 
Similarly, if marker1 has a main effect on any traits, these
will be shown in the row for marker1 and the trait columns.
</p>


<h3>Usage</h3>

<pre><code class="language-R">plot_variant_influences(
  data_obj,
  p_or_q = 0.05,
  min_std_effect = 0,
  plot_all_vals = FALSE,
  standardize = FALSE,
  color_scheme = c("DO/CC", "other"),
  pos_col = "brown",
  neg_col = "blue",
  not_tested_col = "lightgray",
  show_marker_labels = FALSE,
  show_chr = TRUE,
  label_chr = TRUE,
  show_alleles = TRUE,
  scale_effects = c("log10", "sqrt", "none"),
  pheno_width = NULL,
  covar_width = NULL,
  covar_labels = NULL,
  phenotype_labels = NULL,
  show_not_tested = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>data_obj</code></td>
<td>
<p>a <code>Cape</code> object</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p_or_q</code></td>
<td>
<p>A threshold indicating the maximum p value (or q value
if FDR was used) of significant interactions and main effects</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min_std_effect</code></td>
<td>
<p>An optional filter. The plot will exclude
all pairs with standardized effects below the number set here.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot_all_vals</code></td>
<td>
<p>If TRUE will plot all values regardless of 
significant</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>standardize</code></td>
<td>
<p>Whether to plot effect sizes (FALSE) or standardized
effect sizes (TRUE)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>color_scheme</code></td>
<td>
<p>A character value of either "DO/CC" or other indicating the 
color scheme of main effects. If "DO/CC" allele effects can be plotted with the
DO/CC colors.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pos_col</code></td>
<td>
<p>The color to use for positive main effects and interactions
must be one of "green", "purple", "red", "orange", "blue", "brown", "yellow", "gray"
see <code>get_color</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>neg_col</code></td>
<td>
<p>The color to use for negative main effects and interactions
takes the same values as pos_col.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>not_tested_col</code></td>
<td>
<p>The color to use for marker pairs not tested. Takes
the same values as pos_col and neg_col</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>show_marker_labels</code></td>
<td>
<p>Whether to write the marker labels on the plot</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>show_chr</code></td>
<td>
<p>Whether to show chromosome boundaries</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>label_chr</code></td>
<td>
<p>Whether to label chromosomes if plotted</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>show_alleles</code></td>
<td>
<p>If TRUE, the allele of each marker is indicated by color.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>scale_effects</code></td>
<td>
<p>One of "log10", "sqrt", "none." If some effects are
very large, scaling them can help show contrasts between smaller values.
The default is no scaling.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pheno_width</code></td>
<td>
<p>Each marker and trait gets one column in the matrix. 
If there are many markers, this makes the effects on the traits difficult 
to see. pheno_width increases the number of columns given to the phenotypes.
For example, if pheno_width = 11, the phenotypes will be shown 11 times wider
than individual markers.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>covar_width</code></td>
<td>
<p>See pheno_width. This is the same effect for covariates.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>covar_labels</code></td>
<td>
<p>Labels for covariates if different from those stored in 
the data object.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>phenotype_labels</code></td>
<td>
<p>Labels for traits if different from those stored in 
the data object</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>show_not_tested</code></td>
<td>
<p>Whether to color the marker pairs that were not
tested. If FALSE, they will not be colored in.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>This function invisibly returns the variant influences matrix.
shown in the heat map.
</p>


</div>