<div class="container">

<table style="width: 100%;"><tr>
<td>write_population</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Save the cross data in R/qtl CSV format</h2>

<h3>Description</h3>

<p>This function writes out a cape object
in a csv format readable both by <code>read_population</code>
in Cape or by read.cross in R/qtl.
</p>


<h3>Usage</h3>

<pre><code class="language-R">write_population(
  data_obj,
  geno_obj,
  ref_allele = "A",
  na = NA,
  filename = "capeData.csv"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>data_obj</code></td>
<td>
<p>a <code>Cape</code> object</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>geno_obj</code></td>
<td>
<p>a genotype object</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ref_allele</code></td>
<td>
<p>a character, e.g., "A", that represents the reference allele in the data object</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>na</code></td>
<td>
<p>either NA or a character used to represent a missing data value in the output</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>filename</code></td>
<td>
<p>absolute or relative path to the output file's destination</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Writes a file to the destination path
</p>


<h3>References</h3>

<p>Broman et al. (2003) R/qtl: QTL mapping in experimental crosses. 
Bioinformatics 19:889-890 doi:10.1093/bioinformatics/btg112
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
write_population(data_obj, geno_obj)

## End(Not run)

</code></pre>


</div>