<div class="container">

<table style="width: 100%;"><tr>
<td>read.tab</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Read in a Tab Delimited File.</h2>

<h3>Description</h3>

<p>This function is a slight (genome annotation friendly) variant of the built-in read.delim function in R.  
Two non-standard defaults have been set: stringsAsFactors=TRUE, quote="". 
An additional parameter "check.row.ct", triggering a count.fields call, has been added to further ensure the integrity of large data files.
</p>


<h3>Usage</h3>

<pre><code class="language-R">read.tab(file, check.row.ct = TRUE, stringsAsFactors = FALSE, 
               quote = "", header=TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>file</code></td>
<td>
<p>the name of the file which the data are to be read from.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>check.row.ct</code></td>
<td>
<p>logical: use 'count.fields' to independently verify the number of rows read.table reads into memory?</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>stringsAsFactors</code></td>
<td>
<p>logical: should character vectors be converted to factors?.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>quote</code></td>
<td>
<p>the set of quoting characters.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>header</code></td>
<td>
<p>boolean specifying if the first row serves as labels for the columns</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>other paramters passed to read.delim.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>a dataframe.
</p>


</div>