<div class="container">

<table style="width: 100%;"><tr>
<td>geneSelection</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Methods for selecting genes</h2>

<h3>Description</h3>

<p>Selection of differentially expressed genes.
</p>


<h3>Usage</h3>

<pre><code class="language-R">## S4 method for signature 'micro_array,micro_array,numeric'
geneSelection(
  x,
  y,
  tot.number,
  data_log = TRUE,
  wanted.patterns = NULL,
  forbidden.patterns = NULL,
  peak = NULL,
  alpha = 0.05,
  Design = NULL,
  lfc = 0
)

## S4 method for signature 'list,list,numeric'
geneSelection(
  x,
  y,
  tot.number,
  data_log = TRUE,
  alpha = 0.05,
  cont = FALSE,
  lfc = 0,
  f.asso = NULL
)

## S4 method for signature 'micro_array,numeric'
genePeakSelection(
  x,
  peak,
  y = NULL,
  data_log = TRUE,
  durPeak = c(1, 1),
  abs_val = TRUE,
  alpha_diff = 0.05
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>either a micro_array object or a list of micro_array objects. In
the first case, the micro_array object represents the stimulated
measurements. In the second case, the control unstimulated data (if present)
should be the first element of the list.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>y</code></td>
<td>
<p>either a micro_array object or a list of strings. In the first
case, the micro_array object represents the stimulated measurements. In the
second case, the list is the way to specify the contrast: </p>

<dl>
<dt>First element:</dt>
<dd>
<p> condition, condition&amp;time or pattern. The condition
specification is used when the overall is to compare two conditions.  The
condition&amp;time specification is used when comparing two conditions at two
precise time points. The pattern specification allows to decide which time
point should be differentially expressed.</p>
</dd> <dt>Second element:</dt>
<dd>
<p>a vector
of length 2. The two conditions which should be compared. If a condition is
used as control, it should be the first element of the vector. However, if
this control is not measured throught time, the option cont=TRUE should be
used.</p>
</dd> <dt>Third element:</dt>
<dd>
<p>depends on the first element.  It is no needed
if condition has been specified.  If condition&amp;time has been specified, then
this is a vector containing the time point at which the comparison should be
done. If pattern has been specified, then this is a vector of 0 and 1 of
length T, where T is the number of time points. The time points with desired
differential expression are provided with 1.  </p>
</dd>
</dl>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tot.number</code></td>
<td>
<p>an integer. The number of selected genes. If tot.number &lt;0
all differentially genes are selected. If tot.number &gt; 1, tot.number is the
maximum of diffenrtially genes that will be selected.  If 0&lt;tot.number&lt;1,
tot.number represents the proportion of diffenrentially genes that are
selected.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>data_log</code></td>
<td>
<p>logical (default to TRUE); should data be logged ?</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>wanted.patterns</code></td>
<td>
<p>a matrix with wanted patterns [only for geneSelection].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>forbidden.patterns</code></td>
<td>
<p>a matrix with forbidden patterns [only for geneSelection].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>peak</code></td>
<td>
<p>interger. At which time points measurements should the genes be
selected [optionnal for geneSelection].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alpha</code></td>
<td>
<p>float; the risk level. Default to 'alpha=0.05'</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Design</code></td>
<td>
<p>the design matrix of the experiment. Defaults to 'NULL'.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>lfc</code></td>
<td>
<p>log fold change value used in limma's 'topTable'. Defaults to 0.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cont</code></td>
<td>
<p>use contrasts. Defaults to 'FALSE'.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>f.asso</code></td>
<td>
<p>function used to assess the association between the genes. 
The default value 'NULL' implies the use of the usual 'mean' function.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>durPeak</code></td>
<td>
<p>vector of size 2 (default to c(1,1)) ; the first elements gives the length of the peak at
the left, the second at the right. [only for genePeakSelection]</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>abs_val</code></td>
<td>
<p>logical (default to TRUE) ; should genes be selected on the
basis of their absolute value expression ? [only for genePeakSelection]</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alpha_diff</code></td>
<td>
<p>float; the risk level</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A micro_array object.
</p>


<h3>Author(s)</h3>

<p>Nicolas Jung, Frédéric Bertrand , Myriam Maumy-Bertrand.
</p>


<h3>References</h3>

<p>Jung, N., Bertrand, F., Bahram, S., Vallat, L., and
Maumy-Bertrand, M. (2014). Cascade: a R-package to study, predict and
simulate the diffusion of a signal through a temporal gene network.
<em>Bioinformatics</em>, btt705.
</p>
<p>Vallat, L., Kemper, C. A., Jung, N., Maumy-Bertrand, M., Bertrand, F.,
Meyer, N., ... &amp; Bahram, S. (2013). Reverse-engineering the genetic
circuitry of a cancer cell with predicted intervention in chronic
lymphocytic leukemia. <em>Proceedings of the National Academy of
Sciences</em>, 110(2), 459-464.
</p>


<h3>Examples</h3>

<pre><code class="language-R">

 if(require(CascadeData)){
	data(micro_US)
	micro_US&lt;-as.micro_array(micro_US,time=c(60,90,210,390),subject=6)
	data(micro_S)
	micro_S&lt;-as.micro_array(micro_S,time=c(60,90,210,390),subject=6)

  #Basically, to find the 50 more significant expressed genes you will use:
  Selection_1&lt;-geneSelection(x=micro_S,y=micro_US,
  tot.number=50,data_log=TRUE)
  summary(Selection_1)
  
  #If we want to select genes that are differentially 
  #at time t60 or t90 :
  Selection_2&lt;-geneSelection(x=micro_S,y=micro_US,tot.number=30,
  wanted.patterns=
  rbind(c(0,1,0,0),c(1,0,0,0),c(1,1,0,0)))
  summary(Selection_2)

  #To select genes that have a differential maximum of expression at a specific time point.
  
  Selection_3&lt;-genePeakSelection(x=micro_S,y=micro_US,peak=1,
  abs_val=FALSE,alpha_diff=0.01)
  summary(Selection_3)
  }

 if(require(CascadeData)){
data(micro_US)
micro_US&lt;-as.micro_array(micro_US,time=c(60,90,210,390),subject=6)
data(micro_S)
micro_S&lt;-as.micro_array(micro_S,time=c(60,90,210,390),subject=6)
#Genes with differential expression at t1
Selection1&lt;-geneSelection(x=micro_S,y=micro_US,20,wanted.patterns= rbind(c(1,0,0,0)))
#Genes with differential expression at t2
Selection2&lt;-geneSelection(x=micro_S,y=micro_US,20,wanted.patterns= rbind(c(0,1,0,0)))
#Genes with differential expression at t3
Selection3&lt;-geneSelection(x=micro_S,y=micro_US,20,wanted.patterns= rbind(c(0,0,1,0)))
#Genes with differential expression at t4
Selection4&lt;-geneSelection(x=micro_S,y=micro_US,20,wanted.patterns= rbind(c(0,0,0,1)))
#Genes with global differential expression 
Selection5&lt;-geneSelection(x=micro_S,y=micro_US,20)

#We then merge these selections:
Selection&lt;-unionMicro(list(Selection1,Selection2,Selection3,Selection4,Selection5))
print(Selection)

#Prints the correlation graphics Figure 4:
summary(Selection,3)

##Uncomment this code to retrieve geneids.
#library(org.Hs.eg.db)
#
#ff&lt;-function(x){substr(x, 1, nchar(x)-3)}
#ff&lt;-Vectorize(ff)
#
##Here is the function to transform the probeset names to gene ID.
#
#library("hgu133plus2.db")
#
#probe_to_id&lt;-function(n){  
#x &lt;- hgu133plus2SYMBOL
#mp&lt;-mappedkeys(x)
#xx &lt;- unlist(as.list(x[mp]))
#genes_all = xx[(n)]
#genes_all[is.na(genes_all)]&lt;-"unknown"
#return(genes_all)
#}
#Selection@name&lt;-probe_to_id(Selection@name)
}


</code></pre>


</div>