<div class="container">

<table style="width: 100%;"><tr>
<td>micro_S</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Stimulated dataset</h2>

<h3>Description</h3>

<p>This is the stimulated data part of the GSE39411 dataset,
<a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE39411">https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE39411</a>. Data were
normalized and are ready to use.
</p>


<h3>Format</h3>

<p>A data frame with 54613 probesets measured 6 times throught 4 time
points.
</p>


<h3>Details</h3>

<p>5 leukemic CLL B-lymphocyte of aggressive form were stimulated in vitro with
an anti-IgM antibody, activating the B-cell receptor (BCR). We analyzed the
gene expression at 4 time points (60, 90, 210 and 390 minutes). Each gene
expression measurement is performed both in stimulated cells and in control
unstimulated cells. This is the stimulated cells dataset.
</p>
<p>Data were collected on HG-U133_Plus_2, Affymetrix Human Genome U133 Plus 2.0
Array.
</p>


<h3>References</h3>

<p>Vallat, L., Kemper, C. A., Jung, N., Maumy-Bertrand, M.,
Bertrand, F., ..., Bahram, S. (2013). Reverse-engineering the genetic
circuitry of a cancer cell with predicted intervention in chronic
lymphocytic leukemia. Proceedings of the National Academy of Sciences,
110(2), 459-464, <a href="https://doi.org/10.1073/pnas.1211130110">doi:10.1073/pnas.1211130110</a>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">
data(micro_S)

</code></pre>


</div>