<div class="container">

<table style="width: 100%;"><tr>
<td>predict,micro_array-method</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Prediction of the gene expressions after a knock-out experience
<code>predict</code>
</h2>

<h3>Description</h3>

<p>Prediction of the gene expressions after a knock-out experience
</p>


<h3>Usage</h3>

<pre><code class="language-R">## S4 method for signature 'micro_array'
predict(object, Omega, nv = 0, targets = NULL, adapt = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>object</code></td>
<td>
<p>a micro_array object</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Omega</code></td>
<td>
<p>a network object.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nv</code></td>
<td>
<p>[=0] numeric; the level of the cutoff</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>targets</code></td>
<td>
<p>[NULL] vector; which genes are knocked out?</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>adapt</code></td>
<td>
<p>[TRUE] boolean; do not raise an error if used with vectors 
instead of one column matrices.</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>Nicolas Jung, Frédéric Bertrand , Myriam Maumy-Bertrand.
</p>


<h3>References</h3>

<p>Jung, N., Bertrand, F., Bahram, S., Vallat, L., and
Maumy-Bertrand, M. (2014). Cascade: a R-package to study, predict and
simulate the diffusion of a signal through a temporal gene network.
<em>Bioinformatics</em>, btt705.
</p>
<p>Vallat, L., Kemper, C. A., Jung, N., Maumy-Bertrand, M., Bertrand, F.,
Meyer, N., ... &amp; Bahram, S. (2013). Reverse-engineering the genetic
circuitry of a cancer cell with predicted intervention in chronic
lymphocytic leukemia. <em>Proceedings of the National Academy of
Sciences</em>, 110(2), 459-464.
</p>


<h3>Examples</h3>

<pre><code class="language-R">
data(Selection)
data(network)
#A nv value can chosen using the cutoff function
nv=.11
EGR1&lt;-which(match(Selection@name,"EGR1")==1)
P&lt;-position(network,nv=nv)

#We predict gene expression modulations within the network if EGR1 is experimentaly knocked-out. 
prediction_ko5&lt;-predict(Selection,network,nv=nv,targets=EGR1)

#Then we plot the results. Here for example we see changes at time point t2:
plot(prediction_ko5,time=2,ini=P,label_v=Selection@name)

</code></pre>


</div>