<div class="container">

<table style="width: 100%;"><tr>
<td>CASMAP-class</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Constructor for CASMAP class object.</h2>

<h3>Description</h3>

<p>Constructor for CASMAP class object.
</p>


<h3>Details</h3>

<p>Constructor for CASMAP class object, which needs the <code>mode</code>
parameter to be set by the user. Please see the examples.
</p>


<h3>Fields</h3>


<dl>
<dt><code>mode</code></dt>
<dd>
<p>Either <code>'regionGWAS'</code> or <code>'higherOrderEpistasis'</code>.</p>
</dd>
<dt><code>alpha</code></dt>
<dd>
<p>A numeric value setting the Family-wise Error Rate (FWER).
Must be strictly between <code>0</code> and <code>1</code>. Default
value is <code>0.05</code>.</p>
</dd>
<dt><code>max_comb_size</code></dt>
<dd>
<p>A numeric specifying the maximum length of 
combinations. For example, if set to <code>4</code>, 
then only combinations of size between <code>1</code>
and <code>4</code> (inclusive) will be considered.
To consider combinations of arbitrary (maximal)
length, use value <code>0</code>, which is the default 
value.</p>
</dd>
</dl>
<h3>Base method, for both modes</h3>


<dl>
<dt><code>readFiles</code></dt>
<dd>
<p>Read the data, label and possibly covariates 
files. Parameters are <code>genotype_file</code>, 
for the data, <code>phenotype_file</code> for the
labels and (optional) <code>covariates_file</code>
for the covariates. The option 
<code>plink_file_root</code> is not supported
in the current version, but will be supported
in future versions.</p>
</dd>
<dt><code>setMode</code></dt>
<dd>
<p>Can set/change the mode, but note that any
data files will need to read in again using
the <code>readFiles</code> command.</p>
</dd>
<dt><code>setTargetFWER</code></dt>
<dd>
<p>Can set/change the Family-wise 
Error Rate (FWER). Takes a numeric
parameter <code>alpha</code>, strictly between
<code>0</code> and <code>1</code>.</p>
</dd>
<dt><code>execute</code></dt>
<dd>
<p>Once the data files have been read, can execute the
algorithm. Please note that, depending on the size
of the data files, this could take a long time.</p>
</dd>
<dt><code>getSummary</code></dt>
<dd>
<p>Returns a data frame with a summary of the 
results from the execution, but not any
significant regions/itemsets. See 
<code>getSignificantRegions</code>, 
<code>getSignificantInteractions</code>, and
<code>getSignificantClusterRepresentatives</code>. </p>
</dd>
<dt><code>writeSummary</code></dt>
<dd>
<p>Directly write the information
from <code>getSummary</code> to file.</p>
</dd>
</dl>
<h3>
<code>regionGWAS</code> Methods</h3>


<dl>
<dt><code>getSignificantRegions</code></dt>
<dd>
<p>Returns a data frame with the 
the significant regions. Only valid when
<code>mode='regionGWAS'</code>.</p>
</dd>
<dt><code>getSignificantClusterRepresentatives</code></dt>
<dd>
<p>Returns a data 
frame with the 
the representatives of the significant 
clusters. This will be a subset of the regions
returned from <code>getSignificantRegions</code>. 
Only valid when <code>mode='regionGWAS'</code>.</p>
</dd>
<dt><code>writeSignificantRegions</code></dt>
<dd>
<p>Writes the data from 
<code>getSignificantRegions</code> to file, which
must be specified in the parameter 
<code>path</code>.
Only valid when <code>mode='regionGWAS'</code>.</p>
</dd>
<dt><code>writeSignificantClusterRepresentatives</code></dt>
<dd>
<p>Writes the data 
from 
<code>getSignificantClusterRepresentatives</code> to 
file, which must be specified in the parameter 
<code>path</code>.
Only valid when <code>mode='regionGWAS'</code>.</p>
</dd>
</dl>
<h3>
<code>higherOrderEpistasis</code> Methods</h3>


<dl>
<dt><code>getSignificantInteractions</code></dt>
<dd>
<p>Returns the frame 
from <code>getSignificantInteractions</code> to 
file, which must be specified in the parameter 
<code>path</code>. Only valid 
when <code>mode='higherOrderEpistasis'</code>.</p>
</dd>
<dt><code>writeSignificantInteractions</code></dt>
<dd>
<p>Writes a data frame with  
the significant interactions. Only valid 
when <code>mode='higherOrderEpistasis'</code>.</p>
</dd>
</dl>
<h3>References</h3>

<p>A. Terada, M. Okada-Hatakeyama, K. Tsuda and J. Sese
<em>Statistical significance of combinatorial regulations</em>,
Proceedings of the National Academy of Sciences (2013) 110 
(32): 12996-13001
</p>
<p>F. Llinares-Lopez, D. G. Grimm, D. Bodenham,
U. Gieraths, M. Sugiyama, B. Rowan and K. Borgwardt,
<em>Genome-wide detection of intervals of genetic heterogeneity 
associated with complex traits</em>,
ISMB 2015, Bioinformatics (2015) 31 (12): i240-i249
</p>
<p>L. Papaxanthos, F. Llinares-Lopez, D. Bodenham,
K .Borgwardt,
<em>Finding significant combinations of features in the 
presence of categorical covariates</em>, Advances
in Neural Information Processing Systems 29 (NIPS 2016), 2271-2279.
</p>
<p>F. Llinares-Lopez, L. Papaxanthos, D. Bodenham,
D. Roqueiro and K .Borgwardt,
<em>Genome-wide genetic heterogeneity discovery 
with categorical covariates</em>.
Bioinformatics 2017, 33 (12): 1820-1828.
</p>


<h3>Examples</h3>

<pre><code class="language-R">
## An example using the "regionGWAS" mode
fastcmh &lt;- CASMAP(mode="regionGWAS")      # initialise object

datafile &lt;- getExampleDataFilename()      # file name of example data
labelsfile &lt;- getExampleLabelsFilename()  # file name of example labels
covfile &lt;- getExampleCovariatesFilename() # file name of example covariates 

# read the data, labels and covariate files
fastcmh$readFiles(genotype_file=getExampleDataFilename(),
                  phenotype_file=getExampleLabelsFilename(), 
                  covariate_file=getExampleCovariatesFilename() )

# execute the algorithm (this may take some time)
fastcmh$execute()

#get the summary results
summary_results &lt;- fastcmh$getSummary()

#get the significant regions
sig_regions &lt;- fastcmh$getSignificantRegions()

#get the clustered representatives for the significant regions
sig_cluster_rep &lt;- fastcmh$getSignificantClusterRepresentatives()


## Another example of regionGWAS
fais &lt;- CASMAP(mode="regionGWAS")      # initialise object

# read the data and labels, but no covariates
fastcmh$readFiles(genotype_file=getExampleDataFilename(),
                  phenotype_file=getExampleLabelsFilename())


## Another example, doing higher order epistasis search
facs &lt;- CASMAP(mode="higherOrderEpistasis")      # initialise object

</code></pre>


</div>