<div class="container">

<table style="width: 100%;"><tr>
<td>run_sgrna_quant</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>A function to run a sgRNA quantification algorithm from NGS sample</h2>

<h3>Description</h3>

<p>A function to run a sgRNA quantification algorithm from NGS sample
</p>


<h3>Usage</h3>

<pre><code class="language-R">run_sgrna_quant(lib_path, design, map_path = NULL, ncores = 1, verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>lib_path</code></td>
<td>
<p>The path of the FASTA file.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>design</code></td>
<td>
<p>A table contains the study design. It must contain 'fastq_path' and 'sample_name.'</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>map_path</code></td>
<td>
<p>The path of file contains gene-sgRNA mapping.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ncores</code></td>
<td>
<p>The number that indicates how many processors will be used with a parallelization. 
The parallelization will be enabled if users do not set the parameter as '-1“ 
(it means the full physical cores will be used) or greater than '1'.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>Display some logs during the quantification if it is set to 'TRUE'</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>It will return a list, and the list contains three elements. 
The first element (‘count’) is a data frame contains the result of the quantification for each sample. 
The second element (‘total’) is a numeric vector contains the total number of reads of each sample.
The last element (‘sequence’) a data frame contains the sequence of each sgRNA in the library.
</p>


<h3>Examples</h3>

<pre><code class="language-R">library(CB2)
library(magrittr)
library(tibble)
library(dplyr)
library(glue)
FASTA &lt;- system.file("extdata", "toydata", "small_sample.fasta", package = "CB2")
ex_path &lt;- system.file("extdata", "toydata", package = "CB2")

df_design &lt;- tribble(
  ~group, ~sample_name,
  "Base", "Base1",  
  "Base", "Base2", 
  "High", "High1",
  "High", "High2") %&gt;% 
    mutate(fastq_path = glue("{ex_path}/{sample_name}.fastq"))

cb2_count &lt;- run_sgrna_quant(FASTA, df_design)

</code></pre>


</div>