<div class="container">

<table style="width: 100%;"><tr>
<td>get_genetics_by_sample</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Get All Genomic Information By Sample IDs</h2>

<h3>Description</h3>

<p>Get All Genomic Information By Sample IDs
</p>


<h3>Usage</h3>

<pre><code class="language-R">get_genetics_by_sample(
  sample_id = NULL,
  study_id = NULL,
  sample_study_pairs = NULL,
  genes = NULL,
  panel = NULL,
  add_hugo = TRUE,
  base_url = NULL,
  return_segments = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>sample_id</code></td>
<td>
<p>a vector of sample IDs (character)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>study_id</code></td>
<td>
<p>A string indicating the study ID from which to pull data. If no study ID, will
guess the study ID based on your URL and inform. Only 1 study ID can be passed. If mutations/cna from
more than 1 study needed, see <code>sample_study_pairs</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sample_study_pairs</code></td>
<td>
<p>A dataframe with columns: <code>sample_id</code>, <code>study_id</code> and <code>molecular_profile_id</code> (optional). Variations in capitalization of column names are accepted.
This can be used in place of <code>sample_id</code>, <code>study_id</code>, <code>molecular_profile_id</code> arguments above if you
need to pull samples from several different studies at once. If passed this will take overwrite <code>sample_id</code>, <code>study_id</code>, <code>molecular_profile_id</code> if also passed.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genes</code></td>
<td>
<p>A vector of Entrez ids or Hugo symbols. If Hugo symbols are supplied, they will be converted to entrez ids using the <code>get_entrez_id()</code> function.
If <code>panel</code> and <code>genes</code> are both supplied, genes from both arguments will be returned. If both are NULL (default), it will return gene results for all available genomic data for that sample.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>panel</code></td>
<td>
<p>One or more panel IDs to query (e.g. 'IMPACT468').
If <code>panel</code> and <code>genes</code> are both supplied, genes from both arguments will be returned. If both are NULL (default), it will return gene results for all available genomic data for that sample.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>add_hugo</code></td>
<td>
<p>Logical indicating whether <code>HugoGeneSymbol</code> should be added to your resulting data frame, if not already present in raw API results.
Argument is <code>TRUE</code> by default. If <code>FALSE</code>, results will be returned as is (i.e. any existing Hugo Symbol columns in raw results will not be removed).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>base_url</code></td>
<td>
<p>The database URL to query
If <code>NULL</code> will default to URL set with <code style="white-space: pre;">⁠set_cbioportal_db(&lt;your_db&gt;)⁠</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>return_segments</code></td>
<td>
<p>Default is <code>FALSE</code> where copy number segmentation data won't be returned in addition to the mutation, cna and structural variant data.
<code>TRUE</code> will return any available segmentation data with results.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A list of mutations, cna and structural variants (including fusions), if available. Will also return copy number segmentation data if <code>return_segments = TRUE</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
get_genetics_by_sample(sample_id = c("TCGA-OR-A5J2-01","TCGA-OR-A5J6-01"),
 study_id = "acc_tcga",
 return_segments = TRUE)

## End(Not run)
</code></pre>


</div>