<div class="container">

<table style="width: 100%;"><tr>
<td>ccRemover</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Removes the effect of the cell-cycle</h2>

<h3>Description</h3>

<p><code>ccRemover</code> returns a data matrix with the effects of the cell-cycle
removed.
</p>


<h3>Usage</h3>

<pre><code class="language-R">ccRemover(dat, cutoff = 3, max_it = 4, nboot = 200, ntop = 10,
  bar = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>dat</code></td>
<td>
<p>A list containing a data frame , <code>x</code>, that contains gene expression
measurements with each column representing a sample and each row
representing a gene and a logical vector, <code>if_cc</code>, that indicates
which of the genes/rows are related to the cell-cycle or factor of interest.
</p>
<p>It is recommended that the elements of x are log-transformed and centered
for each gene. For example if <code>x</code> contains TPM measurements then we
suggest the following two-steps:
Step 1: <code>dat$x &lt;- log(dat$x + 1)</code>
Step 2: <code>dat$x</code> - rowMeans(dat$x)
ccRemover requires that the samples have been properly normalized for
sequencing depth and we recommend doing so prior to applying the above steps.
</p>
<p>The <code>if_cc</code> vector must be the same length as the number of rows in
<code>x</code> and have elements equal to <code>TRUE</code> for genes which are related
to the cell-cycle and and elements equal to <code>FALSE</code> for genes which
are unrelated to the cell-cycle.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cutoff</code></td>
<td>
<p>The significance cutoff for identifying sources of variation
related to the cell-cycle. The default value is 3, which roughly corresponds
to a p-value of 0.01.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>max_it</code></td>
<td>
<p>The maximum number of iterations for the algorithm. The
default value is 4.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nboot</code></td>
<td>
<p>The number of bootstrap repititions to be carried out on each
iteration to determine the significance of each component.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ntop</code></td>
<td>
<p>The number of components considered tested at each iteration as
cell-cycle effects. The default value if 10</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bar</code></td>
<td>
<p>Whether to display a progress bar or not. The progress bar will
not work in R-markdown enviornments so this option may be turned off. The
default value is <code>TRUE</code>.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Implements the algorithm described in Barron, M. &amp; Li, J.
"Identifying and removing the cell-cycle effect from scRNA-Sequencing data"
(2016), Scientific Reports. This function takes a normalized,
log-transformed and centered matrix of scRNA-seq  expression data
and a list of genes which are known to be related to the cell-cycle effect.
It then captures the main sources of variation in the data and determines
which of these are related to the cell-cycle before removing those that are.
Please see the original manuscript for further details.
</p>


<h3>Value</h3>

<p>A data matrix with the effects of the cell-cycle removed.
</p>


<h3>Examples</h3>

<pre><code class="language-R">set.seed(10)
# Load in example data
data(t.cell_data)
head(t.cell_data[,1:5])
# Center data and select small sample for example
t_cell_data_cen &lt;- t(scale(t(t.cell_data[,1:20]), center=TRUE, scale=FALSE))
# Extract gene names
gene_names &lt;- rownames(t_cell_data_cen)
# Determine which genes are annotated to the cell-cycle
cell_cycle_gene_indices &lt;- gene_indexer(gene_names,
species = "mouse", name_type = "symbol")
# Create "if_cc" vector
if_cc &lt;- rep(FALSE,nrow(t_cell_data_cen))
if_cc[cell_cycle_gene_indices] &lt;- TRUE
# Move data into list
dat &lt;- list(x=t_cell_data_cen, if_cc=if_cc)
# Run ccRemover
## Not run: 
 xhat &lt;- ccRemover(dat, cutoff = 3, max_it = 4, nboot = 200, ntop = 10)

## End(Not run)
# Run ccRemover with reduced bootstrap repetitions for example only
xhat &lt;- ccRemover(dat, cutoff = 3, max_it = 4, nboot = 20, ntop = 10)
head(xhat[,1:5])
# Run ccRemover with more compoents considered
## Not run: 
xhat &lt;- ccRemover(dat, cutoff = 3, max_it = 4, nboot = 200, ntop = 15)
 
## End(Not run)

</code></pre>


</div>