<div class="container">

<table style="width: 100%;"><tr>
<td>discovery.log</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>

Calculates discovery probability by RNA in situ hybridisation given a sequencing signal
</h2>

<h3>Description</h3>


<p>A set of functions with different assumptions on the probability of RNA in situ staining, given a sequencing coverage.
</p>


<h3>Usage</h3>

<pre><code class="language-R">discovery.log(seq, saturate = 60, bias = 0.01)
discovery.linear(seq, saturate = 60, bias = 0.01)
discovery.identic(seq, saturate=Inf, bias=0)

</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>seq</code></td>
<td>


<p>A vector of sequencing FPKMs.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>saturate</code></td>
<td>


<p>FPKM value from which on maximum discovery probability (=0.99) is assumed (i.e. almost certain true positives).
Value of 60 is default, may need adjustment to sequencing coverage.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bias</code></td>
<td>


<p>Positive staining probability of 0 FPKM transcripts (i.e. false positives). Must be &gt;0.
Default is 0.01, an empirically determined value.
</p>
</td>
</tr>
</table>
<h3>Details</h3>



<ol>
<li> <p><strong>discovery.log</strong> Uses a logarithmic saturation function for discovery probabilities. This relationship was empirically determined from sequencing and hybridisation data.
</p>
</li>
<li> <p><strong>discovery.linear</strong> Linear saturation function for discovery probabilities.
</p>
</li>
<li> <p><strong>discovery.identic</strong> Passes input through. Useful for comparing RNASeq Vs. RNASeq data. Also for cases when the discovery probability for each transcript has been already determined in some other way.
</p>
</li>
</ol>
<h3>Value</h3>






<p>A vector of probabilities. Element names are preserved.
</p>


<h3>See Also</h3>


<p><code>seqVsInsitu</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">plot(0:80, discovery.log(0:80),
  ylim=c(0,1.1), type="l",
  xlab="FPKM", ylab="p(discovery insitu hybridization)")

plot(0:80, discovery.linear(0:80),
  ylim=c(0,1.1), type="l",
  xlab="FPKM", ylab="p(discovery insitu hybridization)")

</code></pre>


</div>