<div class="container">

<table style="width: 100%;"><tr>
<td>vncMedianCoverage.tsv</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>

Drosophila melanogaster embryo ventral nerve cord RNASeq coverage
</h2>

<h3>Description</h3>


<p>Median RNAseq read coverages from 3 dissected embryonic (stage 11) fruit fly ventral nerve cords. The sequencing coverages are measured within the probing intervals of high-confidence BDGP insitu probes, as described in cellOrigins-package.
</p>


<h3>Format</h3>

<p>The format is: probe name, coverage, chromosome, probe beginn, probe end, strand.
</p>


<h3>Source</h3>


<p>Molnar, D 2015, 'Single embryo-single organ transcriptomics of Drosophila embryos', PhD thesis, University of Cambridge.
</p>


<h3>Examples</h3>

<pre><code class="language-R">  fpath &lt;- system.file("extdata", "vncMedianCoverage.tsv", package="cellOrigins")
vncExpression &lt;- read.delim(file = fpath, header=FALSE, as.is=TRUE)
</code></pre>


</div>