<div class="container">

<table style="width: 100%;"><tr>
<td>ceRNA.cernia</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
identifying miRNA sponge interactions using ceRNA.cernia function.
</h2>

<h3>Description</h3>

<p>identifying miRNA sponge interactions using ceRNA.cernia.In this function We
implement cernia score to identify miRNA sponge interactions.
The fraction of common miRNAs;
1. The fraction of common miRNAs;
2. The density of the MREs for all shared miRNAs;
3. The distribution of MREs of the putative ceRNAs;
4. The relation between the overall number of MREs for a putative ceRNA, 
compared with the number of miRNAs that yield these MREs;
5. The density of the hybridization energies related to MREs for
all shared miRNAs;
6. The DT-Hybrid recommendation scores; 
7. The pairwise Pearson correlation between putative ceRNA
expressions.
</p>


<h3>Usage</h3>

<pre><code class="language-R">ceRNA.cernia(miRtar, targetce = NULL, geneexp, miRexp, mres, numMIR = 3, cor_cutoff = 0,
             s_cutoff = 0.5)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>miRtar</code></td>
<td>

<p>A data frame representing the relationship between miRNA and target. The data frame contains the name of 
the miRNA and target regulatory relationship.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>targetce</code></td>
<td>

<p>a character string (vector) specifying candidate target name to analyse (default (targetce = NULL)).
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>geneexp</code></td>
<td>

<p>An input target expression data frame, the columns are genes and
the rows are samples.The expression value may be  gene expression ,non-coding RNA expression  or 
expression values of circRNAs and so on.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>miRexp</code></td>
<td>

<p>An input miRNA expression data frame, the columns are miRNA and
the rows are samples.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mres</code></td>
<td>

<p>miRNA response elements (mres) data frame, each row
contains five elements: mirna, target, energy, gap_l, gap_r. 
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>numMIR</code></td>
<td>

<p>a numeric vector that specify the minimum number of 2 gene-shared miRNAs.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cor_cutoff</code></td>
<td>

<p>a numeric vector of the form method specifying threshold between
ceRNA interactions correlation,default(cor_cutoff=0).
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>s_cutoff</code></td>
<td>

<p>the threshold of seven comprehensive scores. 
</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Note:All the arguments without default value must be assigned.The miRNA 
in the file of the target-miRNA regulatory relationship pair should also be 
present in the expression profile data file.
Internal functions (parMM, graphWeights, recommendation, dtHybrid) of cernia method are from
the website: https://github.com/dsardina/cernia Copyright 2016 Rosalba Giugno Licensed under
the Apache License, Version 2.0 (the 'License')
</p>


<h3>Value</h3>

<p>A list of identified miRNA sponge interactions containing following components:
</p>

<ul>
<li> <p><code>targetce</code> represented target names,respectively.
</p>
</li>
<li> <p><code>anotherce</code> names of modulators that another target(modulators) constitutes a ceRNA interaction relation.
</p>
</li>
<li> <p><code>miRNAs_num</code> names of miRNA in the triplet.
</p>
</li>
<li> <p><code>Score 1</code> The fraction of common miRNAs;
</p>
</li>
<li> <p><code>Score 2</code> The density of the MREs for all shared miRNAs;
</p>
</li>
<li> <p><code>Score 3</code> The distribution of MREs of the putative ceRNAs;
</p>
</li>
<li> <p><code>Score 4</code> The relation between the overall number of MREs for a putative ceRNA, 
compared with the number of miRNAs that yield these MREs;
</p>
</li>
<li> <p><code>Score 5</code> The density of the hybridization energies related to MREs for
all shared miRNAs;
</p>
</li>
<li> <p><code>Score 6</code> The DT-Hybrid recommendation scores; 
</p>
</li>
<li> <p><code>Score 7</code> The pairwise Pearson correlation between putative ceRNA
expressions.
</p>
</li>
<li> <p><code>Normalized score</code> combine the sum of 7 scores and standardize by z-score.
</p>
</li>
</ul>
<h3>References</h3>

<p>Sardina D S , Alaimo S , Ferro A , et al. A novel computational method for inferring 
competing endogenous interactions[J]. Briefings in Bioinformatics, 2016:bbw084.
</p>


<h3>Examples</h3>

<pre><code class="language-R">##identifying miRNA sponge interactions.
##Here we take six candidate targets(modulators) and corresponding expression 
##data for example,Specify target(PTEN) to predict ceRNA interaction.
ceRNA.cernia(miRtar=dataset[["miRtar"]], targetce = "PTEN", geneexp=dataset[["geneexp"]],
          numMIR = 1, miRexp=dataset[["miRexp"]], mres=dataset[["mres"]], 
		  cor_cutoff = 0.2, s_cutoff = 0.5) 
</code></pre>


</div>