<div class="container">

<table style="width: 100%;"><tr>
<td>MKMEP</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Megakaryocyte-Erythroid Progenitors
</h2>

<h3>Description</h3>

<p>In a study about the formation of blood cells, Psaila <em>and others</em> (2016) have recently identified three distinct sub-populations of cells, which are all derived from hematopoietic stem cells through cell differentiation. One of these sub-populations, denoted by MK-MEP, is a previously unknown, rare population of cells that are bipotent but primarily generate megakaryocytic progeny.
</p>
<p>Multiplex RT-qPCR has been used to profile 63 genes and 48 single human MK-MEP cells. RT-qPCR data are typically right-censored with a limit of detection fixed by the manufacturer to 40. Raw data have been mean normalized using the method proposed in Pipelers <em>and others</em> (2017). See Section 5 in Augugliaro <em>and others</em> (2018) for more details.
</p>
<p>‘<code>MKMEP</code>’ is a matrix containing a subset of the data available from Psaila <em>and others</em> (2016).
</p>


<h3>Usage</h3>

<pre><code class="language-R">data("MKMEP")</code></pre>


<h3>References</h3>

<p>Augugliaro L., Sottile G., Wit E.C., and Vinciotti V. (2023) &lt;<a href="https://doi.org/10.18637/jss.v105.i01">doi:10.18637/jss.v105.i01</a>&gt;.
cglasso: An R Package for Conditional Graphical Lasso Inference with Censored and Missing Values.
<em>Journal of Statistical Software</em> <b>105</b>(1), 1–58.
</p>
<p>Augugliaro, L., Sottile, G., and Vinciotti, V. (2020) &lt;<a href="https://doi.org/10.1007/s11222-020-09945-7">doi:10.1007/s11222-020-09945-7</a>&gt;.
The conditional censored graphical lasso estimator.
<em>Statistics and Computing</em> <b>30</b>, 1273–1289.
</p>
<p>Augugliaro, L., Abbruzzo, A., and Vinciotti, V. (2020) &lt;<a href="https://doi.org/10.1093/biostatistics/kxy043">doi:10.1093/biostatistics/kxy043</a>&gt;.
<code class="reqn">\ell_1</code>-Penalized censored Gaussian graphical model.
<em>Biostatistics</em> <b>21</b>, e1–e16.
</p>
<p>Pipelers, P., Clement, L., Vynck, M., Hellemans, J., Vandesompele, J. and Thas, O. (2017) &lt;doi: 10.1371/journal.pone.0182832&gt;.
A unified censored normal regression model for qPCR differential gene expression analysis.
<em>PLoS One</em> <b>12</b>, e0182832.
</p>
<p>Psaila, B., Barkas, N., Iskander, D., Roy, A., Anderson, S., Ashley, N., Caputo, V. S., Lichtenberg, J., Loaiza, S., Bodine, D. M. <em>and others</em>. (2016) &lt;doi: 10.1186/s13059-016-0939-7&gt;
Single-cell profiling of human megakaryocyte-erythroid progenitors identifies distinct megakaryocyte and erythroid differentiation pathways.
<em>Genome Biology</em> <b>17</b>, 83–102.
</p>


<h3>See Also</h3>

<p><code>cglasso</code>, <code>to_graph</code>, and the method functions <code>summary</code>, <code>coef</code>, <code>plot</code>, <code>AIC.cglasso</code> and <code>BIC.cglasso</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">data("MKMEP")
is.matrix(MKMEP)
Z &lt;- datacggm(Y = MKMEP, up = 40)
str(Z)
is.datacggm(Z)
print(Z, width = 80L)
</code></pre>


</div>