<div class="container">

<table style="width: 100%;"><tr>
<td>map_taxa</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Map taxonomic names to NCBI or GTDB taxonomy</h2>

<h3>Description</h3>

<p>Maps taxonomic names to NCBI (RefSeq) or GTDB taxonomy by automatic matching of taxonomic names, with manual mappings for some groups.
</p>


<h3>Usage</h3>

<pre><code class="language-R">  map_taxa(taxacounts = NULL, refdb = "GTDB_214", taxon_AA = NULL, quiet = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>taxacounts</code></td>
<td>
<p>data frame with taxonomic name and abundances</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>refdb</code></td>
<td>
<p>character, name of reference database (‘<span class="samp">⁠GTDB_214⁠</span>’ or ‘<span class="samp">⁠RefSeq_206⁠</span>’)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>taxon_AA</code></td>
<td>
<p>data frame, amino acid compositions of taxa, used to bypass <code>refdb</code> specification</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>quiet</code></td>
<td>
<p>logical, suppress printed messages?</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function maps taxonomic names to the NCBI (RefSeq) or GTDB taxonomy.
<code>taxacounts</code> should be a data frame generated by either <code>read_RDP</code> or <code>ps_taxacounts</code>.
Input names are made by combining the taxonomic rank and name with an underscore separator (e.g. ‘<span class="samp">⁠genus_ Escherichia/Shigella⁠</span>’).
Input names are then matched to the taxa listed in ‘<span class="file">taxon_AA.csv.xz</span>’ found under ‘<span class="file">RefDB/RefSeq_206</span>’ or ‘<span class="file">RefDB/GTDB_214</span>’.
The <code>protein</code> and <code>organism</code> columns in these files hold the rank and taxon name extracted from the RefSeq or GTDB database.
Only exactly matching names are automatically mapped.
</p>
<p>For mapping to the NCBI (RefSeq) taxonomy, some group names are manually mapped as follows (see Dick and Tan, 2023):
</p>

<table>
<tr>
<td style="text-align: left;">
    <strong>RDP training set</strong> </td>
<td style="text-align: left;"> <strong>NCBI</strong> </td>
</tr>
<tr>
<td style="text-align: left;">
    genus_Escherichia/Shigella </td>
<td style="text-align: left;"> genus_Escherichia </td>
</tr>
<tr>
<td style="text-align: left;">
    phylum_Cyanobacteria/Chloroplast </td>
<td style="text-align: left;"> phylum_Cyanobacteria </td>
</tr>
<tr>
<td style="text-align: left;">
    genus_Marinimicrobia_genera_incertae_sedis </td>
<td style="text-align: left;"> species_Candidatus Marinimicrobia bacterium </td>
</tr>
<tr>
<td style="text-align: left;">
    class_Cyanobacteria </td>
<td style="text-align: left;"> phylum_Cyanobacteria </td>
</tr>
<tr>
<td style="text-align: left;">
    genus_Spartobacteria_genera_incertae_sedis </td>
<td style="text-align: left;"> species_Spartobacteria bacterium LR76 </td>
</tr>
<tr>
<td style="text-align: left;">
    class_Planctomycetacia </td>
<td style="text-align: left;"> class_Planctomycetia </td>
</tr>
<tr>
<td style="text-align: left;">
    class_Actinobacteria </td>
<td style="text-align: left;"> phylum_Actinobacteria </td>
</tr>
<tr>
<td style="text-align: left;">
    order_Rhizobiales </td>
<td style="text-align: left;"> order_Hyphomicrobiales </td>
</tr>
<tr>
<td style="text-align: left;">
    genus_Gp1 </td>
<td style="text-align: left;"> genus_Acidobacterium </td>
</tr>
<tr>
<td style="text-align: left;">
    genus_Gp6 </td>
<td style="text-align: left;"> genus_Luteitalea </td>
</tr>
<tr>
<td style="text-align: left;">
    genus_GpI </td>
<td style="text-align: left;"> genus_Nostoc </td>
</tr>
<tr>
<td style="text-align: left;">
    genus_GpIIa </td>
<td style="text-align: left;"> genus_Synechococcus </td>
</tr>
<tr>
<td style="text-align: left;">
    genus_GpVI </td>
<td style="text-align: left;"> genus_Pseudanabaena </td>
</tr>
<tr>
<td style="text-align: left;">
    family_Family II </td>
<td style="text-align: left;"> family_Synechococcaceae </td>
</tr>
<tr>
<td style="text-align: left;">
    genus_Subdivision3_genera_incertae_sedis </td>
<td style="text-align: left;"> family_Verrucomicrobia subdivision 3 </td>
</tr>
<tr>
<td style="text-align: left;">
    order_Clostridiales </td>
<td style="text-align: left;"> order_Eubacteriales </td>
</tr>
<tr>
<td style="text-align: left;">
    family_Ruminococcaceae </td>
<td style="text-align: left;"> family_Oscillospiraceae
</td>
</tr>
</table>
<p>To avoid manual mapping, GTDB can be used for both taxonomic assignemnts and reference proteomes.
Taxonomic assignments based on 16S rRNA sequences from GTDB can be made using training files for the RDP Classifier (<a href="https://doi.org/10.5281/zenodo.7633099">doi:10.5281/zenodo.7633099</a>) or <span class="pkg">dada2</span> (<a href="https://doi.org/10.5281/zenodo.2541238">doi:10.5281/zenodo.2541238</a>) (make sure to choose the appropriate GTDB version).
Example files created using the RDP Classifier are provided under ‘<span class="file">extdata/RDP-GTDB_207</span>’.
An example dataset created with DADA2 is <code>data(mouse.GTDB_214)</code>; this is a <code>phyloseq-class</code> object that can be processed with functions described at <code>physeq</code>.
</p>
<p>Change <code>quiet</code> to TRUE to suppress printing of messages about manual mappings, most abundant unmapped groups, and overall percentage of mapped names.
</p>


<h3>Value</h3>

<p>Integer vector with length equal to number of rows of <code>taxacounts</code>.
Values are rownumbers in the data frame generated by reading <code>taxon_AA.csv.xz</code>, or NA for no matching taxon.
Attributes <code>unmapped_groups</code> and <code>unmapped_percent</code> have the input names of unmapped groups and their percentage of the total classification count.
</p>


<h3>References</h3>

<p>Dick JM, Tan J. 2023. Chemical links between redox conditions and estimated community proteomes from 16S rRNA and reference protein sequences. <em>Microbial Ecology</em> <b>85</b>: 1338–1355. <a href="https://doi.org/10.1007/s00248-022-01988-9">doi:10.1007/s00248-022-01988-9</a>
</p>


<h3>Examples</h3>

<pre><code class="language-R"># Mapping taxonomic classifications from RDP training set to NCBI taxonomy
file &lt;- system.file("extdata/RDP/SMS+12.tab.xz", package = "chem16S")
RDP &lt;- read_RDP(file)
map &lt;- map_taxa(RDP, refdb = "RefSeq_206")
# About 24% of classifications are unmapped
sum(attributes(map)$unmapped_percent)

# Mapping from GTDB training set to GTDB taxonomy
file &lt;- system.file("extdata/RDP-GTDB_207/SMS+12.tab.xz", package = "chem16S")
RDP.GTDB &lt;- read_RDP(file)
map.GTDB &lt;- map_taxa(RDP.GTDB)
# The classifications were made with GTDB r207, but the reference database 
# in chem16S has been updated to GTDB r214, so the mapping rate is less than 100 %
sum(attributes(map.GTDB)$unmapped_percent)
</code></pre>


</div>