<div class="container">

<table style="width: 100%;"><tr>
<td>prepare.data</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>prepare.data</h2>

<h3>Description</h3>

<p>Prepare data for running interaction calling. Takes a BAM file and baits and restriction fragments as input, and returns a data table with data ready for analysis.
</p>


<h3>Usage</h3>

<pre><code class="language-R">prepare.data(
  bam,
  baits,
  fragments,
  replicate.merging.method = "sum",
  include.zeros = c("none", "cis", "all"),
  remove.adjacent = FALSE,
  temp.directory = NULL,
  keep.files = FALSE,
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>bam</code></td>
<td>
<p>Path to a BAM file</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>baits</code></td>
<td>
<p>Path to a BED file containing the baits</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fragments</code></td>
<td>
<p>Path to a BED file containing all restriction fragments in the genome</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>replicate.merging.method</code></td>
<td>
<p>Method that should be used for merging replicates, if applicable</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>include.zeros</code></td>
<td>
<p>String specifying what zero counts to include. Options are none (default), cis, and all.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>remove.adjacent</code></td>
<td>
<p>Logical indicating whether to remove all reads mapping to adjacent restriction fragments.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>temp.directory</code></td>
<td>
<p>Directory where temporary files should be stored. Defaults to current directory.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>keep.files</code></td>
<td>
<p>Logical indicating whether to keep temporary files</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>Logical indicating whether to print progress reports.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Data table object with columns
</p>
<table>
<tr style="vertical-align: top;">
<td><code>target.id</code></td>
<td>
<p>String in chrN:start-end format identifying target fragment</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bait.id</code></td>
<td>
<p>String in chrN:start-end format identifying bait fragment</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>target.chr</code></td>
<td>
<p>Chromosome of target fragment</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>target.start</code></td>
<td>
<p>Start coordinate of target fragment (zero-based)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>target.end</code></td>
<td>
<p>End coordinate of target fragment</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bait.chr</code></td>
<td>
<p>Chromosome of bait fragment</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bait.start</code></td>
<td>
<p>Start coordinate of bait fragment (zero-based)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bait.end</code></td>
<td>
<p>End coordinate of bait fragment</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bait.to.bait</code></td>
<td>
<p>Boolean indicating if the interaction is bait-to-bait (i.e. the fragment listed as target is also a bait)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>count</code></td>
<td>
<p>The number of reads linking the two fragments</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bait.trans.count</code></td>
<td>
<p>The number of reads linking the bait to fragments in trans (a measure of "interactibility")</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>target.trans.count</code></td>
<td>
<p>The number of reads linking the target to fragments in trans (a measure of "interactibility")</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>distance</code></td>
<td>
<p>Distance between the midpoints of the bait and target fragments (basepairs). NA for trans interactions</p>
</td>
</tr>
</table>
<h3>Examples</h3>

<pre><code class="language-R">
if( bedtools.installed() ) {
  bam &lt;- system.file('extdata', 'Bre80_2q35.bam', package = 'chicane');
  baits &lt;- system.file('extdata', '2q35.bed', package = 'chicane');
  fragments &lt;- system.file('extdata', 'GRCh38_HindIII_chr2.bed.gz', package = 'chicane');
  input.data &lt;- prepare.data(
	bam = bam, 
	baits = baits, 
	fragments = fragments
	);
 }


</code></pre>


</div>