<div class="container">

<table style="width: 100%;"><tr>
<td>create.locus.plot</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>create.locus.plot</h2>

<h3>Description</h3>

<p>Create a file compatible with WashU Epigenome Browser from CHiCANE interaction calls.
</p>


<h3>Usage</h3>

<pre><code class="language-R">create.locus.plot(
  genome = "hg38",
  chr = NULL,
  start = NULL,
  end = NULL,
  gene.data = NULL,
  genomic.features = NULL,
  feature.name = NULL,
  fdr.filter = 0.05,
  interaction.data = NULL,
  file.name = NULL,
  height = 5.5,
  width = 8.5,
  track.heights = c(0.2, 0.5, 0.8, 0.5, 1.5, 2),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>genome</code></td>
<td>
<p>Name of genome build (e.g. 'hg38' or 'hg37')</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chr</code></td>
<td>
<p>Chromosome number for desired locus including 'chr' (e.g. 'chr1')</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>start</code></td>
<td>
<p>Start coordinate of desired locus</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>end</code></td>
<td>
<p>End coordinate of desired locus</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gene.data</code></td>
<td>
<p>Path to chosen genome annotation file in .gtf format</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genomic.features</code></td>
<td>
<p>Path to BED file with coordinates of desired feature track</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>feature.name</code></td>
<td>
<p>Title to appear above genomic features</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fdr.filter</code></td>
<td>
<p>Q-value filter threshold for interaction calls to be included</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>interaction.data</code></td>
<td>
<p>Path to unfiltered CHiCANE calls output</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>file.name</code></td>
<td>
<p>Path to output file</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>height</code></td>
<td>
<p>Height in inches for desired plot</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>width</code></td>
<td>
<p>Width in inches of desired plot</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>track.heights</code></td>
<td>
<p>Vector of length 6 indicating desired height of individual tracks</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Any additional parameters to <code>Gviz::plotTracks</code></p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>TRUE if plot was successfully created
</p>


<h3>Author(s)</h3>

<p>Andrea Gillespie, Syed Haider
</p>


<h3>Examples</h3>

<pre><code class="language-R"># In order to conserve memory only significant interactions are included in example 
# interaction.data file. However, in order to show raw counts, unfiltered calls should be 
# included and only significant interactions (as set by fdr.filter) wil be displayed

gene.data &lt;- system.file('extdata', 'gencode_2q35.gtf', package = 'chicane');
genomic.features &lt;- system.file('extdata', '2q35.bed', package = 'chicane');
interaction.data &lt;- system.file(
  'extdata', 'T47D_2q35_filtered_chicane_calls.txt', 
  package = 'chicane'
  );
file.name &lt;- file.path(tempdir(), "chr2_interactions.pdf");


create.locus.plot(
   genome = 'hg38', 
   chr = 'chr2', 
   start = 216600000,
   end = 217200000,
   gene.data = gene.data,
   genomic.features = genomic.features,
   feature.name = 'baits',
   interaction.data = interaction.data,
   file.name = file.name,
   collapseTranscripts = TRUE,
   shape = "arrow"
   );
 

</code></pre>


</div>