<div class="container">

<table style="width: 100%;"><tr>
<td>chromoMap</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>render interactive chromosome plots of any living organism and annotate elements</h2>

<h3>Description</h3>

<p>render an interactive graphics visualization of entire chromosomes
or chromosomal regions of any living organism. Chromosomal elements such as genes 
can be annotated easily using this tool.
</p>
<p>required for creating widgets
</p>


<h3>Usage</h3>

<pre><code class="language-R">chromoMap(
  ch.files,
  data.files,
  title = c(),
  ch_gap = 5,
  ploidy = 1,
  top_margin = 25,
  left_margin = 50,
  chr_width = 15,
  chr_length = 4,
  chr_color = c("black"),
  data_based_color_map = FALSE,
  segment_annotation = FALSE,
  lg_x = 0,
  lg_y = 0,
  data_type = c("numeric", "categorical"),
  labels = FALSE,
  canvas_width = NULL,
  canvas_height = NULL,
  data_colors = list(),
  anno_col = c("#10B85F"),
  chr_text = c(TRUE),
  discrete.domain = NULL,
  legend = c(FALSE),
  hlinks = FALSE,
  aggregate_func = c("avg"),
  plots = c("none"),
  tag_filter = list(c("none", 0)),
  plot_height = c(30),
  plot_ticks = c(4),
  plot_color = c("blue"),
  plot_y_domain = list(c(0, 0)),
  ch2D.colors = NULL,
  ch2D.cat.order = NULL,
  ch2D.lg_x = 0,
  ch2D.lg_y = 0,
  ref_line = c(FALSE),
  refl_pos = c(0),
  refl_color = c("grey"),
  refl_stroke_w = c(2),
  tagColor = c("red"),
  heat_map = c(TRUE),
  text_font_size = c(10),
  chr_curve = 5,
  title_font_size = 12,
  label_font = 9,
  label_angle = -90,
  vertical_grid = FALSE,
  grid_array = c(0, 5000, 10000),
  grid_color = "grey",
  grid_text = NULL,
  grid_text_size = 12,
  grid_text_y = 20,
  plot_filter = list(c("none", 0)),
  id = c("chromap"),
  region = NULL,
  show.links = FALSE,
  loci_links = "none",
  directed.edges = F,
  y_chr_scale = 0,
  links.colors = NULL,
  links.lg_x = 0,
  links.lg_y = 0,
  n_win.factor = 1,
  chr.scale.ticks = 5,
  export.options = F,
  fixed.window = F,
  window.size = NULL,
  win.summary.display = F,
  remove.last.window = T,
  guides = F,
  guides_color = "lightgrey",
  ann.h = 1,
  chr.2D.plot = F,
  display.chr = T,
  plot.shift = c(1),
  plot.legend.labels = c(""),
  cat.legend.label = "",
  plot.y.labels = c(""),
  plot.y.lab.x = 10,
  plot.y.lab.y = 0,
  plot.y.lab.size = 15,
  scale.suffix = "bp",
  numeric.domain = NULL,
  interactivity = T
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>ch.files</code></td>
<td>
<p>filename(s) as character vector OR list of data.frames containing co-ordinates of the chromosomes to render</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>data.files</code></td>
<td>
<p>filename(s) as character vector OR list of data.frames containing data to annotate on the chromosomes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>title</code></td>
<td>
<p>a character string to be used as a title in plot</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ch_gap</code></td>
<td>
<p>provide spacing between chromosomes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ploidy</code></td>
<td>
<p>specify the number of sets of chromsomes being passed.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>top_margin</code></td>
<td>
<p>specify the margin from top of the plot</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>left_margin</code></td>
<td>
<p>specify the margin from the left of the plot</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chr_width</code></td>
<td>
<p>specify the width of each chromsome</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chr_length</code></td>
<td>
<p>specify the length of each chromsome.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chr_color</code></td>
<td>
<p>a vector specifying the color of each chromsome in a set. A color
can be assigned to each set by passing a different color values as vector</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>data_based_color_map</code></td>
<td>
<p>a boolean to tell the plot to use the data provided in file for
visualizing annotation</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>segment_annotation</code></td>
<td>
<p>a boolean to use segment-annotation algorithm</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>lg_x</code></td>
<td>
<p>specify the x or horizontal distance of the legend from origin(bottom right corner)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>lg_y</code></td>
<td>
<p>specify the y or vertical distnce of the legend from the origin</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>data_type</code></td>
<td>
<p>specifying the data type of the data used. takes value either 'categorical' or
'numeric'</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>labels</code></td>
<td>
<p>a boolean to include labels in plot</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>canvas_width</code></td>
<td>
<p>width of the plot</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>canvas_height</code></td>
<td>
<p>height of the plot</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>data_colors</code></td>
<td>
<p>specify annotation colors for the data</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>anno_col</code></td>
<td>
<p>a vector to specify annotation color for each set.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chr_text</code></td>
<td>
<p>a boolean vector to enable or disable chromsome texts for each ploidy.set</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>discrete.domain</code></td>
<td>
<p>manually specify the order of categories.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>legend</code></td>
<td>
<p>a boolean vector to enable or disable legend for each set/ploidy</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>hlinks</code></td>
<td>
<p>a boolean to use hyperlinks supplied in data</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>aggregate_func</code></td>
<td>
<p>takes either 'sum' or 'avg' to specift aggregate function for each loci</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plots</code></td>
<td>
<p>specify the type of plot to visualize. takes either 'scatter' , 'bar' or 'tags'.(default: 'none')</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tag_filter</code></td>
<td>
<p>a list to specify the filter operation and operands for each ploidy.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot_height</code></td>
<td>
<p>specify plot height for each ploidy. default: c(30)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot_ticks</code></td>
<td>
<p>specify number of ticks for plot axis. default: c(4)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot_color</code></td>
<td>
<p>specify the plot color for each ploidy. default: c("blue")</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot_y_domain</code></td>
<td>
<p>specify plot y-axis domain. default: list(c(0,0))</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ch2D.colors</code></td>
<td>
<p>specify the group colors for visualizing categories on 2D chromosome plots</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ch2D.cat.order</code></td>
<td>
<p>manually setting the order of categories for 2D-Chromsome plot</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ch2D.lg_x</code></td>
<td>
<p>specify the x or horizontal distance of 2D plot legend from the origin(bottom right corner)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ch2D.lg_y</code></td>
<td>
<p>specify the y or vertical distance of 2D plot legend</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ref_line</code></td>
<td>
<p>a boolean to use horizontal reference line in plot. default: c(FALSE)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>refl_pos</code></td>
<td>
<p>specify the position of reference line. default: c(0)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>refl_color</code></td>
<td>
<p>specify the color of the reference line. default: c("grey")</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>refl_stroke_w</code></td>
<td>
<p>specify the stroke width of the reference line. default: c(2)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tagColor</code></td>
<td>
<p>specify the color of tags. default: c("red")</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>heat_map</code></td>
<td>
<p>a boolean to use if chromosome heatmaps are shown. default: c(TRUE),</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>text_font_size</code></td>
<td>
<p>specify chromosome text font-size. default: c(10)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chr_curve</code></td>
<td>
<p>specify the chromosome curves at the telomeres or centromere loci. default:5</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>title_font_size</code></td>
<td>
<p>specify the font-size of the title. default:12</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>label_font</code></td>
<td>
<p>specify the font-size of the labels. default:9</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>label_angle</code></td>
<td>
<p>specify the angle of rotation of labels. default: -90</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>vertical_grid</code></td>
<td>
<p>a boolean to use vertical grid lines. default: FALSE</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>grid_array</code></td>
<td>
<p>specify the position(s) of grid line(s) in bp to highlight locations across genome. default: c(0,5000,10000)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>grid_color</code></td>
<td>
<p>specify the color of the grid lines. default: "grey"</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>grid_text</code></td>
<td>
<p>specify the text to be attached at the top end of gridlines</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>grid_text_size</code></td>
<td>
<p>specify the font-size of the text</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>grid_text_y</code></td>
<td>
<p>specify the y-distance (from top) for the text</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot_filter</code></td>
<td>
<p>a list specify the plot filter operation, operands, and filter-color for each ploidy.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>id</code></td>
<td>
<p>specify a unique id doe chromoMap plot. default: c("chromap")</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>region</code></td>
<td>
<p>specify the region of interest for chromosome(s) for zoom-in. Format: "chrName:Ploidy:Start:Stop"</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>show.links</code></td>
<td>
<p>a boolean to specify whether links are visualized. default: FALSE</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>loci_links</code></td>
<td>
<p>a character vector specifying file name or a data.frame for links input data</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>directed.edges</code></td>
<td>
<p>a boolean to visualize directed edges</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>y_chr_scale</code></td>
<td>
<p>adjust the chromosome scale along y-axis</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>links.colors</code></td>
<td>
<p>specify the links colors</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>links.lg_x</code></td>
<td>
<p>specify x or horizontal distance of links legend from the origin</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>links.lg_y</code></td>
<td>
<p>specify y or vertical distance of links</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n_win.factor</code></td>
<td>
<p>specify the factor by which the chr will be scaled;increases number of windows (default:1)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chr.scale.ticks</code></td>
<td>
<p>specify the number of ticks for chr scale (default:5)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>export.options</code></td>
<td>
<p>boolean to include export buttons in the plot</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fixed.window</code></td>
<td>
<p>Boolean to specify wether to use fixed window visualization</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>window.size</code></td>
<td>
<p>specify the window size, if fixed.window is TRUE</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>win.summary.display</code></td>
<td>
<p>boolean to display window summary to console</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>remove.last.window</code></td>
<td>
<p>For fixed window analysis, boolean to specify whether to include last window of chromosomes</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>guides</code></td>
<td>
<p>boolean to display guides</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>guides_color</code></td>
<td>
<p>set guides color.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ann.h</code></td>
<td>
<p>set annotation bar height in 2D-Chromosome plot</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chr.2D.plot</code></td>
<td>
<p>boolean to specify visualize 2d Chromosome plot</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>display.chr</code></td>
<td>
<p>boolean to show.hide chromosome</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot.shift</code></td>
<td>
<p>shifting the plots in y direction in case hiding chromosomes</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot.legend.labels</code></td>
<td>
<p>specify plot legend labels</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cat.legend.label</code></td>
<td>
<p>specify categorical-data legends label</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot.y.labels</code></td>
<td>
<p>specify plots y-axis labels</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot.y.lab.x</code></td>
<td>
<p>adjust plot y labels in x-direction</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot.y.lab.y</code></td>
<td>
<p>adjust plot y labels in y-direction</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot.y.lab.size</code></td>
<td>
<p>set size of plot y labels</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>scale.suffix</code></td>
<td>
<p>set the suffix for chromosome scale(default:'bp')</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>numeric.domain</code></td>
<td>
<p>manually set data domain(min,max) for heat colors for numeric data</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>interactivity</code></td>
<td>
<p>boolean to enable/disable interactivity on chromosomes</p>
</td>
</tr>
</table>
<h3>Examples</h3>

<pre><code class="language-R">## Not run: 

library(chromoMap)

#simple annotations
chromoMap("chromosome_file.txt","annotation_file.txt")

#polyploidy example
chromoMap(c("chromosome_set1.txt","chromosome_set2.txt"),
          c("annotation_set1.txt","annotation_set2.txt"), ploidy=2)

#plotting group annotation
chromoMap("chromosome_file.txt","annotation_file.txt",
           data_base_color_map=T, data_type="categorical")

#plotting chromsome heatmaps
chromoMap("chromosome_file.txt","annotation_file.txt",
           data_based_color_map=T, data_type="numeric")

#enabling hyperlinks
chromoMap("chromosome_file.txt","annotation_file.txt", hlinks=T)

#enabling labels
chromoMap("chromosome_file.txt","annotation_file.txt", labels=T)

#change chromosome color
chromoMap("chromosome_file.txt","annotation_file.txt", chr_color="red")


## End(Not run)





</code></pre>


</div>