<div class="container">

<table style="width: 100%;"><tr>
<td>generate_data</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Data generation function</h2>

<h3>Description</h3>

<p>Function to generate data with <code>n</code> observations of a primary
outcome <code>Y</code>, secondary outcome <code>K</code>, exposure <code>X</code>, and
measured as well as unmeasured confounders <code>L</code> and <code>U</code>, where
the primary outcome is a quantitative normally-distributed variable
(<code>setting</code> = <code>"GLM"</code>) or censored time-to-event outcome under
an accelerated failure time (AFT) model (<code>setting</code> = <code>"AFT"</code>).
Under the AFT setting, the observed time-to-event variable <code>T=exp(Y)</code>
as well as the censoring indicator <code>C</code> are also computed. <code>X</code>
is generated as a genetic exposure variable in the form of a single
nucleotide variant (SNV) in 0-1-2 additive coding with minor allele
frequency <code>maf</code>. <code>X</code> can be generated independently of <code>U</code>
(<code>X_orth_U</code> = <code>TRUE</code>) or dependent on <code>U</code>
(<code>X_orth_U</code> = <code>FALSE</code>). For more details regarding the underlying
model, see the vignette.
</p>


<h3>Usage</h3>

<pre><code class="language-R">generate_data(setting = "GLM", n = 1000, maf = 0.2, cens = 0.3,
  a = NULL, b = NULL, aXK = 0.2, aXY = 0.1, aXL = 0, aKY = 0.3,
  aLK = 0, aLY = 0, aUY = 0, aUL = 0, mu_X = NULL, sd_X = NULL,
  X_orth_U = TRUE, mu_U = 0, sd_U = 1, mu_K = 0, sd_K = 1, mu_L = 0,
  sd_L = 1, mu_Y = 0, sd_Y = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>setting</code></td>
<td>
<p>String with value <code>"GLM"</code> or <code>"AFT"</code> indicating
whether the primary outcome is generated as a
normally-distributed quantitative outcome (<code>"GLM"</code>) or
censored time-to-event outcome (<code>"AFT"</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n</code></td>
<td>
<p>Numeric. Sample size.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>maf</code></td>
<td>
<p>Numeric. Minor allele frequency of the genetic exposure variable.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cens</code></td>
<td>
<p>Numeric. Desired percentage of censored individuals and has to be
specified under the AFT setting. Note that the actual censoring
rate is generated through specification of the parameters
<code>a</code> and <code>b</code>, and <code>cens</code> is mostly used as a check
whether the desired censoring rate is obtained through <code>a</code>
and <code>b</code> (otherwise, a warning is issued).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>a</code></td>
<td>
<p>Integer for generating the desired censoring rate under the AFT
setting. Has to be specified under the AFT setting.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>b</code></td>
<td>
<p>Integer for generating the desired censoring rate under the AFT
setting. Has to be specified under the AFT setting.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>aXK</code></td>
<td>
<p>Numeric. Size of the effect of <code>X</code> on <code>K</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>aXY</code></td>
<td>
<p>Numeric. Size of the effect of <code>X</code> on <code>Y</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>aXL</code></td>
<td>
<p>Numeric. Size of the effect of <code>X</code> on <code>L</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>aKY</code></td>
<td>
<p>Numeric. Size of the effect of <code>K</code> on <code>Y</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>aLK</code></td>
<td>
<p>Numeric. Size of the effect of <code>L</code> on <code>K</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>aLY</code></td>
<td>
<p>Numeric. Size of the effect of <code>L</code> on <code>Y</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>aUY</code></td>
<td>
<p>Numeric. Size of the effect of <code>U</code> on <code>Y</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>aUL</code></td>
<td>
<p>Numeric. Size of the effect of <code>U</code> on <code>L</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mu_X</code></td>
<td>
<p>Numeric. Expected value of <code>X</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sd_X</code></td>
<td>
<p>Numeric. Standard deviation of <code>X</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>X_orth_U</code></td>
<td>
<p>Logical. Indicator whether <code>X</code> should be generated
independently of <code>U</code> (<code>X_orth_U</code> = <code>TRUE</code>)
or dependent on <code>U</code> (<code>X_orth_U</code> = <code>FALSE</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mu_U</code></td>
<td>
<p>Numeric. Expected value of <code>U</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sd_U</code></td>
<td>
<p>Numeric. Standard deviation of <code>U</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mu_K</code></td>
<td>
<p>Numeric. Expected value of <code>K</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sd_K</code></td>
<td>
<p>Numeric. Standard deviation of <code>K</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mu_L</code></td>
<td>
<p>Numeric. Expected value of <code>L</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sd_L</code></td>
<td>
<p>Numeric. Standard deviation of <code>L</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mu_Y</code></td>
<td>
<p>Numeric. Expected value of <code>Y</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sd_Y</code></td>
<td>
<p>Numeric. Standard deviation of <code>Y</code>.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A dataframe containing <code>n</code> observations of the variables <code>Y</code>,
<code>K</code>, <code>X</code>, <code>L</code>, <code>U</code>. Under the AFT setting,
<code>T=exp(Y)</code> and the censoring indicator <code>C</code> (0 = censored,
1 = uncensored) are also computed.
</p>


<h3>Examples</h3>

<pre><code class="language-R"># Generate data under the GLM setting with default values
dat_GLM &lt;- generate_data()
head(dat_GLM)

# Generate data under the AFT setting with default values
dat_AFT &lt;- generate_data(setting = "AFT", a = 0.2, b = 4.75)
head(dat_AFT)

</code></pre>


</div>