<div class="container">

<table style="width: 100%;"><tr>
<td>run_enrichment</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>run_enrichment</h2>

<h3>Description</h3>

<p>This function is run, if the enrichment pipeline wants to be called afterwards.
Setting reference genome to the same genome which cinaR was run should be given to this function!
</p>


<h3>Usage</h3>

<pre><code class="language-R">run_enrichment(
  results,
  geneset = NULL,
  experiment.type = "ATAC-Seq",
  reference.genome = NULL,
  enrichment.method = NULL,
  enrichment.FDR.cutoff = 1,
  background.genes.size = 20000,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>results</code></td>
<td>
<p>list, DA peaks list for different contrasts</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>geneset</code></td>
<td>
<p>Pathways to be used in enrichment analyses. If not set vp2008 (Chaussabel, 2008)
immune modules will be used. This can be set to any geneset using 'read.gmt' function from 'qusage'
package. Different modules are available: https://www.gsea-msigdb.org/gsea/downloads.jsp.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>experiment.type</code></td>
<td>
<p>The type of experiment either set to "ATAC-Seq" or "RNA-Seq"</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>reference.genome</code></td>
<td>
<p>genome of interested species. It should be 'hg38', 'hg19' or 'mm10'.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>enrichment.method</code></td>
<td>
<p>There are two methodologies for enrichment analyses,
Hyper-geometric p-value (HPEA) or Geneset Enrichment Analyses (GSEA).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>enrichment.FDR.cutoff</code></td>
<td>
<p>FDR cut-off for enriched terms, p-values
are corrected by Benjamini-Hochberg procedure</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>background.genes.size</code></td>
<td>
<p>number of background genes for hyper-geometric p-value
calculations. Default is 20,000.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>prints messages through running the pipeline</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>list, enrichment analyses results along with corresponding differential analyses outcomes
</p>


<h3>Examples</h3>

<pre><code class="language-R">
library(cinaR)
data(atac_seq_consensus_bm) # calls 'bed'

# a vector for comparing the examples
contrasts &lt;- sapply(strsplit(colnames(bed), split = "-", fixed = TRUE),
                    function(x){x[1]})[4:25]

results &lt;- cinaR(bed, contrasts, reference.genome = "mm10", run.enrichment = FALSE)

results_with_enrichment &lt;- run_enrichment(results, reference.genome = "mm10")

</code></pre>


</div>