<div class="container">

<table style="width: 100%;"><tr>
<td>cinaR</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>cinaR</h2>

<h3>Description</h3>

<p>Runs differential analyses and enrichment pipelines
</p>


<h3>Usage</h3>

<pre><code class="language-R">cinaR(
  matrix,
  contrasts,
  experiment.type = "ATAC-Seq",
  DA.choice = 1,
  DA.fdr.threshold = 0.05,
  DA.lfc.threshold = 0,
  comparison.scheme = "OVO",
  save.DA.peaks = FALSE,
  DA.peaks.path = NULL,
  norm.method = "cpm",
  filter.method = "custom",
  library.threshold = 2,
  cpm.threshold = 1,
  TSS.threshold = 50000,
  show.annotation.pie = FALSE,
  reference.genome = NULL,
  batch.correction = FALSE,
  batch.information = NULL,
  additional.covariates = NULL,
  sv.number = NULL,
  run.enrichment = TRUE,
  enrichment.method = NULL,
  enrichment.FDR.cutoff = 1,
  background.genes.size = 20000,
  geneset = NULL,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>matrix</code></td>
<td>
<p>either bed formatted consensus peak matrix (peaks by 3+samples)
CHR, START, STOP and raw peak counts OR count matrix (genes by 1+samples).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>contrasts</code></td>
<td>
<p>user-defined contrasts for comparing samples</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>experiment.type</code></td>
<td>
<p>The type of experiment either set to "ATAC-Seq" or "RNA-Seq"</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>DA.choice</code></td>
<td>
<p>determines which pipeline to run:
(1) edgeR, (2) limma-voom, (3) limma-trend, (4) DEseq2.
Note: Use limma-trend if consensus peaks are already normalized, otherwise use other methods.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>DA.fdr.threshold</code></td>
<td>
<p>fdr cut-off for differential analyses</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>DA.lfc.threshold</code></td>
<td>
<p>log-fold change cutoff for differential analyses</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>comparison.scheme</code></td>
<td>
<p>either one-vs-one (OVO) or one-vs-all (OVA) comparisons.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>save.DA.peaks</code></td>
<td>
<p>saves differentially accessible peaks to an excel file</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>DA.peaks.path</code></td>
<td>
<p>the path which the excel file of the DA peaks will be saved,
if not set it will be saved to current directory.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>norm.method</code></td>
<td>
<p>normalization method for consensus peaks</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>filter.method</code></td>
<td>
<p>filtering method for low expressed peaks</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>library.threshold</code></td>
<td>
<p>number of libraries a peak occurs so that it is not filtered default set to 2</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cpm.threshold</code></td>
<td>
<p>count per million threshold for not to filter a peak</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>TSS.threshold</code></td>
<td>
<p>Distance to transcription start site in base-pairs. Default set to 50,000.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>show.annotation.pie</code></td>
<td>
<p>shows the annotation pie chart produced with ChipSeeker</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>reference.genome</code></td>
<td>
<p>genome of interested species. It should be 'hg38', 'hg19' or 'mm10'.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>batch.correction</code></td>
<td>
<p>logical, if set will run unsupervised batch correction
via sva (default) or if the batch information is known 'batch.information'
argument should be provided by user.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>batch.information</code></td>
<td>
<p>character vector, given by user.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>additional.covariates</code></td>
<td>
<p>vector or data.frame, this parameter will be directly added to design
matrix before running the differential analyses, therefore won't affect the batch corrections but
adjust the results in down-stream analyses.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sv.number</code></td>
<td>
<p>number of surrogate variables to be calculated using SVA, best left untouched.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>run.enrichment</code></td>
<td>
<p>logical, turns off enrichment pipeline</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>enrichment.method</code></td>
<td>
<p>There are two methodologies for enrichment analyses,
Hyper-geometric p-value (HPEA) or Geneset Enrichment Analyses (GSEA).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>enrichment.FDR.cutoff</code></td>
<td>
<p>FDR cut-off for enriched terms, p-values
are corrected by Benjamini-Hochberg procedure</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>background.genes.size</code></td>
<td>
<p>number of background genes for hyper-geometric p-value
calculations. Default is 20,000.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>geneset</code></td>
<td>
<p>Pathways to be used in enrichment analyses. If not set vp2008 (Chaussabel, 2008)
immune modules will be used. This can be set to any geneset using 'read.gmt' function from 'qusage'
package. Different modules are available: https://www.gsea-msigdb.org/gsea/downloads.jsp.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>prints messages through running the pipeline</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>returns differentially accessible peaks
</p>


<h3>Examples</h3>

<pre><code class="language-R">
data(atac_seq_consensus_bm) # calls 'bed'

# a vector for comparing the examples
contrasts &lt;- sapply(strsplit(colnames(bed), split = "-", fixed = TRUE),
                    function(x){x[1]})[4:25]

results &lt;- cinaR(bed, contrasts, reference.genome = "mm10")



</code></pre>


</div>