<div class="container">

<table style="width: 100%;"><tr>
<td>generate_phenodata</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Functions to generate phenotype data.</h2>

<h3>Description</h3>

<p>Functions to generate standard normal or binary phenotypes based on provided genetic
data, for specified effect sizes.
The functions <code>generate_phenodata_1_simple</code> and
<code>generate_phenodata_1</code> generate one phenotype Y conditional on
single nucleotide variants (SNVs) and two covariates.
<code>generate_phenodata_2_bvn</code> as well as <code>generate_phenodata_2_copula</code>
generate two phenotypes <code class="reqn">Y_1</code>, <code class="reqn">Y_2</code> with dependence Kendall's tau conditional on
the provided SNVs and two covariates.
</p>


<h3>Usage</h3>

<pre><code class="language-R">generate_phenodata_1_simple(genodata = NULL, type = "quantitative",
  b = 0, a = c(0, 0.5, 0.5))

generate_phenodata_1(genodata = NULL, type = "quantitative", b = 0.6,
  a = c(0, 0.5, 0.5), MAF_cutoff = 1, prop_causal = 0.1,
  direction = "a")

generate_phenodata_2_bvn(genodata = NULL, tau = NULL, b1 = 0,
  b2 = 0, a1 = c(0, 0.5, 0.5), a2 = c(0, 0.5, 0.5))

generate_phenodata_2_copula(genodata = NULL, phi = NULL, tau = 0.5,
  b1 = 0.6, b2 = 0.6, a1 = c(0, 0.5, 0.5), a2 = c(0, 0.5, 0.5),
  MAF_cutoff = 1, prop_causal = 0.1, direction = "a")
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>genodata</code></td>
<td>
<p>Numeric input vector or dataframe containing the genetic
variant(s) in columns. Must be in allelic coding 0, 1, 2.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>type</code></td>
<td>
<p>String with value <code>"quantitative"</code> or <code>"binary"</code>
specifying whether normally-distributed or binary phenotypes
are generated.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>b</code></td>
<td>
<p>Integer or vector specifying the genetic effect size(s) of
the provided SNVs (<code>genodata</code>) in the data generation.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>a</code></td>
<td>
<p>Numeric vector specifying the effect sizes of the covariates <code class="reqn">X_1</code>, <code class="reqn">X_2</code>
in the data generation.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>MAF_cutoff</code></td>
<td>
<p>Integer specifying a minor allele frequency cutoff to
determine among which SNVs the causal SNVs are
sampled for the phenotype generation.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>prop_causal</code></td>
<td>
<p>Integer specifying the desired percentage of causal SNVs
among all SNVs.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>direction</code></td>
<td>
<p>String with value <code>"a"</code>, <code>"b"</code>, or <code>"c"</code>
specifying whether all causal SNVs have a positive effect on
the phenotypes (<code>"a"</code>), 20% of the causal SNVs have a
negative effect and 80% a positive effect on the phenotypes
(<code>"b"</code>), or 50% of the causal SNVs have a negative
effect and 50% a positive effect on the phenotypes (<code>"c"</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tau</code></td>
<td>
<p>Integer specifying Kendall's tau, which determines the
dependence between the two generated phenotypes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>b1</code></td>
<td>
<p>Integer or vector specifying the genetic effect size(s) of
the provided SNVs (<code>genodata</code>) on the first phenotype
in the data generation.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>b2</code></td>
<td>
<p>Integer or vector specifying the genetic effect size(s) of
the provided SNVs (<code>genodata</code>) on the second phenotype
in the data generation.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>a1</code></td>
<td>
<p>Numeric vector specifying the effect sizes of the covariates <code class="reqn">X_1</code>, <code class="reqn">X_2</code>
on the first phenotype in the data generation.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>a2</code></td>
<td>
<p>Numeric vector specifying the effect sizes of the covariates <code class="reqn">X_1</code>, <code class="reqn">X_2</code>
on the second phenotype in the data generation.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>phi</code></td>
<td>
<p>Integer specifying the parameter <code class="reqn">\phi</code> for
the dependence between the two generated phenotypes.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>In more detail, the function <code>generate_phenodata_1_simple</code>
generates a quantitative or binary phenotype Y with n observations,
conditional on the specified SNVs with given effect sizes and conditional
on one binary and one standard normally-distributed covariate with
specified effect sizes. n is given through the provided SNVs.
</p>
<p><code>generate_phenodata_1</code> provides an extension of
<code>generate_phenodata_1_simple</code> and allows to further select
the percentage of causal SNVs, a minor allele frequency cutoff on the
causal SNVs, and varying effect directions. n is given through the
provided SNVs.
</p>
<p>The function <code>generate_phenodata_2_bvn</code> generates
two quantitative phenotypes <code class="reqn">Y_1</code>, <code class="reqn">Y_2</code> conditional on one binary and one
standard normally-distributed covariate <code class="reqn">X_1</code>, <code class="reqn">X_2</code> from the bivariate
normal distribution so that they have have dependence <code class="reqn">\tau</code> given
by Kendall's <code>tau</code>.
</p>
<p>The function <code>generate_phenodata_2_copula</code> generates
two quantitative phenotypes <code class="reqn">Y_1</code>, <code class="reqn">Y_2</code> conditional on one binary and one
standard normally-distributed covariate <code class="reqn">X_1</code>, <code class="reqn">X_2</code> from the Clayton copula
so that <code class="reqn">Y_1</code>, <code class="reqn">Y_2</code> are marginally normally distributed and have dependence
Kendall's tau specified by <code>tau</code> or <code>phi</code>, using the function
<code>generate_clayton_copula</code>.
</p>
<p>The genetic effect sizes are the specified numeric values <code>b</code> and
<code>b1, b2</code>, respectively, in the functions <code>generate_phenodata_1_simple</code>
and <code>generate_phenodata_2_bvn</code>. In
<code>generate_phenodata_1</code> and <code>generate_phenodata_2_copula</code>,
the genetic effect sizes are computed by multiplying <code>b</code> or <code>b1, b2</code>,
respectively, with the absolute value of the log10-transformed
minor allele frequencies, so that rarer variants have larger effect sizes.
</p>


<h3>Value</h3>

<p>A dataframe containing n observations of the phenotype Y or phenotypes
<code class="reqn">Y_1</code>, <code class="reqn">Y_2</code> and of the covariates <code class="reqn">X_1</code>, <code class="reqn">X_2</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">
# Generate genetic data:
set.seed(10)
genodata &lt;- generate_genodata(n_SNV = 20, n_ind = 1000)
compute_MAF(genodata)

# Generate different phenotype data:
phenodata1 &lt;- generate_phenodata_1_simple(genodata = genodata[,1],
                                          type = "quantitative", b = 0)
phenodata2 &lt;- generate_phenodata_1_simple(genodata = genodata[,1],
                                          type = "quantitative", b = 2)
phenodata3 &lt;- generate_phenodata_1_simple(genodata = genodata,
                                          type = "quantitative", b = 2)
phenodata4 &lt;- generate_phenodata_1_simple(genodata = genodata,
                                          type = "quantitative",
                                          b = seq(0.1, 2, 0.1))
phenodata5 &lt;- generate_phenodata_1_simple(genodata = genodata[,1],
                                          type = "binary", b = 0)
phenodata6 &lt;- generate_phenodata_1(genodata = genodata[,1],
                                   type = "quantitative", b = 0,
                                   MAF_cutoff = 1, prop_causal = 0.1,
                                   direction = "a")
phenodata7 &lt;- generate_phenodata_1(genodata = genodata,
                                   type = "quantitative", b = 0.6,
                                   MAF_cutoff = 0.1, prop_causal = 0.05,
                                   direction = "a")
phenodata8 &lt;- generate_phenodata_1(genodata = genodata,
                                   type = "quantitative",
                                   b = seq(0.1, 2, 0.1),
                                   MAF_cutoff = 0.1, prop_causal = 0.05,
                                   direction = "a")
phenodata9 &lt;- generate_phenodata_2_bvn(genodata = genodata[,1],
                                       tau = 0.5, b1 = 0, b2 = 0)
phenodata10 &lt;- generate_phenodata_2_bvn(genodata = genodata,
                                        tau = 0.5, b1 = 0, b2 = 0)
phenodata11 &lt;- generate_phenodata_2_bvn(genodata = genodata,
                                        tau = 0.5, b1 = 1,
                                        b2 = seq(0.1,2,0.1))
phenodata12 &lt;- generate_phenodata_2_bvn(genodata = genodata,
                                        tau = 0.5, b1 = 1, b2 = 2)
par(mfrow = c(3, 1))
hist(phenodata12$Y1)
hist(phenodata12$Y2)
plot(phenodata12$Y1, phenodata12$Y2)

phenodata13 &lt;- generate_phenodata_2_copula(genodata = genodata[,1],
                                           MAF_cutoff = 1, prop_causal = 1,
                                           tau = 0.5, b1 = 0, b2 = 0)
phenodata14 &lt;- generate_phenodata_2_copula(genodata = genodata,
                                           MAF_cutoff = 1, prop_causal = 0.5,
                                           tau = 0.5, b1 = 0, b2 = 0)
phenodata15 &lt;- generate_phenodata_2_copula(genodata = genodata,
                                           MAF_cutoff = 1, prop_causal = 0.5,
                                           tau = 0.5, b1 = 0, b2 = 0)
phenodata16 &lt;- generate_phenodata_2_copula(genodata = genodata,
                                           MAF_cutoff = 1, prop_causal = 0.5,
                                           tau = 0.2, b1 = 0.3,
                                           b2 = seq(0.1, 2, 0.1))
phenodata17 &lt;- generate_phenodata_2_copula(genodata = genodata,
                                           MAF_cutoff = 1, prop_causal = 0.5,
                                           tau = 0.2, b1 = 0.3, b2 = 0.3)
par(mfrow = c(3, 1))
hist(phenodata17$Y1)
hist(phenodata17$Y2)
plot(phenodata17$Y1, phenodata17$Y2)






</code></pre>


</div>