<div class="container">

<table style="width: 100%;"><tr>
<td>get_estimates_naive</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Naive estimates of the copula and marginal parameters.</h2>

<h3>Description</h3>

<p>Function to compute naive estimates of the copula parameter(s)
and maximum likelihood (ML) estimates of the marginal parameters in a joint
copula model of <code>Y1</code> and <code>Y2</code> given the predictors of <code>Y1</code>
and <code>Y2</code>. The main use of the function is to provide parameter
starting values for the optimization of the log-likelihood function of the
joint copula model in <code>cjamp</code> in order to obtain maximum
likelihood estimates in the copula model.
</p>


<h3>Usage</h3>

<pre><code class="language-R">get_estimates_naive(Y1 = NULL, Y2 = NULL, predictors_Y1 = NULL,
  predictors_Y2 = NULL, copula_param = "both")
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>Y1</code></td>
<td>
<p>Numeric vector containing the first phenotype.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Y2</code></td>
<td>
<p>Numeric vector containing the second phenotype.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>predictors_Y1</code></td>
<td>
<p>Dataframe containing the predictors of <code>Y1</code>
in columns.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>predictors_Y2</code></td>
<td>
<p>Dataframe containing the predictors of <code>Y2</code>
in columns.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>copula_param</code></td>
<td>
<p>String indicating whether estimates should be computed
for <code class="reqn">\phi</code> (<code>"phi"</code>), for <code class="reqn">\theta</code>
(<code>"theta"</code>), or both (<code>"both"</code>).</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The estimates of the copula parameter(s) include estimates of <code class="reqn">\phi</code>
(if <code>copula_param == "phi"</code>), <code class="reqn">\theta</code> (if
<code>copula_param == "theta"</code>) or both (if <code>copula_param == "both"</code>).
They are obtained by computing Kendall's tau between <code>Y1</code> and <code>Y2</code>
and using the relationship <code class="reqn">\tau = \phi/(\phi+2)</code> of the Clayton
copula to obtain an estimate of <code class="reqn">\phi</code> and <code class="reqn">\tau = (\theta-1)/\theta</code>
of the Gumbel copula to obtain an estimate of <code class="reqn">\theta</code>.
</p>
<p>The ML estimates of the marginal parameters include estimates of the log standard
deviations of <code>Y1</code>, <code>Y2</code> given their predictors (<code class="reqn">log(\sigma1), log(\sigma2)</code>)
and of the effects of <code>predictors_Y1</code> on <code>Y1</code> and
<code>predictors_Y2</code> on <code>Y2</code>. The estimates of the marginal effects are
obtained from linear regression models of <code>Y1</code> given <code>predictors_Y1</code>
and <code>Y2</code> given <code>predictors_Y2</code>, respectively. If single nucleotide
variants (SNVs) are included as predictors, the genetic effect estimates
are obtained from an underlying additive genetic model if SNVs are provided
as 0-1-2 genotypes and from an underlying dominant model if SNVs are provided
as 0-1 genotypes.
</p>


<h3>Value</h3>

<p>Vector of the numeric estimates of the copula parameters
<code class="reqn">log(\phi)</code> and/or <code class="reqn">log(\theta-1)</code>, of the marginal
parameters (<code class="reqn">log(\sigma1), log(\sigma2)</code>, and estimates
of the effects of the predictors <code>predictors_Y1</code>
on <code>Y1</code> and <code>predictors_Y2</code> on <code>Y2</code>).
</p>


<h3>Examples</h3>

<pre><code class="language-R"># Generate genetic data:
set.seed(10)
genodata &lt;- generate_genodata(n_SNV = 20, n_ind = 1000)

# Generate phenotype data:
phenodata &lt;- generate_phenodata_2_copula(genodata = genodata, MAF_cutoff = 1,
                                         prop_causal = 0.5, tau = 0.2,
                                         b1 = 0.3, b2 = 0.3)
predictors &lt;- data.frame(X1 = phenodata$X1, X2 = phenodata$X2,
                         SNV = genodata$SNV1)

get_estimates_naive(Y1 = phenodata$Y1, Y2 = phenodata$Y2,
                    predictors_Y1 = predictors, predictors_Y2 = predictors,
                    copula_param = "both")

</code></pre>


</div>