<div class="container">

<table style="width: 100%;"><tr>
<td>generate_genodata</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Functions to generate genetic data.</h2>

<h3>Description</h3>

<p>Functions to generate genetic data in the form of single
nucleotide variants (SNVs). The function
<code>generate_singleton_data</code> generates singletons (i.e. SNVs with
one observed minor allele); <code>generate_doubleton_data</code>
generates doubletons (i.e. SNVs with two observed minor alleles), and the
function <code>generate_genodata</code> generates <code>n_ind</code>
observations of <code>n_SNV</code> SNVs with random minor allele frequencies.
</p>


<h3>Usage</h3>

<pre><code class="language-R">generate_genodata(n_SNV = 100, n_ind = 1000)

generate_singleton_data(n_SNV = 100, n_ind = 1000)

generate_doubleton_data(n_SNV = 100, n_ind = 1000)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>n_SNV</code></td>
<td>
<p>Integer specifying the number of SNVs that are generated.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n_ind</code></td>
<td>
<p>Integer specifying the number of observations that are generated.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A dataframe containing <code>n_ind</code> observations of <code>n_SNV</code> SNVs.
</p>


<h3>Examples</h3>

<pre><code class="language-R">set.seed(10)
genodata1 &lt;- generate_singleton_data()
compute_MAF(genodata1)

genodata2 &lt;- generate_doubleton_data()
compute_MAF(genodata2)

genodata3 &lt;- generate_genodata()
compute_MAF(genodata3)

</code></pre>


</div>