<div class="container">

<table style="width: 100%;"><tr>
<td>compute_expl_var</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Phenotypic variance explained by genetic variants.</h2>

<h3>Description</h3>

<p>Function to estimate the percentage of the variance of a phenotype
that can be explained by given single nucleotide variants
(SNVs).
</p>


<h3>Usage</h3>

<pre><code class="language-R">compute_expl_var(genodata = NULL, phenodata = NULL,
  type = "Rsquared_unadj", causal_idx = NULL, effect_causal = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>genodata</code></td>
<td>
<p>Numeric vector or dataframe containing the genetic variant(s) in
columns. Must be in allelic coding 0, 1, 2.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>phenodata</code></td>
<td>
<p>Numeric vector or dataframe of the phenotype.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>type</code></td>
<td>
<p>String (vector) specifying the estimation approach(es) that are computed.
Available are the methods <code>"Rsquared_unadj"</code>,
<code>"Rsquared_adj"</code>, <code>"MAF_based"</code>, and
<code>"MAF_based_Y_adjusted"</code>. See below for more details.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>causal_idx</code></td>
<td>
<p>Vector with entries <code>TRUE</code>, <code>FALSE</code> specifying which
SNVs are causal. Has to be supplied for the approaches
<code>MAF_based</code> and <code>MAF_based_Y_adjusted</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>effect_causal</code></td>
<td>
<p>Numeric vector containing the effect sizes of the causal SNVs.
Has to be supplied for the approaches <code>MAF_based</code> and
<code>MAF_based_Y_adjusted</code>.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Four different approaches are available to estimate the percentage of
explained phenotypic variance (Laird &amp; Lange, 2011):
</p>
<p>(1) <code>"Rsquared_unadj"</code>: Unadjusted <code class="reqn">R^2</code> from a linear regression of
the phenotype conditional on all provided SNVs.
</p>
<p>(2) <code>"Rsquared_adj"</code>: Adjusted <code class="reqn">R^2</code> from a linear regression of
the phenotype conditional on all provided SNVs.
</p>
<p>(3) <code>"MAF_based"</code>: Expected explained phenotypic variance computed based on the
MAF and effect size of the provided causal SNVs.
</p>
<p>(4) <code>"MAF_based_Y_adjusted"</code>: Expected explained phenotypic variance computed
based on the MAF and effect size of the causal SNVs, with respect to the empirical
phenotypic variance, which is the broad-sense heritability relative to the
empirical phenotypic variance.
</p>
<p>References:
</p>
<p>Laird NM, Lange C (2011). The fundamentals of modern statistical genetics. New York: Springer.
</p>


<h3>Value</h3>

<p>A list containing the estimated percentage of explained phenotypic variance.
</p>


<h3>Examples</h3>

<pre><code class="language-R">
set.seed(10)
genodata &lt;- generate_genodata(n_SNV = 20, n_ind = 1000)
phenodata &lt;- generate_phenodata_1_simple(genodata = genodata[,1],
                                         type = "quantitative", b = 0)
compute_expl_var(genodata = genodata, phenodata = phenodata$Y,
                 type = c("Rsquared_unadj", "Rsquared_adj"),
                 causal_idx = NULL, effect_causal = NULL)

</code></pre>


</div>