<div class="container">

<table style="width: 100%;"><tr>
<td>variantT</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Classes for Variant T-tests</h2>

<h3>Description</h3>

<p>Classes to perform row-by-row paired or unequal variance t-tests on
microarray or proteomics data.
</p>


<h3>Usage</h3>

<pre><code class="language-R">MultiTtestPaired(data, classes, pairing)
MultiTtestUnequal(data, classes)
## S4 method for signature 'MultiTtestPaired'
summary(object, ...)
## S4 method for signature 'MultiTtestUnequal'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>data</code></td>
<td>
<p> Either a data frame or matrix with numeric values or an
<code>ExpressionSet</code> as defined
in the BioConductor tools for analyzing microarray data.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>classes</code></td>
<td>
<p> If <code>data</code> is a data frame or matrix, then classes
must be either a logical vector or a factor. If <code>data</code> is an
<code>ExpressionSet</code>, then <code>classes</code> can be a character string that
names one of the factor columns in the associated
<code>phenoData</code> subobject.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pairing</code></td>
<td>
<p>A numerical vector indicating which samples are
paired.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>object</code></td>
<td>
<p>A <code>MultiTtest</code> object</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Unused; optional extra parameters for <code>summary</code>.</p>
</td>
</tr>
</table>
<h3>Creating objects</h3>

<p>Although objects can be created using <code>new</code>, the better method is
to use the <code>MultiTtestPaired</code> or <code>MultiTtestUnequal</code>
functions. In the simplest case, you simply pass in a data matrix
and a logical vector assigning classes to the columns (and, in the
case of a paired t-test, a numeric vector describing the pairing), and
the constructor performs row-by-row two-sample t-tests and computes
the associated (single test) p-values.  To adjust for multiple
testing, you can pass the p-values on to the <code>Bum</code> class.
</p>
<p>If you use a factor instead of a logical vector, then the t-test
compares the first level of the factor to everything else.  To handle
the case of multiple classes, see the <code>MultiLinearModel</code>
class.
</p>
<p>As with other class comparison functions that are part of the OOMPA,
we can also perform statistical tests on
<code>ExpressionSet</code> objects from
the BioConductor libraries. In this case, we pass in an
<code>ExpressionSet</code> object along with the name of a factor to use for
splitting the data.
</p>


<h3>Extends</h3>

<p>Both classes extend class <code>MultiTtest</code>, directly. See that
class for descriptions of the inherited methods and slots.
</p>


<h3>Slots</h3>


<dl>
<dt>
<code>df</code>:</dt>
<dd>
<p>The <code>MultiTtestUnequal</code> class adds a slot to
record e gene-by-gene degrees of freedom, which can change along
with the variances.</p>
</dd>
</dl>
<h3>Methods</h3>


<dl>
<dt>summary</dt>
<dd>
<p><code>signature(object = MultiTtestPaired)</code>:
Write out a summary of the object. </p>
</dd>
<dt>summary</dt>
<dd>
<p><code>signature(object = MultiTtestUnequal)</code>:
Write out a summary of the object. </p>
</dd>
</dl>
<h3>Author(s)</h3>

<p>Kevin R. Coombes <a href="mailto:krc@silicovore.com">krc@silicovore.com</a>
</p>


<h3>References</h3>

<p>OOMPA</p>


<h3>See Also</h3>

<p><code>Bum</code>,
<code>MultiTtest</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">showClass("MultiTtestPaired")
showClass("MultiTtestUnequal")
ng &lt;- 10000
ns &lt;- 50
dat &lt;- matrix(rnorm(ng*ns), ncol=ns)
cla &lt;- factor(rep(c('A', 'B'), each=25))
res &lt;- MultiTtestUnequal(dat, cla)
summary(res)
hist(res, breaks=101)
plot(res, res@p.values)

pairing &lt;- rep(1:25, 2)
res &lt;- MultiTtestPaired(dat, cla, pairing)
summary(res)
plot(res)
hist(res@p.values, breaks=101)
</code></pre>


</div>