<div class="container">

<table style="width: 100%;"><tr>
<td>Mosaic</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Class "Mosaic"</h2>

<h3>Description</h3>

<p>Produce “Eisen” plots of microarray or proteomics data.
</p>


<h3>Usage</h3>

<pre><code class="language-R">Mosaic(data,
       sampleMetric="pearson",
       sampleLinkage="average",
       geneMetric="euclid",
       geneLinkage="average",
       usecor=FALSE,
       center=FALSE,
       name="My mosaic")
## S4 method for signature 'Mosaic'
pltree(x, colors, labels, ...)
## S4 method for signature 'Mosaic,missing'
plot(x, main=x@name, center=FALSE,
  scale=c("none", "row", "column"), limits=NULL,
  sampleColors=NULL, sampleClasses=NULL,
  geneColors=NULL, geneClasses=NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>data</code></td>
<td>
<p> Either a data frame or matrix with numeric values or an
<code>ExpressionSet</code> as defined
in the BioConductor tools for analyzing microarray data.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sampleMetric</code></td>
<td>
<p>Any valid distance metric that can be passed to the
<code>distanceMatrix</code> function</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sampleLinkage</code></td>
<td>
<p>Any valid linkage method that can be passed to the
<code>hclust</code> function</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>geneMetric</code></td>
<td>
<p>Any valid distance metric that can be passed to the
<code>distanceMatrix</code> function</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>geneLinkage</code></td>
<td>
<p>Any valid linkage method that can be passed to the
<code>hclust</code> function</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>center</code></td>
<td>
<p>logical scalar. If <code>TRUE</code>, center the data rows.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>usecor</code></td>
<td>
<p>logical scalar. If <code>TRUE</code>, scale the data rows to have
standard deviation one.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>name</code></td>
<td>
<p>character string specifying the name of this object</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>object of class <code>Mosaic</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>scale</code></td>
<td>
<p>Same as in <code>heatmap</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>colors</code></td>
<td>
<p>An optional vector of character strings containing color
names to be used when labeling the trees in the dendrogram. If
provided, then the length should equal the number of columns in the
original data matrix.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>labels</code></td>
<td>
<p>An optional vector of character strings used to label
the leaves in the dendrogram. If omitted, the column names are
used.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>main</code></td>
<td>
<p>character string specifying the main title for the plot</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>limits</code></td>
<td>
<p>An numeric vector. If provided, the data is truncated
for display purposes, both above and below, at the minimum and
maximum values of <code>limits</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sampleColors</code></td>
<td>
<p>An optional character vector containing colors
that will be used to label different sample types with a color bar
across the top of the heat map.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sampleClasses</code></td>
<td>
<p>A logical vector or factor used to classify the
samples into groups. Alternatively, an integer specifying the number
<code>k</code> of groups into which to cut the sample dendrogram.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>geneColors</code></td>
<td>
<p>An optional character vector containing colors
that will be used to label different gene types with a color bar
along the side of the heat map.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>geneClasses</code></td>
<td>
<p>A logical vector or factor used to classify the
genes into groups. Alternatively, an integer specifying the number
<code>k</code> of groups into which to cut the gene dendrogram.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Additional parameters for <code>heatmap</code>.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>One of the earliest papers in the microarray literature used
independent clustering of the genes (rows) and samples (columns) to
produce dendrograms that were plotted along with a red-green heat map
of the centered expression values. Since that time, literally thousand
of additional papers have published variations on these red-green
images. R includes a function, <code>heatmap</code> that builds such
figures.  However, that function is general purpose and has numerous
optional parameters to tweak the display. The purpose of the
<code>Mosaic</code> class is to provide a simplified object-oriented wrapper
around <code>heatmap</code>, which as a side benefit allows us to
keep track of the distance metrics and linkage rules that were used to
produce the resulting figure.
</p>


<h3>Value</h3>

<p>The <code>Mosaic</code> function constructs and returns a valid object of
the <code>Mosaic</code> class.
</p>


<h3>Objects from the Class</h3>

<p>Objects should be created with the <code>Mosaic</code> function.
</p>


<h3>Slots</h3>


<dl>
<dt>
<code>data</code>:</dt>
<dd>
<p>The <code>matrix</code> containing the numerical data </p>
</dd>
<dt>
<code>samples</code>:</dt>
<dd>
<p>A dendrogram of class <code>hclust</code> produced
by clustering the biological samples (columns of <code>data</code>). </p>
</dd>
<dt>
<code>genes</code>:</dt>
<dd>
<p>A dendrogram of class <code>hclust</code> produced by
clustering the genes (columns of <code>data</code>).</p>
</dd>
<dt>
<code>sampleMetric</code>:</dt>
<dd>
<p>A <code>character</code> string; the distance
metric used to cluster the samples. </p>
</dd>
<dt>
<code>sampleLinkage</code>:</dt>
<dd>
<p>A <code>character</code> string; the linkage
rule used to cluster the samples. </p>
</dd>
<dt>
<code>geneMetric</code>:</dt>
<dd>
<p>A <code>character</code> string; the distance
metric used to cluster the genes. </p>
</dd>
<dt>
<code>geneLinkage</code>:</dt>
<dd>
<p>A <code>character</code> string; the linkage
rule used to cluster the genes. </p>
</dd>
<dt>
<code>call</code>:</dt>
<dd>
<p>An object of class <code>call</code> recording how the
object was constructed. </p>
</dd>
<dt>
<code>name</code>:</dt>
<dd>
<p>A <code>character</code> string; the name of this object. </p>
</dd>
</dl>
<h3>Methods</h3>


<dl>
<dt>plot</dt>
<dd>
<p><code>signature(x = Mosaic, y = missing)</code>: Produce the
“Eisen” plot, using <code>heatmap</code>. </p>
</dd>
<dt>pltree</dt>
<dd>
<p><code>signature(x = Mosaic)</code>: Plot the sample class
dendrogram in the object. </p>
</dd>
<dt>summary</dt>
<dd>
<p><code>signature(object = Mosaic)</code>: Write out a
summary of the object. </p>
</dd>
</dl>
<h3>Author(s)</h3>

<p>Kevin R. Coombes <a href="mailto:krc@silicovore.com">krc@silicovore.com</a>,
P. Roebuck <a href="mailto:proebuck@mdanderson.org">proebuck@mdanderson.org</a>
</p>


<h3>References</h3>

<p>Eisen MB, Spellman PT, Brown PO, Botstein D.<br><em>Cluster analysis and display of genome-wide expression patterns.</em><br>
Proc Natl Acad Sci U S A. 1998 Dec 8;95(25):14863-8.
</p>


<h3>See Also</h3>

<p><code>cutree</code>,
<code>hclust</code>,
<code>heatmap</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">showClass("Mosaic")

## simulate data from three different sample groups
d1 &lt;- matrix(rnorm(100*10, rnorm(100, 0.5)), nrow=100, ncol=10, byrow=FALSE)
d2 &lt;- matrix(rnorm(100*10, rnorm(100, 0.5)), nrow=100, ncol=10, byrow=FALSE)
d3 &lt;- matrix(rnorm(100*10, rnorm(100, 0.5)), nrow=100, ncol=10, byrow=FALSE)
dd &lt;- cbind(d1, d2, d3)
kind &lt;- factor(rep(c('red', 'green', 'blue'), each=10))

## prepare the Mosaic object
m &lt;- Mosaic(dd,
            sampleMetric='pearson',
            geneMetric='spearman',
            center=TRUE,
            usecor=TRUE)
summary(m)

## The default plot with red-green color map
plot(m, col=redgreen(64))

## change to a blue-yellow color map, and mark the four top splits in the
## sample direction with a color bar along the top
plot(m, col=blueyellow(128), sampleClasses=4,
     sampleColors=c('red', 'green', 'blue', 'black'))

## This time, mark the three classes that we know are there
plot(m, col=blueyellow(128), sampleClasses=kind,
     sampleColors=c('red', 'green', 'blue'))

plot(m, col=blueyellow(128),
     geneClasses=3, geneColors=c('red', 'green', 'black'))

## In addition, mark the top 5 splits in the gene dendrogram
plot(m,
     col=blueyellow(128),
     sampleClasses=kind,
     sampleColors=c('red', 'green', 'black'),
     geneClasses=5,
     geneColors=c('cyan', 'magenta', 'royalblue', 'darkgreen', 'orange'))

## plot the sample dendrogram by itself
cols &lt;- as.character(kind)
pltree(m, labels=1:30, colors=cols)

## cleanup
rm(d1, d2, d3, dd, kind, cols, m)
</code></pre>


</div>