<div class="container">

<table style="width: 100%;"><tr>
<td>MultiWilcoxonTest-class</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Class "MultiWilcoxonTest"</h2>

<h3>Description</h3>

<p>The <code>MultiWilcoxonTest</code> class is used to perform row-by-row Wilcoxon
rank-sum tests on a data matrix. Significance cutoffs are determined by the
empirical Bayes method of Efron and Tibshirani.
</p>


<h3>Usage</h3>

<pre><code class="language-R">MultiWilcoxonTest(data, classes, histsize=NULL)
## S4 method for signature 'MultiWilcoxonTest'
summary(object, prior=1, significance=0.9, ...)
## S4 method for signature 'MultiWilcoxonTest'
hist(x, xlab='Rank Sum',
 ylab='Prob(Different | Y)', main='', ...)
## S4 method for signature 'MultiWilcoxonTest,missing'
plot(x, prior=1, significance=0.9,
 ylim=c(-0.5, 1), xlab='Rank Sum', ylab='Prob(Different | Y)', ...)
## S4 method for signature 'MultiWilcoxonTest'
cutoffSignificant(object, prior, significance, ...)
## S4 method for signature 'MultiWilcoxonTest'
selectSignificant(object, prior, significance, ...)
## S4 method for signature 'MultiWilcoxonTest'
countSignificant(object, prior, significance, ...)
## S4 method for signature 'MultiWilcoxonTest'
probDiff(object, p0, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>data</code></td>
<td>
<p>either a data frame or matrix with numeric values, or an
<code>ExpressionSet</code> as defined
in the BioConductor tools for analyzing microarray data.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>classes</code></td>
<td>
<p> If <code>data</code> is a data frame or matrix, then classes
must be either a logical vector or a factor. If <code>data</code> is an
<code>ExpressionSet</code>, then <code>classes</code> can be a character string that
names one of the factor columns in the associated
<code>phenoData</code> subobject.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>histsize</code></td>
<td>
<p>An integer; the number of bins used for the histogram
summarizing the Wilcoxon statistics. When <code>NULL</code>, each discrete
rank-sum value gets its own bin.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>object</code></td>
<td>
<p>an object of the <code>MultiWilcoxonTest</code> class.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>an object of the <code>MultiWilcoxonTest</code> class.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>xlab</code></td>
<td>
<p>character string specifying label for the x axis</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ylab</code></td>
<td>
<p>character string specifying label for the y axis</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ylim</code></td>
<td>
<p>Plotting limits on the y-axis</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>main</code></td>
<td>
<p>character string specifying graph title</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p0</code></td>
<td>
<p>see prior.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>prior</code></td>
<td>
<p>Prior probability that an arbitrary gene is not differentially
expressed, or that an arbitrary row does not yield a significant
Wilcoxon rank-sum statistic. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>significance</code></td>
<td>
<p>Desired level of posterior probability</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>extra arguments for generic or plotting routines</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>See the paper by Efron and Tibshirani.
</p>


<h3>Value</h3>

<p>The standard methods <code>summary</code>, <code>hist</code>, and <code>plot</code>
return what you would expect.
</p>
<p>The <code>cutoffSignificant</code> method returns a list of two
integers. Rank-sum values smaller than the first value or larger than
the second value are statistically significant in the sense that their
posterior probability exceeds the specified <code>significance</code> level
given the assumptions about the <code>prior</code> probability of not being
significant.
</p>
<p>The <code>selectSignificant</code> method returns a vector of logical values
identifying the significant test results, and <code>countSignificant</code>
returns an integer counting the number of significant test results.
</p>


<h3>Creating Objects</h3>

<p>As usual, objects can be created by <code>new</code>, but better methods are
available in the form of the <code>MultiWilcoxonTest</code> function. The
inputs to this function are the same as those used for row-by-row
statistical tests throughout the ClassComparison package; a detailed
description can be found in the <code>MultiTtest</code> class.
</p>
<p>The constructor computes row-by-row Wilcoxon rank-sum statistics on
the input <code>data</code>, comparing the two groups defined by the
<code>classes</code> argument. It also estimates the observed and
theoretical (expected) density functions for the collection of
rank-sum statistics.
</p>
<p>The additional input argument, <code>histsize</code> is usually best left to
its default value. In certain pathological cases, we have found it
necessary to use fewer bins; one suspects that the underlying model
does not adequately capture the complexity of those situations.
</p>


<h3>Slots</h3>


<dl>
<dt>
<code>statistics</code>:</dt>
<dd>
<p>numeric vector containing the
computed rank-sum statistics.</p>
</dd>
<dt>
<code>xvals</code>:</dt>
<dd>
<p>numeric vector, best thought of as the vector
of possible rank-sum statistics given the sizes of the two groups.</p>
</dd>
<dt>
<code>theoretical.pdf</code>:</dt>
<dd>
<p>numeric vector containing the
theoretical density function evaluated at the points of
<code>xvals</code>.</p>
</dd>
<dt>
<code>pdf</code>:</dt>
<dd>
<p>numeric vector containing the empirical density
function computed at the points of <code>xvals</code>.</p>
</dd>
<dt>
<code>unravel</code>:</dt>
<dd>
<p>numeric vector containing a smoothed
estimate (by Poisson regression using B-splines) of the empirical
density function evaluated at <code>xvals</code>.</p>
</dd>
<dt>
<code>groups</code>:</dt>
<dd>
<p>A vector containing the names of the groups
defined by <code>classes</code>.</p>
</dd>
<dt>
<code>call</code>:</dt>
<dd>
<p>object of class <code>call</code> representing the
function call that created the object.</p>
</dd>
</dl>
<h3>Methods</h3>


<dl>
<dt>summary(object, prior=1, significance=0.9, ...)</dt>
<dd>
<p>Write out a summary of the object. For a given value of the
<code>prior</code> probability of not being differentially expressed and
a given significance cutoff on the posterior probability, reports
the cutoffs and number of items in both tails of the distribution.</p>
</dd>
<dt>hist(x, xlab='Rank Sum', ylab='Prob(Different|Y)', main=‚Äù, ...)</dt>
<dd>
<p>Plot a histogram of the rank-sum statistics, with overlaid curves
representing the expected and observed distributions.  Colors of
the curves are controlled by
<code>oompaColor$EXPECTED</code> and 
<code>oompaColor$OBSERVED</code>.
</p>
</dd>
<dt>plot(x, prior=1, significance=0.9, ylim=c(-0.5, 1), xlab='Rank
Sum', ylab='Prob(Different | Y)', ...)</dt>
<dd>
<p>Plots the posterior probability of being differentially expressed
for given values of the <code>prior</code>.  Horizontal lines are added
at each specified <code>significance</code> level for the posterior
probability.
</p>
</dd> 
<dt>cutoffSignificant(object, prior, significance, ...)</dt>
<dd>
<p>Determine
cutoffs on the rank-sum statistic at the desired significance level.</p>
</dd>
<dt>selectSignificant(object, prior, significance, ...)</dt>
<dd>
<p>Compute a
logical vector for selecting significant test results.</p>
</dd>
<dt>countSignificant(object, prior, significance, ...)</dt>
<dd>
<p>Count the
number of significant test results.</p>
</dd>
<dt>probDiff(object, p0, ...)</dt>
<dd>
<p>Compute the probabilty that an
observed value comes from the "unusual" part of the mixture
distribution. Only exported so it can be inherited by other
classes....</p>
</dd> 
</dl>
<h3>Author(s)</h3>

<p>Kevin R. Coombes <a href="mailto:krc@silicovore.com">krc@silicovore.com</a>
</p>


<h3>References</h3>

<p>Efron B, Tibshirani R.<br><em>Empirical bayes methods and false discovery rates for microarrays.</em><br>
Genet Epidemiol 2002, 23: 70-86.
</p>
<p>Pounds S, Morris SW.<br><em>Estimating the occurrence of false positives and false negatives in
microarray studies by approximating and partitioning the empirical
distribution of p-values.</em><br>
Bioinformatics. 2003 Jul 1;19(10):1236-42. 
</p>


<h3>See Also</h3>

<p>Implementation is handled in part by the functions <code>dwil</code>
and <code>rankSum</code>. The empirical Bayes results for
alternative tests (such as <code>MultiTtest</code>) can be obtained
using the beta-uniform mixture model in the <code>Bum</code> class.
</p>


<h3>Examples</h3>

<pre><code class="language-R">showClass("MultiWilcoxonTest")
ng &lt;- 10000
ns &lt;- 15
nd &lt;- 200
fake.class &lt;- factor(rep(c('A', 'B'), each=ns))
fake.data &lt;- matrix(rnorm(ng*ns*2), nrow=ng, ncol=2*ns)
fake.data[1:nd, 1:ns] &lt;- fake.data[1:nd, 1:ns] + 2
fake.data[(nd+1):(2*nd), 1:ns] &lt;- fake.data[(nd+1):(2*nd), 1:ns] - 2

a &lt;- MultiWilcoxonTest(fake.data, fake.class)
hist(a)
plot(a)
plot(a, prior=0.85)
abline(h=0)

cutoffSignificant(a, prior=0.85, signif=0.95)
countSignificant(a, prior=0.85, signif=0.95)
</code></pre>


</div>