<div class="container">

<table style="width: 100%;"><tr>
<td>DNAStringSetTPR</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Curing of biological sequences</h2>

<h3>Description</h3>

<p>Curates biological sequences of two restriction enzyme primers or cloning vectors.This cleaning is required for techniques such as cDNA-AFLP.This cleaning is required for techniques such as cDNA-AFLP.
</p>


<h3>Usage</h3>

<pre><code class="language-R">DNAStringSetTPR(SEQs, PrimerF, PrimerR)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>SEQs</code></td>
<td>
<p>file with fasta format containing biological sequences that are to be cleaned.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>PrimerF</code></td>
<td>
<p>dnastring containing the foward primer/vector sequences to be removed.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>PrimerR</code></td>
<td>
<p>dnastring containing the reverse primer/vector sequences to be removed.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>clean biological sequences and visualization of the alignments
</p>


<h3>Author(s)</h3>

<p>Florencia I Pozzi, Silvina A. Felitti
</p>


<h3>Examples</h3>

<pre><code class="language-R">SEQs = readDNAStringSet(system.file("sequences","SeqInputTPR.fasta", package = "CleanBSequences"))
PrimerR= DNAString ("GACTGCGTACCATGC")
PrimerF = DNAString("GATGAGTCCTGACCGAA")
DNAStringSetTPR (SEQs,PrimerF,PrimerR)
</code></pre>


</div>