<div class="container">

<table style="width: 100%;"><tr>
<td>geno_freq_phased</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Calculate phased genotype frequencies from allele frequencies,
assuming Hardy-Weinberg equilibrium</h2>

<h3>Description</h3>

<p>A function to calculate the population frequencies of the phased genotypes
at a single autosomal genetic locus that has given allele frequencies and
is at Hardy-Weinberg equilibrium.  Phased genotypes can be used to
investigate parent-of-origin effects, e.g. see (van Vliet et al., 2011).
</p>


<h3>Usage</h3>

<pre><code class="language-R">geno_freq_phased(p_alleles, annotate = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>p_alleles</code></td>
<td>
<p>A vector of strictly positive numbers that sum to <code>1</code>,
with <code>p_alleles[i]</code> interpreted as the allele frequency of the <code>i</code>th allele
of the genetic locus.  When <code>annotate</code> is <code>TRUE</code>, the names of the alleles
will be taken to be <code>names(p_alleles)</code> or, if <code>names(p_alleles)</code> is <code>NULL</code>,
to be <code>1:length(p_alleles)</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>annotate</code></td>
<td>
<p>A logical flag.  When <code>FALSE</code> (the default), the function
returns a vector suitable to be used as the <code>geno_freq</code> argument of
<code>pedigree_loglikelihood</code>.  When <code>TRUE</code>, the function adds a
<code>names</code> attribute to this vector to indicate which element corresponds to
which phased genotype.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>For a genetic locus that is at Hardy-Weinberg equilibrium in a
particular population, the population allele frequencies at
the locus determine the population genotype frequencies; see Sections 1.2 and
1.3 of (Lange, 2002) for the unphased version of this law.  When a genetic
locus is at Hardy-Weinberg equilibrium, the maternal and paternal alleles of a
random person from the population are independent.  A phased genotype at a
genetic locus is an ordered pair consisting of a maternal
and paternal allele at the locus.  So to any heterozygous unphased genotype,
there are two corresponding phased genotypes, and these two phased genotypes
have equal frequencies under Hardy-Weinberg equilibrium.
</p>
<p>Given a vector <code>p_alleles</code> containing the allele frequencies,
this function returns the frequencies of the possible phased genotypes,
in a particular order that can be viewed by setting <code>annotate</code> to <code>TRUE</code>.
If the alleles are named <code>1:length(p_alleles)</code>, so that <code>p_alleles[i]</code> is
the frequency of allele <code>i</code>, then the phased genotypes are of the form
<code style="white-space: pre;">⁠1|1, 1|2, ...⁠</code>, where <code>a|b</code> means the maternal allele is <code>a</code> and the
paternal allele is <code>b</code>. Note that if the output of this function is to be
used as the <code>geno_freq</code> argument of <code>pedigree_loglikelihood</code>
then the <code>annotate</code> option must be set to <code>FALSE</code>.
</p>


<h3>Value</h3>

<p>A vector of strictly positive numbers (the genotype frequencies)
that sum to <code>1</code>, named with the genotype names if <code>annotate</code> is <code>TRUE</code>.
</p>


<h3>References</h3>

<p>Lange K.  Mathematical and Statistical Methods for Genetic Analysis (second edition).
Springer, New York.  2002.
</p>
<p>van Vliet CM, Dowty JG, van Vliet JL, et al. Dependence of colorectal cancer
risk on the parent-of-origin of mutations in DNA mismatch repair genes.
Hum Mutat. 2011;32(2):207-212.
</p>


<h3>Examples</h3>

<pre><code class="language-R"># Genotype frequencies for a biallelic locus at Hardy-Weinberg equilibrium
# and with a minor allele frequency of 10%
p_alleles &lt;- c(0.9, 0.1)
geno_freq_phased(p_alleles, annotate = TRUE)

# Genotype frequencies for a triallelic locus at Hardy-Weinberg equilibrium
p_alleles &lt;- c(0.85, 0.1, 0.05)
geno_freq_phased(p_alleles, annotate = TRUE)
sum(geno_freq_phased(p_alleles))

</code></pre>


</div>