<div class="container">

<table style="width: 100%;"><tr>
<td>trans_phased</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>The transmission matrix for phased genotypes</h2>

<h3>Description</h3>

<p>A function to calculate the transmission matrix for a single autosomal
genetic locus with an arbitrary number of alleles and phased genotypes,
based on Mendel's laws of inheritance.  Phased genotypes can be used to
investigate parent-of-origin effects, e.g. see (van Vliet et al., 2011).
</p>


<h3>Usage</h3>

<pre><code class="language-R">trans_phased(n_alleles, annotate = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>n_alleles</code></td>
<td>
<p>A positive integer, interpreted as the number of possible
alleles at the genetic locus.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>annotate</code></td>
<td>
<p>A logical flag. When <code>FALSE</code> (the default), the function
returns a matrix suitable to be used as the <code>trans</code> argument of
<code>pedigree_loglikelihood</code>. When <code>TRUE</code>, the function annotates
this matrix (and converts it to a data frame) to make the output more
easily understood by humans.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>When <code>annotate</code> is <code>FALSE</code>, a matrix of genetic transmission
probabilities is returned, with rows corresponding to the possible joint
parental genotypes and columns corresponding to the possible offspring
genotypes.
There are <code>ngeno = n_alleles^2</code> possible phased genotypes,
and by choosing an order on these genotypes (which can be
viewed by setting <code>annotate</code> to <code>TRUE</code>, see below)
we can label the set of possible phased genotypes as <code>1:ngeno</code>.
Then the <code style="white-space: pre;">⁠(ngeno * gm + gf - ngeno, go)⁠</code>th element of the outputted matrix is
the conditional probability that a person has genotype <code>go</code>, given that his
or her biological mother and father have genotypes <code>gm</code> and <code>gf</code>,
respectively.
</p>
<p>When <code>annotate</code> is <code>TRUE</code>, the function converts this matrix to a data frame,
adds column names giving the offspring genotype corresponding to each
column, and adds columns <code>gm</code> and <code>gf</code> describing the parental genotypes
corresponding to each row.
In this data frame, phased genotypes are written in the usual form
<code style="white-space: pre;">⁠1|1, 1|2, ...⁠</code> for the alleles <code>1:n_alleles</code>, where <code>a|b</code> means the
maternal allele is <code>a</code> and the paternal allele is <code>b</code>.
</p>
<p>Note that if the output of this function is to be used as the <code>trans</code>
argument of <code>pedigree_loglikelihood</code> then the <code>annotate</code> option
must be set to <code>FALSE</code>.
</p>


<h3>Value</h3>

<p>Either a matrix of genetic transmission probabilities suitable to be
used as the <code>trans</code> argument of <code>pedigree_loglikelihood</code>
(if <code>annotate</code> is <code>FALSE</code>), or a data frame that is an annotated version of
this matrix (if <code>annotate</code> is <code>TRUE</code>).
</p>


<h3>References</h3>

<p>van Vliet CM, Dowty JG, van Vliet JL, et al. Dependence of colorectal cancer
risk on the parent-of-origin of mutations in DNA mismatch repair genes.
Hum Mutat. 2011;32(2):207-212.
</p>


<h3>Examples</h3>

<pre><code class="language-R"># The transition matrix for a biallelic, autosomal locus with phased genotypes
trans_phased(2)
trans_phased(2, annotate = TRUE)

</code></pre>


</div>