<div class="container">

<table style="width: 100%;"><tr>
<td>Simulating Clones</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2> Simulating Tumor Clones </h2>

<h3>Description</h3>

<p>Simulating copy number segmentation data and sequencing mutation data
for tumors composed of multiple clones.
</p>


<h3>Usage</h3>

<pre><code class="language-R">generateTumorData(tumor, snps.seq, snps.cgh, mu, sigma.reads,
                  sigma0.lrr, sigma0.baf, density.sigma)
plotTumorData(tumor, data)
tumorGen(...)
dataGen(tumor, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>tumor</code></td>
<td>
<p> an object of the <code>Tumor</code> class. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>snps.seq</code></td>
<td>
<p>an integer; the total number of germline variants and
somatic mutations to simulate in the tumor genome.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>snps.cgh</code></td>
<td>
<p>an integer; the number of single nucleotide
polymorphisms (SNPs) to simulate as meaurements made to estimate
copy number.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mu</code></td>
<td>
<p>an integer; the average read depth of a simulated sequencing
study giving rise to mutations.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sigma.reads</code></td>
<td>
<p>a real number; the standard deviation of the number
of simulated sequencing reads per base.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sigma0.lrr</code></td>
<td>
<p>a real number; the standard deviation of the
simulated per-SNP log R ratio (LRR) for assessing copy number.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sigma0.baf</code></td>
<td>
<p>a real number; the standard deviation of the
simulated B allele frequency (BAF) for assessing copy number.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>density.sigma</code></td>
<td>
<p>a real number; the standard deviation of a beta
distribution used to simulate the number of SNP markers per copy
number segment.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>data</code></td>
<td>
<p> a list containing two data frames, <code>cn.data</code> and
<code>seq.data</code>, as produced by <code>generateTumorData</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>additional variables</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Copy number and mutation data are simulated essentially
independently. Each simulation starts with a single "normal" genome, 
and CNVs and/or mutations are randomly generated for each new "branch"
or subclone. (The number of subclones depends on the input
parameters.) Each successive branch is randomly determined to descend
from one of the existing clones, and therefore contains both the
aberrations belonging to its parent clone and the novel aberrations
assigned to it. Depending on input parameters, the algorithm can also
randomly select some clones for extinction in the process of
generating the heterogeneous tumor, to yield a more realistic
population structure. 
</p>
<p>Note that <code>tumorGen</code> (an alias for <code>Tumor</code> that returns a
list instead of a Tumor object) and <code>dataGen</code> (an alias for
<code>generateTumorData</code>) are DEPRECATED.
</p>


<h3>Value</h3>

<p>The <code>generateTumorData</code> function returns a list with two
components, <code>cn.data</code> and <code>seq.data</code>. Each component is
itself a data frame. Note that in some cases, one of these data frames
may have zero rows or may be returned as an <code>NA</code>.
</p>
<p>The <code>cn.data</code> component contains seven columns:
</p>

<dl>
<dt><code>chr</code></dt>
<dd>
<p>the chromosome number;</p>
</dd>
<dt><code>seq</code></dt>
<dd>
<p>a unique segment identifier;</p>
</dd>
<dt><code>LRR</code></dt>
<dd>
<p>simulated segment-wise log ratios;</p>
</dd>
<dt><code>BAF</code></dt>
<dd>
<p>simulated segment-wise B allele frequencies;</p>
</dd>
<dt>
<code>X</code> and <code>Y</code>
</dt>
<dd>
<p>simulated intensities for two separate
alleles/haplotypes per segment; and</p>
</dd> 
<dt><code>markers</code></dt>
<dd>
<p>the simulated number of SNPS per segment.</p>
</dd>
</dl>
<p>The <code>seq.data</code> component contains eight columns:
</p>

<dl>
<dt><code>chr</code></dt>
<dd>
<p>the chromosome number;</p>
</dd>
<dt><code>seq</code></dt>
<dd>
<p>a unique "segment" identifier;</p>
</dd>
<dt><code>mut.id</code></dt>
<dd>
<p>a unique mutation identifier;</p>
</dd>
<dt>
<code>refCounts</code> and <code>varCounts</code>
</dt>
<dd>
<p>the simulated numbers
of reference and variant counts per mutation;</p>
</dd>
<dt><code>VAF</code></dt>
<dd>
<p>the simulated variant allele frequency;</p>
</dd>
<dt><code>totalCounts</code></dt>
<dd>
<p>the simulated total number of read counts;
and</p>
</dd>
<dt><code>status</code></dt>
<dd>
<p>a character (that should probably be a factor)
indicating whether a variant should be viewed as somatic or
germline.</p>
</dd>
</dl>
<p>The <code>plotTumorData</code> function invisibly returns its <code>data</code>
argument.
</p>


<h3>Author(s)</h3>

<p>Kevin R. Coombes <a href="mailto:krc@silicovore.com">krc@silicovore.com</a>,
Mark Zucker <a href="mailto:zucker.64@buckeyemail.osu.edu">zucker.64@buckeyemail.osu.edu</a>
</p>


<h3>Examples</h3>

<pre><code class="language-R">psis &lt;- c(0.6, 0.3, 0.1) # three clones
# create tumor with copy number but no mutation data
tumor &lt;- Tumor(psis, rounds = 400, nu = 0, pcnv = 1, norm.contam = FALSE)
# simulate the dataset
dataset &lt;- generateTumorData(tumor, 10000, 600000, 70, 25, 0.15, 0.03, 0.1)
#plot it
plotTumorData(tumor, dataset)
</code></pre>


</div>