<div class="container">

<table style="width: 100%;"><tr>
<td>Seeking Clones</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Seeking Tumor Clones From Data</h2>

<h3>Description</h3>

<p>Starting with copy number segmentation data and/or sequencing mutation
data for a tumor, seek the number of clones, the fraction of cells
belonging to each clone, and the likely set of abnormalities in each
clone.
</p>


<h3>Usage</h3>

<pre><code class="language-R">seekClones(cndata, vardata, cnmodels, psiset, pars, imputedCN = NULL)
runAlg(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>cndata</code></td>
<td>
<p> A data frame with seven columns; can also be NULL. The
names of the required columns are enumerated in the  man page for
<code>generateTumorData</code>. These are the same as the output typically
produced by the <code>DNAcopy</code> algorithm, where each row represents a
segment (a contiguous region of a chromosome) where the copy number is
believed to be constant.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>vardata</code></td>
<td>
<p> A data frame with eight columns; can also be NULL. The
names of the required columns are enumerated in the man page for
<code>generateTumorData</code>. These are typical of the output of a DNA
sequencing experiment that has been processed to identify variants,
which may be either germline or somatic.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cnmodels</code></td>
<td>
<p> A matrix. Each row represents a model to be considered;
each column represents a clone.  The entries are integers specifying
the number of DNA copies present in that clone. See details.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>psiset</code></td>
<td>
<p> A matrix. Each column represents a clone, and each row
represents a different possible model of the fraction of cells per
clone. See details. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pars</code></td>
<td>
<p> A list of algorithm parameters; see details.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>imputedCN</code></td>
<td>
<p>a logical value; if missing, should the copy number be
imputed from the mutation data.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>additional variables</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The algorithm starts with an initial set of 'psi' parameters
(representing the fraction of tumor belonging to each clone). It
computes the best (maximum a posteriori) clonal copy number and/or
number of mutated alleles for each clone for each segment/mutation,
conditional on the data and each of the initial psi vectors. It then
computes the posterior probability for each psi-vector and its
computed copy number and mutation parameters. It uses these posterior
probabilities to resample new possible psi-vectors. The process
repeats iteratively, and with each iterations obtains a better
estimate of psi and the clonal segment copy number and mutation
assignments until it terminates.
</p>
<p>The set of copy number models that we use is typically
generated using the following command:
<code>as.matrix(expand.grid(lapply(1:5, function(i){0:5})))</code>
This setup considers all (7776) possible models with up to five clones,
where the copy number for each clone ranges from 0 to 5. (In the future,
we are likely to make this the default; right now, you have to generate
these models yourself.)
</p>
<p>The set of possible psi-vectors (that is, the fraction of cells allocated
to each clone) that we use is typically generated using the following
command:
<code>psis.20 &lt;- generateSimplex(20,5)</code>
This setup considers all (192) possible divisions of the tumor into up to
five clones, where the fraction of cells per clone is any possible
multiple of 0.05. Each row is sorted to put the most abundant clones first,
which makes it easier to identify specific clones, except in the rare case
when two clones contain exactly the same fraction of cells. (In the future,
we are likely to make this the default; right now, you have to generate
these models yourself.)
</p>
<p>The object <code>pars</code> is a list of numerical algorithm parameters. The
elements are:
</p>

<dl>
<dt>sigma0</dt>
<dd>
<p>The standard deviation of measured allelic copy
number at the SNP level.</p>
</dd>
<dt>ktheta</dt>
<dd>
<p>The probability parameter of the geometric prior
distribution on <code>K</code>, the number of clones.</p>
</dd> 
<dt>theta</dt>
<dd>
<p>The probability parameter of the geometric prior
distribution on genomic copy number.</p>
</dd>
<dt>mtheta</dt>
<dd>
<p>The probability parameter of the geometric prior
distribution on the occurence of point mutations.</p>
</dd>
<dt>alpha</dt>
<dd>
<p>The (repeated) alpha parameter of a symmetric Dirichlet
distributed prior on the  fractions of cells belong to each clone;
default value is 0.5, giving a Jeffreys Prior.</p>
</dd>
<dt>thresh</dt>
<dd>
<p>The threshold determining the smallest possible
detectable clone.</p>
</dd>
<dt>cutoff</dt>
<dd>
<p>SNP array segments with fewer markers than this are
excluded.</p>
</dd> 
<dt>Q</dt>
<dd>
<p>Determines the number of new <code>psi</code> vectors resampled
from the estimated posterior probability distribution at each
iteration of the algorithm </p>
</dd> 
<dt>iters</dt>
<dd>
<p>The number of iterations in the algorithm.</p>
</dd>
</dl>
<p>The default settings we used are from
commonly used unfinformative priors (e.g., alpha=0.5 for the Dirichlet
distribution is the Jeffreys Prior) or based on empirical assessments
of the variation in data (sigma0, for example, which describes
variation in SNP array data).
</p>
<p>Note that <code>runAlg</code> (an alias for <code>seekClones</code>) is DEPRECATED.
</p>


<h3>Value</h3>

<p>The <code>seekClones</code> function returns a (rather long) list containing:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>psi</code></td>
<td>
<p>The most likely posterior psi-vector, given the data. The
number of non-zero entries is the number of clones found, and the non-zero
entries are the fraction of cells per clone</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>A</code></td>
<td>
<p>The most likely copy numbers for the A allele in each segment in each clone.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>B</code></td>
<td>
<p>The most likely copy numbers for the B allele in each segment in each clone.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>psibank</code></td>
<td>
<p>A matrix, where each row is one of the psi-vectors
considered during the analysis.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>psiPosts</code></td>
<td>
<p>A numeric vector, the (marginal) posterior probability of
each psi-vector considered during the analysis.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>indices</code></td>
<td>
<p>???</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>data</code></td>
<td>
<p>a list with two data-frame components containing the data
used during the analysis.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>filtered.data</code></td>
<td>
<p>a list with two data-frame components containing the
filtered data used during the analysis. Filtering removes
non-informative segments that have normal copy number or contain only
germline mutations.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>etaA</code></td>
<td>
<p>A vector of the weighted average allelic copy number for the 'A-Allele' at each segment (that is, the sum of the clonal A-allelic copy number values multiplied by the fraction of the tumor made up by each clone)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>etaB</code></td>
<td>
<p>A vector of the weighted average allelic copy number for the 'B-Allele' at each segment</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>etaM</code></td>
<td>
<p>A vector of the weighted average number of copies of the mutated allele at each mutation</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mutated</code></td>
<td>
<p>A matrix of the number of mutated alleles at each locus in each clone, where the number of rows is the number of somatic mutations in the data and the number of columns is the number of clones</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>Kevin R. Coombes <a href="mailto:krc@silicovore.com">krc@silicovore.com</a>,
Mark Zucker <a href="mailto:zucker.64@buckeyemail.osu.edu">zucker.64@buckeyemail.osu.edu</a>
</p>


<h3>References</h3>

<p>Zucker MR, Abruzzo LV, Herling CD, Barron LL, Keating MJ, Abrams ZB,
Heerema N, Coombes KR. Inferring Clonal Heterogeneity in Cancer using
SNP Arrays and Whole Genome Sequencing. Bioinformatics. To
appear. doi: 10.1093/bioinformatics/btz057.
</p>


<h3>Examples</h3>

<pre><code class="language-R"># set up models
psis.20 &lt;- generateSimplex(20,5)
cnmodels &lt;- as.matrix(expand.grid(lapply(1:5, function(i){ 0:5 })))
# set up algortihm parameters
pars &lt;- list(sigma0=5, theta = 0.9, ktheta = 0.3, mtheta = 0.9,
             alpha = 0.5, thresh = 0.04, cutoff = 100, Q = 100, iters = 4)
# create a tumor
psis &lt;- c(0.6, 0.3, 0.1) # three clones
tumor &lt;- Tumor(psis, rounds = 100, nu = 0, pcnv = 1, norm.contam = FALSE)
# simulate a dataset
dataset &lt;- generateTumorData(tumor, 10000, 600000, 70, 25, 0.15, 0.03, 0.1)
result &lt;- seekClones(dataset$cn.data, dataset$seq.data,
             cnmodels, psis.20, pars = pars, imputedCN = NULL) 
</code></pre>


</div>