<div class="container">

<table style="width: 100%;"><tr>
<td>motif_plot</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Code for plotting the Motif based on a specific CDR3 length and V gene (see netplot_ClusTCR2 for ).</h2>

<h3>Description</h3>

<p>Code for plotting the Motif based on a specific CDR3 length and V gene (see netplot_ClusTCR2 for ).
</p>


<h3>Usage</h3>

<pre><code class="language-R">motif_plot(
  ClusTCR,
  Clust_column_name = "Clust_size_order",
  Clust_selected = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>ClusTCR</code></td>
<td>
<p>Matrix file produce from mcl_cluster</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Clust_column_name</code></td>
<td>
<p>Name of clustering column from mcl_cluster file e.g. cluster</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Clust_selected</code></td>
<td>
<p>Select which cluster to display. Only one at a time.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A ggplot object representing the motif.
</p>


<h3>Examples</h3>

<pre><code class="language-R"># Example usage of mcl_cluster function with a stored file
example_file &lt;- read.csv(system.file("extdata", "my_data.csv",package = "ClusTCR2"))
# Perform clustering using mcl_cluster function
step1 &lt;- ClusTCR(example_file,allele = FALSE)
# perform mcl
step2 &lt;- mcl_cluster(step1)
# print the motif plot for the simple clustering
print(motif_plot(step2,Clust_selected = 1))
</code></pre>


</div>