<div class="container">

<table style="width: 100%;"><tr>
<td>ClusTCR</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Creates ClusTCR matrix
This function identifies similar CDR3 amino acid sequences based on the same length and V_gene</h2>

<h3>Description</h3>

<p>Creates ClusTCR matrix
This function identifies similar CDR3 amino acid sequences based on the same length and V_gene
</p>


<h3>Usage</h3>

<pre><code class="language-R">ClusTCR(my_file, allele = NULL, v_gene = "v_call")
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>my_file</code></td>
<td>
<p>uploaded file with junction_aa (CD3 sequences), variable gene.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>allele</code></td>
<td>
<p>The allele, if present as *00 will be removed if the user requires it.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>v_gene</code></td>
<td>
<p>Variable gene column name</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>X by Y matrix of structurally related CDR3 sequences.
</p>


<h3>Examples</h3>

<pre><code class="language-R"># Example usage of ClusTCR function with a stored file
example_file &lt;- read.csv(system.file("extdata", "my_data.csv", package = "ClusTCR2"))
# Perform clustering using ClusTCR function
step1 &lt;- ClusTCR(example_file, allele = FALSE)
# Print the result
print(step1)
</code></pre>


</div>