<div class="container">

<table style="width: 100%;"><tr>
<td>ClussCluster</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Performs simultaneous detection of cell types and cell-type-specific signature genes</h2>

<h3>Description</h3>

<p><code>ClussCluster</code> takes the single-cell transcriptome data and returns an object containing cell types and type-specific signature gene sets
</p>
<p>Selects the tuning parameter in a permutation approach. The tuning parameter controls the L1 bound on w, the feature weights.
</p>


<h3>Usage</h3>

<pre><code class="language-R">ClussCluster(x, nclust = NULL, centers = NULL, ws = NULL,
  nepoch.max = 10, theta = NULL, seed = 1, nstart = 20,
  iter.max = 50, verbose = FALSE)

ClussCluster_Gap(x, nclust = NULL, B = 20, centers = NULL,
  ws = NULL, nepoch.max = 10, theta = NULL, seed = 1,
  nstart = 20, iter.max = 50, verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>An nxp data matrix. There are n cells and p genes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nclust</code></td>
<td>
<p>Number of clusters desired if the cluster centers are not provided. If both are provided, nclust must equal the number of cluster <code>centers</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>centers</code></td>
<td>
<p>A set of initial (distinct) cluster centres if the number of clusters (<code>nclust</code>) is null. If both are provided, the number of cluster centres must equal <code>nclust</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ws</code></td>
<td>
<p>One or multiple candidate tuning parameters to be evaluated and compared. Determines the sparsity of the selected genes. Should be greater than 1.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nepoch.max</code></td>
<td>
<p>The maximum number of epochs. In one epoch, each cell will be evaluated to determine if its label needs to be updated.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>theta</code></td>
<td>
<p>Optional argument. If provided, <code>theta</code> are used as the initial cluster labels of the ClussCluster algorithm; if not, K-means is performed to produce starting cluster labels.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>seed</code></td>
<td>
<p>This seed is used wherever K-means is used.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nstart</code></td>
<td>
<p>Argument passed to <code>kmeans</code>. It is the number of random sets used in <code>kmeans</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>iter.max</code></td>
<td>
<p>Argument passed to <code>kmeans</code>. The maximum number of iterations allowed.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>Print the updates inside every epoch? If TRUE, the updates of cluster label and the value of objective function will be printed out.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>B</code></td>
<td>
<p>Number of permutation samples.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Takes the normalized and log transformed number of reads mapped to genes (e.g., log(RPKM+1) or log(TPM+1) where RPKM stands for Reads Per Kilobase of transcript per Million mapped reads and TPM stands for transcripts per million) but NOT centered.
</p>


<h3>Value</h3>

<p>a list containing the optimal tuning parameter, <code>s</code>, group labels of clustering, <code>theta</code>, and type-specific weights of genes, <code>w</code>.
</p>
<p>a list containig a vector of candidate tuning parameters, <code>ws</code>, the corresponding values of objective function, <code>O</code>, a matrix of values of objective function for each permuted data and tuning parameter, <code>O_b</code>, gap statistics and their one standard deviations, <code>Gap</code> and <code>sd.Gap</code>, the result given by <code>ClussCluster</code>, <code>run</code>, the tuning parameters with the largest Gap statistic and within one standard deviation of the largest Gap statistic, <code>bestw</code> and <code>onesd.bestw</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(Hou_sim)
hou.dat &lt;-Hou_sim$x
run.ft &lt;- filter_gene(hou.dat)
hou.test &lt;- ClussCluster(run.ft$dat.ft, nclust=3, ws=4, verbose = FALSE)
</code></pre>


</div>