<div class="container">

<table style="width: 100%;"><tr>
<td>clusterRepro</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Gene expression clusters reproducibility and validation</h2>

<h3>Description</h3>

<p>Validate gene expression clusters by determining whether or not they
are reproducible
</p>


<h3>Usage</h3>

<pre><code class="language-R">clusterRepro(Centroids, New.data, Number.of.permutations)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>Centroids</code></td>
<td>
<p>The matrix of centroids with annotated rows.  The
labeled rows are either genes (for gene clusters) or samples (for
sample clusters) and the columns are the centroids.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>New.data</code></td>
<td>
<p>The matrix of gene expression data (with annotated
rows) independent of the dataset used to form the centroids.  For
gene clusters, the rows are samples and the columns are genes.  For
sample clusters, the rows are genes and the columns are samples.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Number.of.permutations</code></td>
<td>
<p>The number of times the centroids will
be permuted to generate the null distribution.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function looks for gene expression clusters found in one
dataset in another independent dataset.  The centroids from the first
dataset are used to classify the independent data and the
corresponding in-group proportions (IGPs) are computed.  These in-group
proportions are compared to null distributions of in-group proportions
to produce p-values.  The IGP null distributions are generated by
repeatedly permuting the centroids within the box aligned with the principal
components, classifying the independent data, and calculating the
corresponding IGPs.</p>


<h3>Value</h3>

<table>
<tr style="vertical-align: top;">
<td><code>Actual.Size</code></td>
<td>
<p>The number of columns of New.data assigned to each centroid.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Actual.IGP</code></td>
<td>
<p>The in-group proportions of the groups formed when
New.data is classified using Centroids.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p.value</code></td>
<td>
<p>The p-values for each of the groups represented by the centroids.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Number</code></td>
<td>
<p>The number of permutations used to compute the
corresponding p-value.</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>Amy Kapp and Robert Tibshirani</p>


<h3>References</h3>

<p>Amy Kapp and Robert Tibshirani.  Are clusters in one dataset present
in another dataset?  To be published.
</p>


<h3>See Also</h3>

<p><code>IGP.clusterRepro</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">### Generate centroids with annotated rows
Centroids &lt;- matrix(rnorm(30, sd = 10), 10)
rownames(Centroids) &lt;- letters[1:nrow(Centroids)]

### Generate data with annotated rows
Data &lt;- cbind(matrix(rep(Centroids[,1], 10), 10),
matrix(rep(Centroids[,2], 15), 10), matrix(rep(Centroids[,3], 20), 10))
Data &lt;- Data + matrix(rnorm(length(Data), sd = 10), nrow(Data))
rownames(Data) &lt;- letters[1:nrow(Data)]

### Classify the data and calculate the corresponding in-group
### proportions and group size
Result &lt;- clusterRepro(Centroids, Data, Number.of.permutations = 1)
Result$Actual.IGP
Result$Actual.Size

### Generate null distributions and compare to actual in-group proportions to obtain p-values
Result2 &lt;- clusterRepro(Centroids, Data, Number.of.permutations = 1000)

### If the number of rows in the centroid matrix does not match the
### number of rows in the data matrix and the row labels are unique, this
### function will only use the rows that the two matrices have in common. 
Data &lt;- matrix(rnorm(200), 20)
rownames(Data) &lt;- letters[(nrow(Data)+6):7]
Result &lt;- IGP.clusterRepro(Data, Centroids)
Result2 &lt;- clusterRepro(Centroids, Data, Number.of.permutations = 1000)
</code></pre>


</div>