<div class="container">

<table style="width: 100%;"><tr>
<td>imd</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Calculate of the inter-mutational distance (IMD).
</h2>

<h3>Description</h3>

<p>The inter-mutational distance (IMD) is the distance between each somatic substitution and the substitution immediately prior (Nik-Zainal et al. 2012). The <code>imd()</code> is used to graph the rainfall plot (Nik-Zainal et al. 2012). This graph localizes the regional clustering of mutations and represents the IMD on the y-axis on a log base 10 scale, with mutations ranked and ordered on the x-axis from the first variant on the short arm of chromosome 1 to the last variant on the long arm of chromosome X.
</p>


<h3>Usage</h3>

<pre><code class="language-R">imd(data = NULL, chr = NULL, position, extra = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>data</code></td>
<td>
<p>: somatic substitution mutations of the cancer genome data set.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chr</code></td>
<td>
<p>: chromosome where the somatic mutation is located.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>position</code></td>
<td>
<p>: position of somatic mutations in the DNA sequence of the cancer genome.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>extra</code></td>
<td>
<p>: additional information to posterior graph.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p><code>imd()</code> returns a data set with six variables: 
</p>
<table>
<tr style="vertical-align: top;">
<td><code>chr</code></td>
<td>
<p>: chromosome where the somatic mutation is located.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>position</code></td>
<td>
<p>: position of somatic mutations in the DNA sequence of the cancer genome.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>extra</code></td>
<td>
<p>: additional information.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>number</code></td>
<td>
<p>: ordered mutations.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>distance</code></td>
<td>
<p>: Euclidean distance between each somatic substitution and the substitution immediately prior (IMD) (Nik-Zainal et al. 2012).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>log10distance</code></td>
<td>
<p>: Euclidean distance between each somatic substitution and the substitution immediately prior in base 10.</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>David Lora
</p>


<h3>References</h3>

<p>Nik-Zainal S, Alexandrov LB, Wedge DC, et al; Breast Cancer Working Group of the International Cancer Genome Consortium. Mutational processes molding the genomes of 21 breast cancers. Cell. 2012 May 25;149(5):979-93.
</p>
<p>Wickham H. ggplot2: elegant graphics for data analysis. Springer New York, 2009.
</p>


<h3>See Also</h3>

<p><code>plot</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">
data(PD4107a)

###New variable
extra &lt;- factor(c(),levels=c("T&gt;C","T&gt;G","T&gt;A","C&gt;T","C&gt;G","C&gt;A"))
extra[PD4107a$Ref_base=="A" &amp; PD4107a$Mutant_base=="G"]&lt;-"T&gt;C"
extra[PD4107a$Ref_base=="T" &amp; PD4107a$Mutant_base=="C"]&lt;-"T&gt;C"
extra[PD4107a$Ref_base=="A" &amp; PD4107a$Mutant_base=="C"]&lt;-"T&gt;G"
extra[PD4107a$Ref_base=="T" &amp; PD4107a$Mutant_base=="G"]&lt;-"T&gt;G"
extra[PD4107a$Ref_base=="A" &amp; PD4107a$Mutant_base=="T"]&lt;-"T&gt;A"
extra[PD4107a$Ref_base=="T" &amp; PD4107a$Mutant_base=="A"]&lt;-"T&gt;A"
extra[PD4107a$Ref_base=="G" &amp; PD4107a$Mutant_base=="A"]&lt;-"C&gt;T"
extra[PD4107a$Ref_base=="C" &amp; PD4107a$Mutant_base=="T"]&lt;-"C&gt;T"
extra[PD4107a$Ref_base=="G" &amp; PD4107a$Mutant_base=="C"]&lt;-"C&gt;G"
extra[PD4107a$Ref_base=="C" &amp; PD4107a$Mutant_base=="G"]&lt;-"C&gt;G"
extra[PD4107a$Ref_base=="G" &amp; PD4107a$Mutant_base=="T"]&lt;-"C&gt;A"
extra[PD4107a$Ref_base=="C" &amp; PD4107a$Mutant_base=="A"]&lt;-"C&gt;A"
PD4107a$extra&lt;-extra

###Generate new data set with intermutational distance;
rainfall&lt;-imd(data=PD4107a,chr=Chr,position=Position,extra=extra)

###Rainfall plot for PD4107a cancer sample;
plot(rainfall$number, rainfall$log10distance,pch=20,
	ylab="Intermutation distance (bp)",xlab="PD4107a",yaxt="n",
	col=c(rep(c("black","red"),14)[rainfall$chr]))
axis(2, at=c(0,1,2,3,4,6), labels=c("1","10","100","1000","10000","1000000"),
	las=2, cex.axis=0.6)

###Rainfall plot for PD4107a cancer sample (Nik-Zainal et al. 2012);
#require(ggplot2)
#graph &lt;- qplot(data=rainfall,number,log10distance,colour=extra, ylim=c(0.2,8), 
#	ylab="log10", xlab="PD4107a")
#graph &lt;- graph + 
#	scale_colour_manual(values =c("T&gt;C"="yellow","T&gt;G"="green","T&gt;A"="pink",
#		"C&gt;T"="red","C&gt;G"="black","C&gt;A"="blue"))
#graph &lt;- graph + theme(legend.title=element_blank())
#graph &lt;- graph + scale_y_continuous(breaks = c(0, 1, 2, 3, 4, 6),
#		labels=c("1","10","100","1000","10000","1000000"))
#graph
</code></pre>


</div>