<div class="container">

<table style="width: 100%;"><tr>
<td>features</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Describe the features of mutation in the hyper-mutated zones.
</h2>

<h3>Description</h3>

<p>Several features were observed in the hyper-mutated zones, for example, kataegis is the proposed name for the hyper-mutated zones with a cluster of C&gt;T and/or C&gt;G mutations that are substantially enriched at TpCpN trinucleotides, on the same DNA strand and that co-localize with large-scale genomic structural variation (Alexandrov et al. 2013; Nik-Zainal et al. 2012).
</p>


<h3>Usage</h3>

<pre><code class="language-R">features(data = NULL, chr = NULL, position, refbase, mutantbase,
						 min = 6, max = 5000)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>data</code></td>
<td>
<p>: somatic substitution mutations of the cancer genome data set.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chr</code></td>
<td>
<p>: chromosome where the somatic mutation is located.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>position</code></td>
<td>
<p>: position of somatic mutations in the DNA sequence of the cancer genome.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>refbase</code></td>
<td>
<p>: reference base in the chromosome.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mutantbase</code></td>
<td>
<p>: the mutant base in the chromosome.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min</code></td>
<td>
<p>: a number <code>min</code> or more consecutive mutations.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>max</code></td>
<td>
<p>: a distance less than or equal to a number <code>max</code> of bp.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>By default, <code>features()</code> identifies the mutations in the hyper-mutated zones (min = 6; max=5000). Complex mutations (Roberts et al. 2012; Roberts et al. 2013) are those segments containing &gt;= 2 consecutive mutations with a distance =&lt; 100 bp.
</p>


<h3>Value</h3>

<p><code>features()</code> returns a data set with all mutations in the hyper-mutated zones. The data set contains five variables: 
</p>
<table>
<tr style="vertical-align: top;">
<td><code>clustered</code></td>
<td>
<p>: number of cluster.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chr</code></td>
<td>
<p>: chromosome.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>position</code></td>
<td>
<p>: the position of mutation in the chromosome.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ref_base</code></td>
<td>
<p>: reference base in the chromosome.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mutant_base</code></td>
<td>
<p>: the mutant base in the chromosome.</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>David Lora
</p>


<h3>References</h3>

<p>Alexandrov LB, Nik-Zainal S, Wedge DC, et al. Signatures of mutational processes in human cancer. Nature. 2013 Aug 22;500(7463):415-21.
</p>
<p>Nik-Zainal S, Alexandrov LB, Wedge DC, et al; Breast Cancer Working Group of the International Cancer Genome Consortium. Mutational processes molding the genomes of 21 breast cancers. Cell. 2012 May 25;149(5):979-93.
</p>
<p>Roberts SA, Sterling J, Thompson C, et al. Clustered mutations in  yeast and in human cancers can arise from damaged long single-strand DNA regions. Mol Cell. 2012 May 25;46(4):424-35.
</p>
<p>Roberts SA, Lawrence MS, Klimczak LJ, et al. An APOBEC cytidine deaminase mutagenesis pattern is widespread in human cancers. Nat Genet. 2013 Sep;45(9):970-6. 
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(PD4107a)
###Show the features of mutations in the hypermutated zones;
features(data=PD4107a,chr=Chr,position=Position,refbase=Ref_base,
		mutantbase=Mutant_base)

###Locate complex mutations in the hypermutated zones;
kataegis&lt;-features(data=PD4107a,chr=Chr,position=Position,refbase=Ref_base,
		mutantbase=Mutant_base)
complex &lt;-features(data=PD4107a,chr=Chr,position=Position,
			refbase=Ref_base,mutantbase=Mutant_base,min=2,max=10)
sub.complex &lt;-subset(complex,select=c(clustered,chr,position))
data.hyperm  &lt;-merge(kataegis,sub.complex,by=c("chr","position"),all.x=TRUE) 

###Summary of the number of complex mutations in the hypermutated zones;
table(data.hyperm$clustered.x)
table(data.hyperm$clustered.y)
table(data.hyperm$clustered.y,data.hyperm$clustered.x)
data.hyperm&lt;-transform(data.hyperm,clustered = 
				ifelse(is.na(clustered.y)==FALSE,1,0))
table(data.hyperm$clustered.x)
table(data.hyperm$clustered,data.hyperm$clustered.x)
###All hypermutated zones had more than 6 mutations (without complex mutations).
</code></pre>


</div>