<div class="container">

<table style="width: 100%;"><tr>
<td>clusterhap</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>clusterhap function identifies haplotypes within QTL.</h2>

<h3>Description</h3>

<p>This function groups together all individuals of a population
with the same haplotype.
</p>


<h3>Usage</h3>

<pre><code class="language-R">clusterhap(x, Print = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>a data.frame that should be loaded with read.table function.
Each row represents the individuals while each column represents the markers.
The first column contains the names of the genotypes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Print</code></td>
<td>
<p>option for print the clusterhap result.
The default is FALSE</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Each group contains individual with the same allele in each SNP,
whether or not missing data.
</p>


<h3>Value</h3>

<p>a matrix with the haplotypes
</p>


<h3>Author(s)</h3>

<p>Gaston Quero, Sebastian Simondi, Victoria Bonnecarrere, Lucia Gutierrez.
</p>


<h3>See Also</h3>

<p>read.table function
</p>


<h3>Examples</h3>

<pre><code class="language-R">
#### Simple simulated data
data("sim_qtl")
clusterhap(sim_qtl, Print=TRUE)

### Real experimental data

## Not run: 
data(rice_qtl)
clusterhap(rice_qtl)

## End(Not run)
</code></pre>


</div>