<div class="container">

<table style="width: 100%;"><tr>
<td>clValid</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Validate Cluster Results</h2>

<h3>Description</h3>

<p><code>clValid</code> reports validation measures for clustering
results.  The function returns an object of class
<code>"clValid"</code>, which
contains the clustering results in addition to the validation
measures.  The validation measures fall into three general categories:
"internal", "stability", and "biological".
</p>


<h3>Usage</h3>

<pre><code class="language-R">clValid(obj, nClust, clMethods = "hierarchical", validation =
"stability", maxitems = 600, metric = "euclidean", method = "average",
neighbSize = 10, annotation = NULL, GOcategory = "all",
goTermFreq=0.05, dropEvidence=NULL, verbose=FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>obj</code></td>
<td>
<p>Either a numeric matrix, a data frame, or an ExpressionSet
object.  Data frames must contain all numeric columns.  In all
cases, the rows are the items to be clustered (e.g., genes),
and the columns are the samples.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nClust</code></td>
<td>
<p>A numeric vector giving the numbers of clusters
to be evaluated.  e.g., 4:6 would evaluate the number of clusters
ranging from 4 to 6.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>clMethods</code></td>
<td>
<p>A character vector giving the clustering methods.
Available options are "hierarchical", "kmeans", "diana", "fanny",
"som", "model", "sota", "pam", "clara", and "agnes", with multiple
choices allowed.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>validation</code></td>
<td>
<p>A character vector giving the type of
validation measures to use.  Available options are "internal",
"stability", and "biological", with multiple choices allowed.  </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>maxitems</code></td>
<td>
<p>The maximum number of items (rows in matrix) which can be clustered.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>metric</code></td>
<td>
<p>The metric used to determine the distance
matrix.  Possible choices are "euclidean", "correlation", and "manhattan".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method</code></td>
<td>
<p>For hierarchical clustering (<code>hclust</code> and <code>agnes</code>), the
agglomeration method used.  Available choices are "ward", "single",
"complete", and "average".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>neighbSize</code></td>
<td>
<p>For internal validation, an integer giving the neighborhood size used for the
connectivity measure.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>annotation</code></td>
<td>
<p>For biological validation,
either a character string naming the Bioconductor annotation
package for mapping genes to GO categories, or a list with the names of the functional classes
and the observations belonging to each class.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>GOcategory</code></td>
<td>
<p>For biological validation, gives which GO
categories to use for biological validation.  Can be one of "BP",
"MF", "CC", or "all".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>goTermFreq</code></td>
<td>
<p>For the
BSI, what threshold frequency of GO terms to use for functional
annotation.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dropEvidence</code></td>
<td>
<p>Which GO evidence codes to omit.  Either NULL or a
character vector, see 'Details' below.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>Logical - if TRUE will produce detailed output on the
progress of cluster validation.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Additional arguments to pass to the clustering functions.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function calculates validation measures for a given set of
clustering algorithms and number of clusters.  A variety of clustering
algorithms are available, including hierarchical, self-organizing maps
(SOM), K-means, self-organizing tree algorithm (SOTA), and model-based.
The available validation measures fall into the three general
categories of "internal", "stability", and "biological".  A brief
description of each measure is given below, for further details refer
to the package vignette and the references.
</p>

<dl>
<dt><b>Internal measures:</b></dt>
<dd>
<p>The internal
measures include the connectivity, and Silhouette Width, and Dunn
Index.  The connectivity indicates the degree of connectedness of the
clusters, as determined by the k-nearest neighbors.  The
<code>neighbSize</code> argument specifies the number of neighbors to use.
The connectivity has a value between 0 and infinity and should be minimized.
Both the Silhouette Width and the Dunn Index combine measures of
compactness and separation of the clusters.  The Silhouette Width is
the average of each observation's Silhouette value.  The Silhouette
value measures the degree of confidence in a particular clustering
assignment and lies
in the interval [-1,1], with well-clustered observations having values
near 1 and poorly clustered observations having values near -1.  See
the <code>silhouette</code> function in package <span class="pkg">cluster</span> for
more details.  The
Dunn Index is the ratio between the smallest distance between
observations not in the same cluster to the largest intra-cluster
distance.  It has a value between 0 and infinity and should be maximized.</p>
</dd>
<dt><b>Stability measures:</b></dt>
<dd>
<p>The stability measures are a special version of internal measures
which evaluate the stability of a clustering result by comparing it
with the clusters obtained by removing one column at a time.
These measures include the average proportion of non-overlap (APN),
the average distance (AD), the average distance between means (ADM),
and the figure of merit (FOM).  The APN, AD, and ADM are all based on the
cross-classification table of the original clustering with the
clustering based on the removal of one column.  The APN measures the
average proportion of observations not placed in the same cluster under both
cases, while the AD measures the average distance between observations placed
in the same cluster under both cases and the ADM measures the average
distance between cluster centers for observations placed in the same cluster
under both cases.  The FOM measures the average intra-cluster variance
of the deleted column, where the clustering is based on the remaining
(undeleted) columns.  In all cases the average is taken over all the
deleted columns, and all measures should be minimized.  </p>
</dd>
<dt><b>Biological measures:</b></dt>
<dd>
<p>There are two biological validation measures, the biological homogeneity index (BHI) and
biological stability index (BSI).  The observations are typically taken to
represent a ‘gene’ (e.g., ORF, SAGE
tag, affy ID).  The BHI measures the average
proportion of gene pairs that are clustered together which have
matching biological functional classes.
The BSI is similar to the other stability
measures, but 
inspects the consistency of
clustering for genes with similar biological functionality. Each sample is removed one
at a time, and the cluster membership for genes with similar functional annotation is
compared with the cluster membership using all available samples.
</p>
<p>For biological validation, the user has two options.
The first option is to explicity specify the
functional clustering of the genes via either a named list or logical
matrix. In “list” format, each item in the list is a vector giving genes
belonging to a particular biological class.  In “matrix” format,
each column is a logical vector indicating which genes belong to the
biological class. <code>clValid</code> will convert the biological
annotation to matrix format internally if initially given in list format.
</p>
<p>The second option is to specify the appropriate
annotation package from Bioconductor (<a href="https://www.bioconductor.org/">https://www.bioconductor.org/</a>)
and GO terms to determine the functional classes of
the genes.  To use the second option requires the <span class="pkg">Biobase</span>,
<span class="pkg">annotate</span>, and <span class="pkg">GO</span> packages from Bioconductor, in addition
to the annotation package for the particular data type.
If the annotation package cannot be loaded, <code>clValid</code> will
attempt to automatically download the package from
<a href="https://www.bioconductor.org/">https://www.bioconductor.org/</a> (using the
‘<span class="file">biocLite.R</span>’ installation script).
</p>
<p>The <code>GOcategory</code> options are "MF", "BP", "CC", or "all",
corresponding to molecular function, biological process, cellular
component, and all of the ontologies.
</p>
<p>The <code>dropEvidence</code> argument indicates which GO evidence codes to
omit.  For example, "IEA" is a relatively weak association based only
on electronic information, and users may wish to omit this evidence
when determining the functional annotation classes.
</p>
</dd>
</dl>
<h3>Value</h3>

<p><code>clValid</code> returns an object of class
<code>"clValid"</code>.  See the help file for the class description.
</p>


<h3>Note</h3>

<p>The only package which is automatically attached is <span class="pkg">cluster</span>.  To
use the clustering methods <code>som</code> and <code>Mclust</code> you will need
to load the packages <span class="pkg">kohonen</span> and <span class="pkg">mclust</span>, respectively.
</p>
<p>Unless the the list of genes corresponding to functional classes is prespecified,
to perform biological clustering validation
will require the <span class="pkg">Biobase</span>, <span class="pkg">annotate</span> and <span class="pkg">GO</span> packages from
Bioconductor, and in addition the annotation package for your
particular data type.  Please see <a href="https://www.bioconductor.org/">https://www.bioconductor.org/</a> for installation instructions.
</p>
<p>Further details of the validation measures and instructions in use can
be found in the package vignette.
</p>


<h3>Author(s)</h3>

<p>Guy Brock, Vasyl Pihur, Susmita Datta, Somnath Datta</p>


<h3>References</h3>

<p>Brock, G., Pihur, V., Datta, S. and Datta, S. (2008)
clValid: An R Package for Cluster Validation
Journal of Statistical Software 25(4)
<a href="https://www.jstatsoft.org/v25/i04/">https://www.jstatsoft.org/v25/i04/</a>
</p>
<p>Datta, S. and Datta, S. (2003)
Comparisons and validation of statistical clustering techniques for microarray gene expression data.
Bioinformatics 19(4): 459-466
</p>
<p>Datta, S. and Datta, S. (2006)
Methods for evaluating clustering algorithms for gene expression data
using a reference set of functional classes.
BMC Bioinformatics 7:397
<a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-7-397/">https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-7-397/</a>
</p>
<p>Handl, J., Knowles, K., and Kell, D. (2005)
Computational cluster validation in post-genomic data analysis.
Bioinformatics 21(15): 3201-3212
</p>


<h3>See Also</h3>

<p>For a description of the class 'clValid' and all available methods see <code>clValidObj</code>
or <code>clValid-class</code>.
</p>
<p>For help on the clustering methods see <code>hclust</code> and
<code>kmeans</code>  in package <span class="pkg">stats</span>, 
<code>agnes</code>, <code>clara</code>,  <code>diana</code>,
<code>fanny</code>, and <code>pam</code> in package <span class="pkg">cluster</span>,
<code>supersom</code> in package <span class="pkg">kohonen</span>, <code>Mclust</code>
in package <span class="pkg">mclust</span>, and <code>sota</code> (in this package).
</p>
<p>For additional help on the validation measures see
<code>connectivity</code>,   <code>dunn</code>,
<code>stability</code>, 
<code>BHI</code>, and 
<code>BSI</code>. 
</p>



<h3>Examples</h3>

<pre><code class="language-R">

data(mouse)

## internal validation
express &lt;- mouse[1:25,c("M1","M2","M3","NC1","NC2","NC3")]
rownames(express) &lt;- mouse$ID[1:25]
intern &lt;- clValid(express, 2:4, clMethods=c("hierarchical","kmeans","pam"),
                  validation="internal")

## view results
summary(intern)
optimalScores(intern)
plot(intern)

## stability measures
stab &lt;- clValid(express, 2:4, clMethods=c("hierarchical","kmeans","pam"),
                validation="stability")
optimalScores(stab)
plot(stab)

## biological measures
## first way - functional classes predetermined
fc &lt;- tapply(rownames(express),mouse$FC[1:25], c)
fc &lt;- fc[-match( c("EST","Unknown"), names(fc))]
bio &lt;- clValid(express, 2:4, clMethods=c("hierarchical","kmeans","pam"),
               validation="biological", annotation=fc)
optimalScores(bio)
plot(bio)

## second way - using Bioconductor
if(require("Biobase") &amp;&amp; require("annotate") &amp;&amp; require("GO.db") &amp;&amp; require("moe430a.db")) {
  bio2 &lt;- clValid(express, 2:4, clMethods=c("hierarchical","kmeans","pam"),
                  validation="biological",annotation="moe430a.db",GOcategory="all")
  optimalScores(bio2)
  plot(bio2)
}

</code></pre>


</div>