<div class="container">

<table style="width: 100%;"><tr>
<td>clValid-class</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Class "clValid"</h2>

<h3>Description</h3>

<p>The class <code>"clValid"</code> contains the clustering results
and validation measures from the accompanying call to the function
<code>clValid</code>.
</p>


<h3>Objects from the Class</h3>

<p>Objects can be created using the function <code>clValid</code>.
</p>


<h3>Slots</h3>


<dl>
<dt>
<code>clusterObjs</code>:</dt>
<dd>
<p>Object of class <code>"list"</code>. A list
containing the results from the clustering methods.</p>
</dd>
<dt>
<code>measures</code>:</dt>
<dd>
<p>Object of class <code>"array"</code>. A
3-dimensional array which contains the
validation measures for the clustering results.  The first
dimension indicates the validation measures, the second the number of
clusters, and the third the clustering methods.</p>
</dd>
<dt>
<code>measNames</code>:</dt>
<dd>
<p>Object of class <code>"character"</code>. The
names of the validation measures.</p>
</dd>
<dt>
<code>clMethods</code>:</dt>
<dd>
<p>Object of class <code>"character"</code>. A
character vector giving the clustering methods.</p>
</dd>
<dt>
<code>labels</code>:</dt>
<dd>
<p>Object of class <code>"character"</code>. A
character vector giving the item (gene) labels.</p>
</dd>    
<dt>
<code>nClust</code>:</dt>
<dd>
<p>Object of class <code>"numeric"</code>. A numeric
vector giving the numbers of clusters
which were evaluated.</p>
</dd>
<dt>
<code>validation</code>:</dt>
<dd>
<p>Object of class <code>"character"</code>. A character vector giving the type of
validation measures used, consisting of some combination of
"internal", "stability", or "biological".</p>
</dd>
<dt>
<code>metric</code>:</dt>
<dd>
<p>Object of class <code>"character"</code>. The metric used to determine the distance
matrix.</p>
</dd>
<dt>
<code>method</code>:</dt>
<dd>
<p>Object of class <code>"character"</code>. For
hierarchical clustering, the agglomeration method used.</p>
</dd>
<dt>
<code>neighbSize</code>:</dt>
<dd>
<p>Object of class <code>"numeric"</code>. For internal validation, the neighborhood size used for the
connectivity measure.</p>
</dd>
<dt>
<code>annotation</code>:</dt>
<dd>
<p>Object of class <code>"character or array
	or list"</code>.
Either a character string naming the Bioconductor annotation
package for mapping genes to GO categories, or a list with the names of the functional classes
and the observations belonging to each class.</p>
</dd>
<dt>
<code>GOcategory</code>:</dt>
<dd>
<p>Object of class <code>"character"</code>. For biological validation, gives which GO
categories to use for biological validation.  Can be one of "BP",
"MF", "CC", or "all"</p>
</dd>
<dt>
<code>goTermFreq</code>:</dt>
<dd>
<p>Object of class <code>"numeric"</code>. For the
BSI, what threshold frequency of GO terms to use for functional annotation.</p>
</dd>
<dt>
<code>call</code>:</dt>
<dd>
<p>Object of class <code>"call"</code>.  Gives the call
to <code>clValid</code> used to create the <code>clValid</code> object.</p>
</dd>
</dl>
<h3>Methods</h3>


<dl>
<dt>clusterMethods</dt>
<dd>
<p><code>signature(object = "clValid")</code>: Returns the
names of the clustering methods. </p>
</dd>
<dt>clusters</dt>
<dd>
<p><code>signature(object = "clValid")</code>: Returns the
results from the clustering methods.
</p>
<p>Additional arguments:
</p>

<dl>
<dt><code>method = clMethods(object)</code></dt>
<dd>
<p>The clustering method(s)
to extract.</p>
</dd>
</dl>
</dd>
<dt>measNames</dt>
<dd>
<p><code>signature(object = "clValid")</code>: Returns the
names of the validation measures.</p>
</dd>
<dt>measures</dt>
<dd>
<p><code>signature(object = "clValid")</code>: Returns
the validation measures.
</p>
<p>Additional arguments:
</p>

<dl>
<dt><code>measures = measNames(object)</code></dt>
<dd>
<p> The validation measure(s)
to extract.</p>
</dd>
</dl>
</dd>
<dt>nClusters</dt>
<dd>
<p><code>signature(object = "clValid")</code>: Returns the
numbers of clusters evaluated.</p>
</dd>
<dt>optimalScores</dt>
<dd>
<p><code>signature(object = "clValid")</code>:
Returns the optimal value for each validation measure, along with
the corresponding clustering method and number of clusters.
</p>
<p>Additional arguments:
</p>

<dl>
<dt><code>measures = measNames(object)</code></dt>
<dd>
<p> The validation measure(s)
to extract.</p>
</dd>
</dl>
</dd>
<dt>plot</dt>
<dd>
<p><code>signature(x = "clValid", y = "missing")</code>: Plots
the validation measures.
</p>
<p>Additional arguments:
</p>

<dl>
<dt><code>measures=measNames(x)</code></dt>
<dd>
<p> The validation measures to
plot.</p>
</dd>
<dt><code>legend=TRUE</code></dt>
<dd>
<p> If TRUE provides a legend.</p>
</dd>
<dt><code>legendLoc="topright"</code></dt>
<dd>
<p> The location of the legend.</p>
</dd>
<dt><code>main=NULL</code></dt>
<dd>
<p> Title of graph.</p>
</dd>
<dt><code>pch=NULL</code></dt>
<dd>
<p> Plotting characters to use.</p>
</dd>
<dt><code>type="b"</code></dt>
<dd>
<p> Type of plot.</p>
</dd>
<dt><code>ask=prod(par("mfcol")) &lt; length(measures) &amp;&amp;
	    dev.interactive()</code></dt>
<dd>
<p> Logical.  If <code>TRUE</code> the user is
prompted before each plot.</p>
</dd>
</dl>
</dd>
<dt>print</dt>
<dd>
<p><code>signature(x = "clValid")</code>: Print method for class
<code>clValid</code>.</p>
</dd>
<dt>show</dt>
<dd>
<p><code>signature(object = "clValid")</code>: Same as print.</p>
</dd>
<dt>summary</dt>
<dd>
<p><code>signature(object = "clValid")</code>: Summary method
for class <code>clValid</code>.
</p>
<p>Additional arguments:
</p>

<dl>
<dt><code>digits = max(3,getOption("digits")-3)</code></dt>
<dd>
<p> The number
of significant digits to use.</p>
</dd>
</dl>
</dd>
</dl>
<h3>Note</h3>

<p>See the vignette for an illustration of the class.</p>


<h3>Author(s)</h3>

<p>Guy Brock, Vasyl Pihur, Susmita Datta, Somnath Datta</p>


<h3>References</h3>

<p>Brock, G., Pihur, V., Datta, S. and Datta, S. (2008)
clValid: An R Package for Cluster Validation
Journal of Statistical Software 25(4)
<a href="https://www.jstatsoft.org/v25/i04/">https://www.jstatsoft.org/v25/i04/</a>
</p>
<p>Datta, S. and Datta, S. (2003)
Comparisons and validation of statistical clustering techniques for microarray gene expression data.
Bioinformatics 19(4): 459-466.
</p>
<p>Datta, S. and Datta, S. (2006)
Methods for evaluating clustering algorithms for gene expression data
using a reference set of functional classes.
BMC Bioinformatics 7:397.
<a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-7-397/">https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-7-397/</a>
</p>
<p>Handl, J., Knowles, K., and Kell, D. (2005)
Computational cluster validation in post-genomic data analysis.
Bioinformatics 21(15): 3201-3212.
</p>


<h3>See Also</h3>

<p>For a description of the function 'clValid' see <code>clValid</code>.
</p>
<p>For help on the clustering methods see <code>hclust</code> and
<code>kmeans</code> in package <span class="pkg">stats</span>,
<code>kmeans</code>  in package <span class="pkg">stats</span>, 
<code>agnes</code>, <code>clara</code>,  <code>diana</code>,
<code>fanny</code>, and <code>pam</code> in package <span class="pkg">cluster</span>,
<code>supersom</code> in package <span class="pkg">kohonen</span>, <code>Mclust</code>  
in package <span class="pkg">mclust</span>, and <code>sota</code>.
</p>
<p>For additional help on the validation measures see
<code>connectivity</code>,   <code>dunn</code>,
<code>stability</code>, 
<code>BHI</code>, and 
<code>BSI</code>. 
</p>


<h3>Examples</h3>

<pre><code class="language-R">
## to delete
library(clValid)


data(mouse)

## internal validation
express &lt;- mouse[1:25,c("M1","M2","M3","NC1","NC2","NC3")]
rownames(express) &lt;- mouse$ID[1:25]
intern &lt;- clValid(express, 2:6, clMethods=c("hierarchical","kmeans","pam"),
                  validation="internal")
slotNames(intern)

## view results
intern
summary(intern)
optimalScores(intern)
plot(intern)

## Extract objects from slots
measures(intern)
hierClust &lt;- clusters(intern,"hierarchical")
plot(hierClust)
measNames(intern)
nClusters(intern)

</code></pre>


</div>