<div class="container">

<table style="width: 100%;"><tr>
<td>BSI</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Biological Stability Index</h2>

<h3>Description</h3>

<p>Calculates the biological stability index (BSI) for a given
statistical clustering partition and biological annotation.
</p>


<h3>Usage</h3>

<pre><code class="language-R">BSI(statClust, statClustDel, annotation, names = NULL, category = "all",
goTermFreq = 0.05, dropEvidence=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>statClust</code></td>
<td>
<p>An integer vector indicating the statistical cluster partitioning</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>statClustDel</code></td>
<td>
<p>An integer vector indicating the statistical cluster
partitioning based on one column removed</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>annotation</code></td>
<td>
<p>Either a character string naming the Bioconductor
annotation package for mapping genes to GO categories, or a
matrix where each column is a logical vector indicating which genes
belong to the biological functional class.  See details below.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>names</code></td>
<td>
<p>An optional vector of names for the observations</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>category</code></td>
<td>
<p>Indicates the GO 
categories to use for biological validation.  Can be one of "BP",
"MF", "CC", or "all".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>goTermFreq</code></td>
<td>
<p>What threshold frequency of GO terms to use for
functional annotation.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dropEvidence</code></td>
<td>
<p>Which GO evidence codes to omit.  Either NULL or a
character vector, see 'Details' below.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The BSI  inspects the consistency of clustering for genes with
similar biological functionality.  Each sample is removed, 
and the cluster membership for genes with similar functional annotation
is compared with the cluster membership using all available samples.
The BSI is in the range [0,1], with larger values corresponding to more stable
clusters of the functionally annotated genes.
For details see the package vignette.
</p>
<p>The <code>dropEvidence</code> argument indicates which GO evidence codes to
omit.  For example, "IEA" is a relatively weak association based only
on electronic information, and users may wish to omit this evidence
when determining the functional annotation classes.
</p>
<p>When inputting the biological annotation and functional classes
directly, the <code>BSI</code> function expects the input in “matrix” format,
where each column is a logical vector indicating which genes belong to the
biological class.  For details on how to input the biological
annotation from an Excel file see <code>readAnnotationFile</code> and
for converting from list to matrix format see
<code>annotationListToMatrix</code>.
</p>
<p>NOTE: The <code>BSI</code> function only calculates these measures for
one particular column removed.  To get the
overall scores, the user must average the measures
corresponding to each removed column.
</p>


<h3>Value</h3>

<p>Returns the BSI value corresponding to the particular column that was
removed.  
</p>


<h3>Note</h3>

<p>The main function for cluster validation is <code>clValid</code>, and
users should call this function directly if possible.
</p>
<p>To get the overall BSI value, the BSI values
corresponding to each removed column should be averaged (see the
examples below).
</p>


<h3>Author(s)</h3>

<p>Guy Brock, Vasyl Pihur, Susmita Datta, Somnath Datta</p>


<h3>References</h3>

<p>Datta, S. and Datta, S. (2006).
Methods for evaluating clustering algorithms for gene expression data
using a reference set of functional classes.
BMC Bioinformatics 7:397.
</p>


<h3>See Also</h3>

<p>For a description of the function 'clValid' see <code>clValid</code>.
</p>
<p>For a description of the class 'clValid' and all available methods see
<code>clValidObj</code> or <code>clValid-class</code>.
</p>
<p>For additional help on the other validation measures see
<code>connectivity</code>,   <code>dunn</code>,
<code>stability</code>, and
<code>BHI</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">

data(mouse)
express &lt;- mouse[1:25,c("M1","M2","M3","NC1","NC2","NC3")]
rownames(express) &lt;- mouse$ID[1:25]
## hierarchical clustering
Dist &lt;- dist(express,method="euclidean")
clusterObj &lt;- hclust(Dist, method="average")
nc &lt;- 4 ## number of clusters      
cluster &lt;- cutree(clusterObj,nc)

## first way - functional classes predetermined
fc &lt;- tapply(rownames(express),mouse$FC[1:25], c)
fc &lt;- fc[-match( c("EST","Unknown"), names(fc))]
fc &lt;- annotationListToMatrix(fc, rownames(express))
bsi &lt;- numeric(ncol(express))
## Need loop over all removed samples
for (del in 1:ncol(express)) {
  matDel &lt;- express[,-del]               
  DistDel &lt;- dist(matDel,method="euclidean")
  clusterObjDel &lt;- hclust(DistDel, method="average")
  clusterDel &lt;- cutree(clusterObjDel,nc)
  bsi[del] &lt;- BSI(cluster, clusterDel, fc)
}
mean(bsi)

## second way - using Bioconductor
if(require("Biobase") &amp;&amp; require("annotate") &amp;&amp; require("GO.db") &amp;&amp;
   require("moe430a.db")) {
  bsi &lt;- numeric(ncol(express))
  for (del in 1:ncol(express)) {
    matDel &lt;- express[,-del]               
    DistDel &lt;- dist(matDel,method="euclidean")
    clusterObjDel &lt;- hclust(DistDel, method="average")
    clusterDel &lt;- cutree(clusterObjDel,nc)
    bsi[del] &lt;- BSI(cluster, clusterDel, annotation="moe430a.db",
                    names=rownames(express), category="all")
  }
  mean(bsi)
}
</code></pre>


</div>