<div class="container">

<table style="width: 100%;"><tr>
<td>plot.sota</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Plot Function for a SOTA Object</h2>

<h3>Description</h3>

<p>'plot.sota' is used to obtain a visual representation of profiles within each individual cluster.
Corresponding cluster average profiles are also available. By default, plots for all clusters
are displayed side by side.
</p>


<h3>Usage</h3>

<pre><code class="language-R">## S3 method for class 'sota'
plot(x, cl = 0, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>SOTA object, an object returned by function <code>sota</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cl</code></td>
<td>
<p><code>cl</code> specifies which cluster is to be plotted by setting it to the cluster ID. By default,
<code>cl</code> is equal to 0 and the function plots all clusters side by
side.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Additional arguments to pass to <code>plot</code>.</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>Vasyl Pihur, Guy Brock, Susmita Datta, Somnath Datta</p>


<h3>References</h3>

<p>Herrero, J., Valencia,
A, and Dopazo, J. (2005). A hierarchical unsupervised growing neural
network for clustering gene expression patterns. Bioinformatics, 17, 126-136.</p>


<h3>See Also</h3>

<p><code>sota</code>, <code>print.sota</code> </p>


<h3>Examples</h3>

<pre><code class="language-R">data(mouse)
express &lt;- mouse[,c("M1","M2","M3","NC1","NC2","NC3")]
rownames(express) &lt;- mouse$ID

sotaCl &lt;- sota(as.matrix(express), 4)
names(sotaCl)
sotaCl
plot(sotaCl)
plot(sotaCl, cl=2)

</code></pre>


</div>