<div class="container">

<table style="width: 100%;"><tr>
<td>BHI</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Biological Homogeneity Index</h2>

<h3>Description</h3>

<p>Calculates the biological homogeneity index (BHI) for a given
statistical clustering partition and biological annotation.
</p>


<h3>Usage</h3>

<pre><code class="language-R">BHI(statClust, annotation, names = NULL, category = "all", dropEvidence=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>statClust</code></td>
<td>
<p>An integer vector indicating the statistical cluster partitioning</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>annotation</code></td>
<td>
<p>Either a character string naming the Bioconductor
annotation package for mapping genes to GO categories, or a
matrix where each column is a logical vector indicating which genes
belong to the biological functional class.  See details below.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>names</code></td>
<td>
<p>A vector of labels to associate with the 'genes', to be
used in conjunction with the Bioconductor annotation package.  Not
needed if <code>annotation</code> is a list providing the functional classes. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>category</code></td>
<td>
<p>Indicates the GO 
categories to use for biological validation.  Can be one of "BP",
"MF", "CC", or "all".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dropEvidence</code></td>
<td>
<p>Which GO evidence codes to omit.  Either NULL or a
character vector, see 'Details' below.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The BHI measures how homogeneous the clusters are biologically.  The
measure checks whether genes placed
in the same statistical cluster also belong to the same functional
classes.  The BHI is in the range [0,1], with
larger values corresponding to more biologically homogeneous clusters.
For details see the package vignette.
</p>
<p>When inputting the biological annotation and functional classes
directly, the <code>BSI</code> function expects the input in “matrix” format,
where each column is a logical vector indicating which genes belong to the
biological class.  For details on how to input the biological
annotation from an Excel file see <code>readAnnotationFile</code> and
for converting from list to matrix format see
<code>annotationListToMatrix</code>.
</p>
<p>The <code>dropEvidence</code> argument indicates which GO evidence codes to
omit.  For example, "IEA" is a relatively weak association based only
on electronic information, and users may wish to omit this evidence
when determining the functional annotation classes.  
</p>


<h3>Value</h3>

<p>Returns the BHI measure as a numeric value.
</p>


<h3>Note</h3>

<p>The main function for cluster validation is <code>clValid</code>, and
users should call this function directly if possible.
</p>


<h3>Author(s)</h3>

<p>Guy Brock, Vasyl Pihur, Susmita Datta, Somnath Datta</p>


<h3>References</h3>

<p>Datta, S. and Datta, S. (2006).
Methods for evaluating clustering algorithms for gene expression data
using a reference set of functional classes.
BMC Bioinformatics 7:397.
</p>


<h3>See Also</h3>

<p>For a description of the function 'clValid' see <code>clValid</code>.
</p>
<p>For a description of the class 'clValid' and all available methods see
<code>clValidObj</code> or <code>clValid-class</code>.
</p>
<p>For additional help on the other validation measures see
<code>connectivity</code>,   <code>dunn</code>,
<code>stability</code>, and
<code>BSI</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(mouse)
express &lt;- mouse[1:25,c("M1","M2","M3","NC1","NC2","NC3")]
rownames(express) &lt;- mouse$ID[1:25]
## hierarchical clustering
Dist &lt;- dist(express,method="euclidean")
clusterObj &lt;- hclust(Dist, method="average")
nc &lt;- 4 ## number of clusters      
cluster &lt;- cutree(clusterObj,nc)

## first way - functional classes predetermined
fc &lt;- tapply(rownames(express),mouse$FC[1:25], c)
fc &lt;- fc[-match( c("EST","Unknown"), names(fc))]
fc &lt;- annotationListToMatrix(fc, rownames(express))
BHI(cluster, fc)

## second way - using Bioconductor
if(require("Biobase") &amp;&amp; require("annotate") &amp;&amp; require("GO.db") &amp;&amp;
require("moe430a.db")) {
  BHI(cluster, annotation="moe430a.db", names=rownames(express), category="all")
}

</code></pre>


</div>