<div class="container">

<table style="width: 100%;"><tr>
<td>print.sota</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Print Function for a SOTA Object</h2>

<h3>Description</h3>

<p>A default print method for a SOTA object.
</p>


<h3>Usage</h3>

<pre><code class="language-R">## S3 method for class 'sota'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>a SOTA object as returned by the <code>sota</code> function</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Additional arguments to pass to <code>print</code></p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>The <code>print</code> function does not return anything. It simply displays in the console
window general information about the paritioning (cluster ID, Size, and Diversity Score),
as well as cluster centroids (average profiles within each cluster) and the distance that has
been used.
</p>


<h3>Author(s)</h3>

<p>Vasyl Pihur, Guy Brock, Susmita Datta, Somnath Datta</p>


<h3>References</h3>

<p>Herrero, J., Valencia,
A, and Dopazo, J. (2005). A hierarchical unsupervised growing neural
network for clustering gene expression patterns. Bioinformatics, 17, 126-136.</p>


<h3>See Also</h3>

<p><code>sota</code>, <code>print.sota</code> </p>


<h3>Examples</h3>

<pre><code class="language-R">data(mouse)
express &lt;- mouse[,c("M1","M2","M3","NC1","NC2","NC3")]
rownames(express) &lt;- mouse$ID

sotaCl &lt;- sota(as.matrix(express), 4)
names(sotaCl)
sotaCl
plot(sotaCl)
plot(sotaCl, cl=2)

</code></pre>


</div>