<div class="container">

<table style="width: 100%;"><tr>
<td>segexample</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Example of a segmented copy number table.
</h2>

<h3>Description</h3>

<p>Segmented log ratio data for 5 breast cancer genomes, derived using Representational
Oligonucleotide Microarray Analysis (ROMA) platform. ROMA detects genomic 
amplifications and deletions with boundaries defined at a resolution of 50 kb.
In this segmented table, each row represnts a segment.
</p>


<h3>Usage</h3>

<pre><code class="language-R">data(segexample)</code></pre>


<h3>Format</h3>

<p>A data frame with 479 rows/segments and 12 columns/variables.
</p>

<dl>
<dt><code>ID</code></dt>
<dd>
<p>a character vector of profile IDs</p>
</dd>
<dt><code>start</code></dt>
<dd>
<p>a numeric vector (segment start probe number)</p>
</dd>
<dt><code>end</code></dt>
<dd>
<p>a numeric vector (segment end probe number)</p>
</dd>
<dt><code>num.probes</code></dt>
<dd>
<p>a numeric vector (number of probes in the segment)</p>
</dd>
<dt><code>seg.median</code></dt>
<dd>
<p>a numeric vector (median log ratio)</p>
</dd>
<dt><code>chrom</code></dt>
<dd>
<p>a numeric vector (chromosome number)</p>
</dd>
<dt><code>chrom.pos.start</code></dt>
<dd>
<p>a numeric vector (genomic start)</p>
</dd>
<dt><code>chrom.pos.end</code></dt>
<dd>
<p>a numeric vector (genomic end)</p>
</dd>
<dt><code>cytoband.start</code></dt>
<dd>
<p>a character vector (cytogenetic band start)</p>
</dd>
<dt><code>cytoband.end</code></dt>
<dd>
<p>a character vector (cytogenetic band end)</p>
</dd>
<dt><code>abs.pos.start</code></dt>
<dd>
<p>a numeric vector (genomic start, absolute)</p>
</dd>
<dt><code>abs.pos.end</code></dt>
<dd>
<p>a numeric vector (genomic end, absolute)</p>
</dd>
</dl>
<h3>Details</h3>

<p>Segment medians are computed from log copy number ratio. The corresponding raw data
table is <code>ratexample</code> in this package.
</p>


<h3>Source</h3>

<p>Genome Res. 2006 Dec;16(12):1465-79.
</p>
<p>Novel patterns of genome rearrangement and their association with survival in
breast cancer.
</p>
<p>Hicks J, Krasnitz A, Lakshmi B, Navin NE, Riggs M, Leibu E, Esposito D, 
Alexander J, Troge J, Grubor V, Yoon S, Wigler M, Ye K, Borresen-Dale AL, 
Naume B, Schlicting E, Norton L, Hagerstrom T, Skoog L, Auer G, Maner S,
Lundin P, Zetterberg A.

</p>


<h3>Examples</h3>

<pre><code class="language-R">data(segexample)
</code></pre>


</div>