<div class="container">

<table style="width: 100%;"><tr>
<td>ratexample</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Example of copy number log ratio data
</h2>

<h3>Description</h3>

<p>Log ratio data for 5 breast cancer genomes, derived using Representational 
Oligonucleotide Microarray Analysis (ROMA), an array-based hybridization 
method that uses genomic complexity reduction based on representations.
</p>


<h3>Usage</h3>

<pre><code class="language-R">data(ratexample)</code></pre>


<h3>Format</h3>

<p>A log ratio data matrix with rows of 83055 oligonucleotide probes, and columns of 5
breast tumors.
</p>


<h3>Details</h3>

<p>The values are natural log copy number ratios, consistent with data in
<code>segexample</code> (segmented data for these tumors) and <code>normsegs</code>.
These copy number ratios are normalized using an intensity-based lowess 
curve fitting algorithm.
</p>


<h3>Source</h3>

<p>Genome Res. 2006 Dec;16(12):1465-79.
</p>
<p>Novel patterns of genome rearrangement and their association with survival in
breast cancer.
</p>
<p>Hicks J, Krasnitz A, Lakshmi B, Navin NE, Riggs M, Leibu E, Esposito D, 
Alexander J, Troge J, Grubor V, Yoon S, Wigler M, Ye K, Borresen-Dale AL, 
Naume B, Schlicting E, Norton L, Hagerstrom T, Skoog L, Auer G, Maner S,
Lundin P, Zetterberg A.
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(ratexample)
#Plot the whole genome log ratio data for the first profile 
#Note X and Y chromosomes at the far  right of the plot
plot(ratexample[,1])
</code></pre>


</div>