<div class="container">

<table style="width: 100%;"><tr>
<td>writeMeltedChromosomalMatrixToDisk</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Write a matrix, with genes, of a submatrix of a whole genome interaction matrix to disk.</h2>

<h3>Description</h3>

<p>Writes an RData file with a ggplot2 object within the current directory.
</p>


<h3>Usage</h3>

<pre><code class="language-R">writeMeltedChromosomalMatrixToDisk(
  whole_genome_matrix,
  chrom1,
  chrom2,
  filename,
  extra_data_matrix = NULL,
  transpose = F,
  sequential = T,
  debug = T,
  desired_range_start = 50,
  desired_range_end = 300
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>whole_genome_matrix</code></td>
<td>
<p>A matrix to have edges averaged with genomic coordinates in the form chr1_50_100 set as the column and row names.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chrom1</code></td>
<td>
<p>first chromosome of the two which will subset the matrix. (this is done in row-column fasion).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chrom2</code></td>
<td>
<p>second chromosome of the two which will subset the matrix. (this is done in row-column fasion).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>filename</code></td>
<td>
<p>the filename to be written</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>extra_data_matrix</code></td>
<td>
<p>A matrix with additional variables about each point, one position per row with as many variables as remaining columns.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>transpose</code></td>
<td>
<p>transpose the matrix?</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sequential</code></td>
<td>
<p>Disable paralleization with doParallel? registerDoSEQ() is used for this.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>debug</code></td>
<td>
<p>verbose output for debugging</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>desired_range_start</code></td>
<td>
<p>the downsampled matrix must be of this size (rows &amp; cols) at minimum</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>desired_range_end</code></td>
<td>
<p>the downsampled matrix must be of this size (rows &amp; cols) at maximum</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>ggplotmatrix a matrix with values sufficient to create a ggplot2 heatmap with geom_tile() or with ggiraph's geom_tile_interactive()
</p>


<h3>Examples</h3>

<pre><code class="language-R">load(system.file("extdata","grch37.rda",package = "CNVScope"))
load(system.file("extdata","nbl_result_matrix_sign_small.rda",package = "CNVScope"))
load(system.file("extdata","ensembl_gene_tx_table_prot.rda",package = "CNVScope"))
writeMeltedChromosomalMatrixToDisk(whole_genome_matrix = nbl_result_matrix_sign_small,
chrom1 = 1,chrom2 = 1,desired_range_start = 25, desired_range_end = 25)
file.remove("chr1_chr1_melted.RData")
</code></pre>


</div>