<div class="container">

<table style="width: 100%;"><tr>
<td>create_segsites</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Segregating Sites</h2>

<h3>Description</h3>

<p>These functions create and modify segregating sites
objects, which are one of the basic intermediary statistics that is
calculated in coala. Segregating sites consist primarily of a SNP matrix that
contains all SNPs for one locus, with some additional information attached.
The parts of the S3 class are detailed below.
</p>


<h3>Usage</h3>

<pre><code class="language-R">create_segsites(snps, positions, trio_locus = numeric(0), check = TRUE)

get_snps(segsites)

get_positions(segsites)

set_positions(segsites, positions)

get_trio_locus(segsites)

set_trio_locus(segsites, trio_locus)

is_segsites(segsites)

create_locus_trio(left, middle, right)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>snps</code></td>
<td>
<p>The SNP Matrix (see Details).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>positions</code></td>
<td>
<p>A numeric vector indicating the relative positions of each
SNP on the locus (see Details).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>trio_locus</code></td>
<td>
<p>If the locus consists of a locus trio (see Details).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>check</code></td>
<td>
<p>Whether non-segregating sites are removed from the segregating
sites (<code>TRUE</code>) or not (<code>FALSE</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>segsites</code></td>
<td>
<p>The segregating sites object</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>left</code></td>
<td>
<p>The segregating sites from the left locus</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>middle</code></td>
<td>
<p>The segregating sites from the middle locus</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>right</code></td>
<td>
<p>The segregating sites from the right locus</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>A segregating sites object contains all SNPs for one genetic locus. Each
object consists of three parts: A SNP matrix, a vector of SNP positions and
a vector that states which transcript a SNP belong to, if the locus
consists of multiple transscripts ('locus trio').
</p>

<ul>
<li>
<p>In the <strong>SNP</strong> matrix, each row represents a haplotype and each
column represents a SNP. An entry is either <code>1</code> if the
haplotype carries the derived allele for the SNP, or <code>0</code> if it
carries the ancestral one.
</p>
</li>
<li>
<p>In the <strong>positions</strong> vector, each entry gives the relative
position of SNP in the corresponding column of the SNP matrix.
</p>
</li>
<li>
<p>The <strong>trio_locus</strong> vector contains the trio locus each SNP
belongs to. Entry of <code>-1</code>,<code>0</code>, <code>1</code> represent the
left, middle, and right locus, respectively. For normal loci,
this just consists of <code>0</code>'s
</p>
</li>
</ul>
<h3>Functions</h3>


<ul>
<li> <p><code>create_segsites()</code>: Creates segregating sites
</p>
</li>
<li> <p><code>get_snps()</code>: Returns the SNP matrix from a segregating sites
object.
</p>
</li>
<li> <p><code>get_positions()</code>: Returns the SNP's positions from a segregating
sites  object.
</p>
</li>
<li> <p><code>set_positions()</code>: Sets the SNP's positions in a segregating
sites object.
</p>
</li>
<li> <p><code>get_trio_locus()</code>: Returns the trio locus positions from a
segregating sites  object.
</p>
</li>
<li> <p><code>set_trio_locus()</code>: Sets the trio locus in a segregating sites
object.
</p>
</li>
<li> <p><code>is_segsites()</code>: Checks whether an object is a segsites object.
</p>
</li>
<li> <p><code>create_locus_trio()</code>: Combines three segregating sites to a locus trio
</p>
</li>
</ul>
<h3>Author(s)</h3>

<p>Paul Staab
</p>


<h3>See Also</h3>

<p>For converting biological data to segsites: <code>as.segsites</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">snps &lt;- matrix(c(1, 0, 0,
                 1, 1, 0,
                 0, 0, 1), 3, 3, TRUE)
pos &lt;- c(.1, .3, .5)
segsites &lt;- create_segsites(snps, pos)
print(segsites)
get_snps(segsites)
get_positions(segsites)

# When subsetting individuals, sites that are not
# segregating in these are automatically removed:
segsites[1:2, 1:3]
</code></pre>


</div>