<div class="container">

<table style="width: 100%;"><tr>
<td>sumstat_four_gamete</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Summary Statistic: Four-Gamete-Condition</h2>

<h3>Description</h3>

<p>This summary statistic calculates a number of values (see 'Value')
related to the Four-Gamete-Condition (see 'Details').
It is sensitive for recombination and particularly useful when estimating
recombination rates with <span class="pkg">jaatha</span> or Approximate Bayesian Computation.
</p>


<h3>Usage</h3>

<pre><code class="language-R">sumstat_four_gamete(
  name = "four_gamete",
  population = 1,
  transformation = identity,
  na.rm = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>name</code></td>
<td>
<p>The name of the summary statistic. When simulating a model,
the value of the statistics are written to an entry of the returned list
with this name. Summary statistic names must be unique in a model.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>population</code></td>
<td>
<p>The population for which the statistic is calculated.
Can also be "all" to calculate it from all populations. Default is population 1.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>transformation</code></td>
<td>
<p>An optional function for transforming the results
of the statistic. If specified, the results of the transformation are
returned instead of the original values.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>na.rm</code></td>
<td>
<p>should missing data be ignored? Default is FALSE.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The Four-Gamete-Condition for two SNPs is violated if all four combinations
of derived and ancestral alleles at the SNPs are observed in a gamete/a
haplotype. Under an Infinite-Sites mutation model, a violation indicates that
there must have been at least one recombination event between the SNPs.
</p>


<h3>Value</h3>

<p>The statistic generates a matrix where each row represents one locus, and
the columns give the statistic for different classes of pairs of SNPs:
</p>

<dl>
<dt>mid_near</dt>
<dd>
<p>The value for all pairs of SNPs that are close together,
that is within 10 percent of the locus" length. If locus trios are used,
only pairs of SNPs were both SNPs are on the middle locus are considered.
</p>
</dd>
<dt>mid_far</dt>
<dd>
<p>Same as <code>mid_near</code>, but for pairs of SNPs that are
more that 10 percent of the locus" length apart. </p>
</dd>
<dt>outer</dt>
<dd>
<p>Only when using locus trios. The statistic for pairs
where both SNPs are on the same outer locus.</p>
</dd>
<dt>between</dt>
<dd>
<p>Only when using locus trios. The statistic for pairs
where one SNPs is on the middle locus, and the other is on an outer one.</p>
</dd>
<dt>mid</dt>
<dd>
<p>The value for all pairs on the
middle locus or all pairs when not using trios.</p>
</dd>
<dt>perc_polym</dt>
<dd>
<p>The percentage of positions that are polymorpic.</p>
</dd>
</dl>
<h3>Unphased Data</h3>

<p>For unphased data, the four gamete condition is only counted as violated if
it is violated for all possible phasing of the data.
</p>


<h3>See Also</h3>

<p>To create a demographic model: <code>coal_model</code>
</p>
<p>To calculate this statistic from data: <code>calc_sumstats_from_data</code>
</p>
<p>Other summary statistics: 
<code>sumstat_dna()</code>,
<code>sumstat_file()</code>,
<code>sumstat_ihh()</code>,
<code>sumstat_jsfs()</code>,
<code>sumstat_mcmf()</code>,
<code>sumstat_nucleotide_div()</code>,
<code>sumstat_omega()</code>,
<code>sumstat_seg_sites()</code>,
<code>sumstat_sfs()</code>,
<code>sumstat_tajimas_d()</code>,
<code>sumstat_trees()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">model &lt;- coal_model(5, 2) +
 feat_mutation(50) +
 feat_recombination(10) +
 sumstat_four_gamete()
stats &lt;- simulate(model)
print(stats$four_gamete)
</code></pre>


</div>