<div class="container">

<table style="width: 100%;"><tr>
<td>sumstat_ihh</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Summary Statistic: Integrated Extended Haplotype Homozygosity</h2>

<h3>Description</h3>

<p>This summary statistic calculates a number of values based on
extended haplotype homozygosity (EHH), including iHH, iES
and optionally iHS.
Coala relies on <code>scan_hh</code> from package <span class="pkg">rehh</span> to
calculate this statistic. Please refer
to their documentation for detailed information on the implementation.
Please cite the corresponding publication (see below) if you use the
statistic for a publication.
</p>


<h3>Usage</h3>

<pre><code class="language-R">sumstat_ihh(
  name = "ihh",
  population = 1,
  max_snps = 1000,
  calc_ihs = FALSE,
  transformation = identity
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>name</code></td>
<td>
<p>The name of the summary statistic. When simulating a model,
the value of the statistics are written to an entry of the returned list
with this name. Summary statistic names must be unique in a model.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>population</code></td>
<td>
<p>The population for which the statistic is calculated.
Can also be "all" to calculate it from all populations. Default is population 1.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>max_snps</code></td>
<td>
<p>The maximal number of SNPs per locus that are used for the
calculation. If a locus has more SNPs, only a random subset of them will
be used to increase performance. Set to <code>Inf</code> to use all SNPs.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>calc_ihs</code></td>
<td>
<p>If set to <code>TRUE</code>, additionally standardized iHS is
calculated.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>transformation</code></td>
<td>
<p>An optional function for transforming the results
of the statistic. If specified, the results of the transformation are
returned instead of the original values.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>If <code>calc_ihs = FALSE</code>, a data.frame with values for
iHH and iES is returned. Otherwise, a list of two data frames are
returned, one for IHH and IES values and the other one for IHS values.
</p>
<p>In all <code>data.frames</code> rows are SNPs and the columns present the following
values for each SNP:
</p>

<ul>
<li>
<p>CHR: The SNP's locus
</p>
</li>
<li>
<p>POSITION: The SNP's absolute position on its locus
</p>
</li>
<li>
<p>FREQ_A: The frequency of the ancestral allele
</p>
</li>
<li>
<p>FREQ_D: The frequency of the derived allele
</p>
</li>
<li>
<p>IHH_A: integrated EHH for the ancestral allele
</p>
</li>
<li>
<p>IHH_D: integrated EHH for the derived allele
</p>
</li>
<li>
<p>IES: integrated EHHS
</p>
</li>
<li>
<p>INES: integrated normalized EHHS
</p>
</li>
<li>
<p>IHS: iHS, normalized over all loci.
</p>
</li>
</ul>
<h3>References</h3>


<ul>
<li>
<p>Mathieu Gautier and Renaud Vitalis, rehh: an R package to detect
footprints of selection in genome-wide SNP data from
haplotype structure. Bioinformatics (2012) 28 (8): 1176-1177
first published online March 7, 2012 doi:10.1093/bioinformatics/bts115
</p>
</li>
<li>
<p>Voight et al., A map of recent positive selection in the human
genome. PLoS Biol, 4(3):e72, Mar 2006.
</p>
</li>
</ul>
<h3>Author(s)</h3>

<p>Paul Staab
</p>


<h3>See Also</h3>

<p>To create a demographic model: <code>coal_model</code>
</p>
<p>To calculate this statistic from data: <code>calc_sumstats_from_data</code>
</p>
<p>Other summary statistics: 
<code>sumstat_dna()</code>,
<code>sumstat_file()</code>,
<code>sumstat_four_gamete()</code>,
<code>sumstat_jsfs()</code>,
<code>sumstat_mcmf()</code>,
<code>sumstat_nucleotide_div()</code>,
<code>sumstat_omega()</code>,
<code>sumstat_seg_sites()</code>,
<code>sumstat_sfs()</code>,
<code>sumstat_tajimas_d()</code>,
<code>sumstat_trees()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">  model &lt;- coal_model(20, 1, 1000) +
    feat_mutation(1000) +
    sumstat_ihh()

    stat &lt;- simulate(model)
    print(stat$ihh)
</code></pre>


</div>