<div class="container">

<table style="width: 100%;"><tr>
<td>sumstat_omega</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Summary Statistic: Omega</h2>

<h3>Description</h3>

<p>Calculates the Omega Statistic introduced by
Kim &amp; Nielsen (2004) from the simulated data. The statistic is sensitive for
hard selective sweeps. To calculate
the statistic, coala relies on the command line program
<a href="http://sco.h-its.org/exelixis/web/software/omegaplus/index.html">OmegaPlus</a>,
which needs to be downloaded and compiled manually in order to use the
statistic.
</p>


<h3>Usage</h3>

<pre><code class="language-R">sumstat_omega(
  name = "omega",
  min_win = 100,
  max_win = 1000,
  grid = 1000,
  binary = "automatic",
  transformation = identity
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>name</code></td>
<td>
<p>The name of the summary statistic. When simulating a model,
the value of the statistics are written to an entry of the returned list
with this name. Summary statistic names must be unique in a model.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min_win</code></td>
<td>
<p>The minimum distance from the grid point that a SNP must have
to be included in the calculation of omega.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>max_win</code></td>
<td>
<p>The maximum distance from the grid point that a SNP must have
to be included in the calculation of omega.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>grid</code></td>
<td>
<p>The number of points for which omega is calculated on each
locus. Should be significantly lower than the locus length.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>binary</code></td>
<td>
<p>The path of the binary for OmegaPlus. If set to "automatic",
coala will try to find a binary called "OmegaPlus" using the PATH
environment variable.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>transformation</code></td>
<td>
<p>An optional function for transforming the results
of the statistic. If specified, the results of the transformation are
returned instead of the original values.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A data frame listing of locus, genetic position and the
calculated omega value.
</p>


<h3>References</h3>

<p>Linkage disequilibrium as a signature of selective sweeps.
Y. Kim and R. Nielsen (2004). Genetics, 167, 1513-1524.
</p>
<p>OmegaPlus: a scalable tool for rapid detection of selective
sweeps in whole-genome datasets.
N. Alachiotis, A. Stamatakis and P. Pavlidis (2012).
Bioinformatics Vol. 28 no. 17 2012, pages 2274-2275
doi:10.1093/bioinformatics/bts419
</p>


<h3>See Also</h3>

<p>To create a demographic model: <code>coal_model</code>
</p>
<p>To calculate this statistic from data: <code>calc_sumstats_from_data</code>
</p>
<p>Other summary statistics: 
<code>sumstat_dna()</code>,
<code>sumstat_file()</code>,
<code>sumstat_four_gamete()</code>,
<code>sumstat_ihh()</code>,
<code>sumstat_jsfs()</code>,
<code>sumstat_mcmf()</code>,
<code>sumstat_nucleotide_div()</code>,
<code>sumstat_seg_sites()</code>,
<code>sumstat_sfs()</code>,
<code>sumstat_tajimas_d()</code>,
<code>sumstat_trees()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
model &lt;- coal_model(20, 1, 50000) +
  feat_recombination(50) +
  feat_mutation(1000) +
  feat_selection(strength_A = 1000, time = 0.03) +
  sumstat_omega()
stats &lt;- simulate(model)
plot(stats$omega$omega, type = "l")
## End(Not run)
</code></pre>


</div>