<div class="container">

<table style="width: 100%;"><tr>
<td>as.segsites.GENOME</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Convert PopGenome Data into Coala's Format</h2>

<h3>Description</h3>

<p>Using this function, you can convert genetic data imported with the package
<span class="pkg">PopGenome</span> into coala's segsites format. See <code>as.segsites</code>
for general information on converting genetic data for coala.
</p>


<h3>Usage</h3>

<pre><code class="language-R">## S3 method for class 'GENOME'
as.segsites(data, only_synonymous = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>data</code></td>
<td>
<p>The <code>GENOME</code> data from <span class="pkg">PopGenome</span>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>only_synonymous</code></td>
<td>
<p>Only import synonymous SNPs if set to <code>TRUE</code>.
This requires that <span class="pkg">PopGenome</span> knows where coding regions are., e.g.
by using gff files.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Ignored.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function imports all loci from the <code>GENOME</code> object that have at
least one valid site (<code>data@n.valid.sites</code>). The number of valid sites
is used as length of a locus.
</p>


<h3>See Also</h3>

<p>An example and additional instructions are provided in the
<code>coala-data-import</code> vignette
</p>


</div>