<div class="container">

<table style="width: 100%;"><tr>
<td>feat_ignore_singletons</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Feature: Ignore Singletons</h2>

<h3>Description</h3>

<p>Mutations that are observed in just one haplotype ('singletons') are often
regarded as likely candidates for sequencing errors. Sometimes, it can be
advantageous to exclude them from an analysis. This feature removes all
singletons from the simulated data before the summary statistics are
calculated.
</p>


<h3>Usage</h3>

<pre><code class="language-R">feat_ignore_singletons(locus_group = "all")
</code></pre>


<h3>Arguments</h3>

<table><tr style="vertical-align: top;">
<td><code>locus_group</code></td>
<td>
<p>The loci for which this features is used. Can either be
<code>"all"</code> (default), in which case the feature is used for simulating
all loci, or a numeric vector. In the latter case, the feature is only
used for the loci added in <code>locus_</code> commands  with the corresponding
index starting from 1 in order in which the commands where added to the
model. For example, if a model has
<code>locus_single(10) + locus_averaged(10, 11) + locus_single(12)</code> and
this argument is <code>c(2, 3)</code>, than the feature is used for all but
the first locus (that is locus 2 - 12).</p>
</td>
</tr></table>
<h3>Details</h3>

<p>This function assumes that a singleton is a mutation for which the derived
allele is observed exactly once in all sequences, regardless of the
population structure.
</p>


<h3>Value</h3>

<p>The feature, which can be added to a model created with
<code>coal_model</code> using <code>+</code>.
</p>


<h3>See Also</h3>

<p>For creating a model: <code>coal_model</code>
</p>
<p>Other features: 
<code>feat_growth()</code>,
<code>feat_migration()</code>,
<code>feat_mutation()</code>,
<code>feat_outgroup()</code>,
<code>feat_pop_merge()</code>,
<code>feat_recombination()</code>,
<code>feat_selection()</code>,
<code>feat_size_change()</code>,
<code>feat_unphased()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">model &lt;- coal_model(2, 1) +
  feat_mutation(10) +
  feat_ignore_singletons() +
  sumstat_sfs("n_mut", transformation = sum)
# In this model, all mutations are singletons. Therefore,
# the number of mutations is 0 when removing singletons:
simulate(model)$n_mut
</code></pre>


</div>