<div class="container">

<table style="width: 100%;"><tr>
<td>genomicEBV.w.codata</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Performs genomic prediction based on SNP codata. 
</h2>

<h3>Description</h3>

<p>The main function of the package. The input includes response values, a design matrix for the fixed effects, a matrix with SNP genotype data and a design matrix for the SNP codata.
</p>


<h3>Usage</h3>

<pre><code class="language-R">genomicEBV.w.codata(y, X, Z, X.SNPcodata, Z.test = NULL, max.iter = 100, conv.crit = 1e-5)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>y</code></td>
<td>

<p>Response values
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>X</code></td>
<td>

<p>Design matrix for the fixed effects
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Z</code></td>
<td>

<p>Genotype matrix with element values of 0, 1 or 2
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>X.SNPcodata</code></td>
<td>

<p>Design matrix for the linear predictor of the SNP variances.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Z.test</code></td>
<td>

<p>An optional genotype matrix for a test data set.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>max.iter</code></td>
<td>

<p>The maximum number of iterations
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>conv.crit</code></td>
<td>

<p>The value of the convergence criterion.
</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>By specifying the matrix <code>Z.test</code> in the input, the function computes predicted genomic breeding values for an out-of-sample data set.
</p>


<h3>Value</h3>

<table>
<tr style="vertical-align: top;">
<td><code>gEBV</code></td>
<td>
<p>Genomic breeding values</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>predicted.gEBV</code></td>
<td>
<p>Genomic breeding values based on the genotypes in <code>Z.test</code> </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>w</code></td>
<td>
<p>Computed SNP weights</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>u</code></td>
<td>
<p>Fitted SNP effects</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>beta</code></td>
<td>
<p>Fitted fixed effects</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>disp.beta</code></td>
<td>
<p>Fitted coefficients in the linear predictor for the SNP variance model</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Converge</code></td>
<td>
<p>Shows whether the algorithm has converged or not</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>iter</code></td>
<td>
<p>The number of iterations used</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>Lars Ronnegard
</p>


<h3>Examples</h3>

<pre><code class="language-R">#######
#Simulation part
set.seed(1234)
N &lt;- 200 #Number of individuals
k &lt;- 300 #Number of SNPs with all marker positions including a QTL
Z1 &lt;- matrix(0, N, k )
Z2 &lt;- matrix(0, N, k )
Z1[1:N, 1] &lt;- rbinom(N, 1, 0.5) #Simulated phased SNP matrices
Z2[1:N, 1] &lt;- rbinom(N, 1, 0.5) 
LD.par &lt;- 0.2 #A parameter to simulate LD. 0 gives full LD, and 0.5 no LD
for (j in 2:k) {
  Z1[1:N, j] &lt;- abs( Z1[1:N, j-1] - rbinom(N, 1, LD.par) )
  Z2[1:N, j] &lt;- abs( Z2[1:N, j-1] - rbinom(N, 1, LD.par) )
} 
Z &lt;- Z1 + Z2 #Genotypic SNP matrix
x1 &lt;- c(rep(1,k/2), rep(0,k/2)) #An indicator for the SNPs. 
#The first k/2 SNPs and the last k/2 have different variances
#Simulate linear predictor for the random effect variance
lin.pred &lt;- 0 + 2*x1 
X.snp &lt;- model.matrix( ~ x1 ) #Corresponding design matrix
u &lt;- rnorm(k, 0 , sqrt( exp(lin.pred) )) 
#Took the square root here because it is the SD that is specified.
#and exp() because we are modelling a log link.
u.scaled &lt;- u/as.numeric( sqrt( var( crossprod(t(Z), u) )) ) 
#Scaled by the variance of the breeding values
e &lt;- rnorm(N) #A residual variance
mu &lt;- 0
y &lt;- mu + crossprod(t(Z),u.scaled) + e
######
#Estimation part
mod1 &lt;- genomicEBV.w.codata(y = as.numeric(y), 
          X = matrix(1, N, 1), Z = Z, X.SNPcodata = X.snp)
#To fit gBLUP just specify X.SNPcodata = matrix(1, k, 1)
cat("Correlation between true and estimated BV for the codata model:")
cat(cor(crossprod(t(Z),u.scaled), mod1$gEBV), "\n")
</code></pre>


</div>