<div class="container">

<table style="width: 100%;"><tr>
<td>compute_GL</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Computes genomic relationship matrix
</h2>

<h3>Description</h3>

<p>This function computes the genomic relationship matrix, G, together with its matrix square root, L.
</p>


<h3>Usage</h3>

<pre><code class="language-R">compute_GL(Z, w)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>Z</code></td>
<td>

<p>Scaled matrix with genotype information
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>w</code></td>
<td>

<p>weights
</p>
</td>
</tr>
</table>
<h3>Value</h3>

<table>
<tr style="vertical-align: top;">
<td><code>L</code></td>
<td>
<p>Square root matrix of G </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>svdVec</code></td>
<td>
<p>Vectors in the Single Value Decomposition of G</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>svdD</code></td>
<td>
<p>Diagonal elements in the Single Value Decomposition of G</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>wZt</code></td>
<td>
<p>weights times the transpose of Z</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>Lars Ronnegard
</p>


<h3>Examples</h3>

<pre><code class="language-R">set.seed(1234)
N &lt;- 20 #Number of individuals
k &lt;- 30 #Number of SNPs with all marker positions including a QTL
Z1 &lt;- matrix(0, N, k )
Z2 &lt;- matrix(0, N, k )
Z1[1:N, 1] &lt;- rbinom(N, 1, 0.5) #Simulated phased SNP matrices
Z2[1:N, 1] &lt;- rbinom(N, 1, 0.5) 
LD.par &lt;- 0.2 #A parameter to simulate LD. 0 gives full LD, and 0.5 no LD
for (j in 2:k) {
  Z1[1:N, j] &lt;- abs( Z1[1:N, j-1] - rbinom(N, 1, LD.par) )
  Z2[1:N, j] &lt;- abs( Z2[1:N, j-1] - rbinom(N, 1, LD.par) )
} 
Z &lt;- Z1 + Z2 #Genotypic SNP matrix
sim.res &lt;- compute_GL(Z, w = rep(1,k))
</code></pre>


</div>