<div class="container">

<table style="width: 100%;"><tr>
<td>Transform</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Transforms hat values
</h2>

<h3>Description</h3>

<p>The function calls the hat.transf function.
</p>


<h3>Usage</h3>

<pre><code class="language-R">Transform(X, L, var.e, var.u, v, svdVec, svdD, wZt, w)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>X</code></td>
<td>

<p>Design matrix for the fixed effects
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>L</code></td>
<td>

<p>Square root matrix of the genomic relationship matrix, G
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>var.e</code></td>
<td>

<p>Residual variance
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>var.u</code></td>
<td>

<p>Genetic variance
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>v</code></td>
<td>

<p>Random effects
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>svdVec</code></td>
<td>

<p>Vector from the Single Value Decomposition of G
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>svdD</code></td>
<td>

<p>Diagonal elements of the Single Value Decomposition of G
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>wZt</code></td>
<td>

<p>Weights times the transpose of the scaled genotype matrix
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>w</code></td>
<td>

<p>Fitted SNP weights
</p>
</td>
</tr>
</table>
<h3>Value</h3>

<table>
<tr style="vertical-align: top;">
<td><code>u</code></td>
<td>
<p>SNP effects</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>qu</code></td>
<td>
<p>Hat values for the SNP effects</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>Lars Ronnegard
</p>


</div>