<div class="container">

<table style="width: 100%;"><tr>
<td>scaleZ</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Scales the genotype matrix.
</h2>

<h3>Description</h3>

<p>Scales the genotype matrix so that ZZ' gives the genomic relationship matrix.
</p>


<h3>Usage</h3>

<pre><code class="language-R">scaleZ(Z, freq1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>Z</code></td>
<td>

<p>Genotype matrix with element values 0, 1 and 2
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>freq1</code></td>
<td>

<p>Optional input parameter with allele frequencies. A vector of length equal to the number of columns in Z.
</p>
</td>
</tr>
</table>
<h3>Value</h3>

<table><tr style="vertical-align: top;">
<td><code>Z</code></td>
<td>
<p>Scaled genotype matrix</p>
</td>
</tr></table>
<h3>Author(s)</h3>

<p>Lars Ronnegard
</p>


<h3>Examples</h3>

<pre><code class="language-R">#######
#Simulation part
set.seed(1234)
N &lt;- 200 #Number of individuals
k &lt;- 300 #Number of SNPs with all marker positions including a QTL
Z1 &lt;- matrix(0, N, k )
Z2 &lt;- matrix(0, N, k )
Z1[1:N, 1] &lt;- rbinom(N, 1, 0.5) #Simulated phased SNP matrices
Z2[1:N, 1] &lt;- rbinom(N, 1, 0.5) 
LD.par &lt;- 0.2 #A parameter to simulate LD. 0 gives full LD, and 0.5 no LD
for (j in 2:k) {
  Z1[1:N, j] &lt;- abs( Z1[1:N, j-1] - rbinom(N, 1, LD.par) )
  Z2[1:N, j] &lt;- abs( Z2[1:N, j-1] - rbinom(N, 1, LD.par) )
} 
Z &lt;- Z1 + Z2 #Genotypic SNP matrix
sim.res &lt;- scaleZ(Z)
</code></pre>


</div>