<div class="container">

<table style="width: 100%;"><tr>
<td>check_dataset</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>check_dataset</h2>

<h3>Description</h3>

<p>Check coloc dataset inputs for errors
</p>


<h3>Usage</h3>

<pre><code class="language-R">check_dataset(d, suffix = "", req = c("type", "snp"), warn.minp = 1e-06)

check.dataset(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>d</code></td>
<td>
<p>dataset to check</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>suffix</code></td>
<td>
<p>string to identify which dataset (1 or 2)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>req</code></td>
<td>
<p>names of elements that must be present</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>warn.minp</code></td>
<td>
<p>print warning if no p value &lt; warn.minp</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>arguments passed to check_dataset()</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>A coloc dataset is a list, containing a mixture of vectors
capturing quantities that vary between snps (these vectors must
all have equal length) and scalars capturing quantities that
describe the dataset.
</p>
<p>Coloc is flexible, requiring perhaps only p values, or z scores, or effect
estimates and standard errors, but with this flexibility, also comes
difficulties describing exactly the combinations of items required.
</p>
<p>Required vectors are some subset of
</p>

<dl>
<dt>beta</dt>
<dd>
<p>regression coefficient for each SNP from dataset 1</p>
</dd>
<dt>varbeta</dt>
<dd>
<p>variance of beta</p>
</dd>
<dt>pvalues</dt>
<dd>
<p>P-values for each SNP in dataset 1</p>
</dd>
<dt>MAF</dt>
<dd>
<p>minor allele frequency of the variants</p>
</dd>
<dt>snp</dt>
<dd>
<p>a character vector of snp ids, optional. It will be used to merge dataset1 and dataset2 and will be retained in the results.</p>
</dd>
</dl>
<p>Preferably, give <code>beta</code> and <code>varbeta</code>.  But if these are not available, sufficient statistics can be approximated from <code>pvalues</code> and <code>MAF</code>.
</p>
<p>Required scalars are some subset of
</p>

<dl>
<dt>N</dt>
<dd>
<p>Number of samples in dataset 1</p>
</dd>
<dt>type</dt>
<dd>
<p>the type of data in dataset 1 - either "quant" or "cc" to denote quantitative or case-control</p>
</dd>
<dt>s</dt>
<dd>
<p>for a case control dataset, the proportion of samples in dataset 1 that are cases</p>
</dd>
<dt>sdY</dt>
<dd>
<p>for a quantitative trait, the population standard deviation of the trait.  if not given, it can be estimated from the vectors of varbeta and MAF</p>
</dd>
</dl>
<p>You must always give <code>type</code>.  Then,
</p>

<dl>
<dt>if <code>type</code>=="cc"</dt>
<dd>
<p><code>s</code></p>
</dd>
<dt>if <code>type</code>=="quant" and <code>sdY</code> known</dt>
<dd>
<p><code>sdY</code></p>
</dd>
<dt>if beta, varbeta not known</dt>
<dd>
<p><code>N</code></p>
</dd>
</dl>
<p>If <code>sdY</code> is unknown, it will be approximated, and this will require
</p>

<dl>
<dt>summary data to estimate <code>sdY</code>
</dt>
<dd>
<p><code>beta</code>, <code>varbeta</code>, <code>N</code>, <code>MAF</code></p>
</dd>
</dl>
<p>Optional vectors are
</p>

<dl>
<dt>position</dt>
<dd>
<p>a vector of snp positions, required for <code>plot_dataset</code></p>
</dd>
</dl>
<p><code>check_dataset</code> calls stop() unless a series of expectations on dataset
input format are met
</p>
<p>This is a helper function for use by other coloc functions, but
you can use it directly to check the format of a dataset to be
supplied to coloc.abf(), coloc.signals(), finemap.abf(), or
finemap.signals().
</p>


<h3>Value</h3>

<p>NULL if no errors found
</p>


<h3>Author(s)</h3>

<p>Chris Wallace
</p>


</div>