<div class="container">

<table style="width: 100%;"><tr>
<td>coloc.detail</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Bayesian colocalisation analysis with detailed output</h2>

<h3>Description</h3>

<p>Bayesian colocalisation analysis, detailed output
</p>


<h3>Usage</h3>

<pre><code class="language-R">coloc.detail(
  dataset1,
  dataset2,
  MAF = NULL,
  p1 = 1e-04,
  p2 = 1e-04,
  p12 = 1e-05
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>dataset1</code></td>
<td>
<p>a list with specifically named elements defining the dataset
to be analysed. See <code>check_dataset</code> for details.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dataset2</code></td>
<td>
<p>as above, for dataset 2</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>MAF</code></td>
<td>
<p>Common minor allele frequency vector to be used for both dataset1 and dataset2, a shorthand for supplying the same vector as parts of both datasets</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p1</code></td>
<td>
<p>prior probability a SNP is associated with trait 1, default 1e-4</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p2</code></td>
<td>
<p>prior probability a SNP is associated with trait 2, default 1e-4</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p12</code></td>
<td>
<p>prior probability a SNP is associated with both traits, default 1e-5</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function replicates coloc.abf, but outputs more detail for
further processing using coloc.process
</p>
<p>Intended to be called internally by coloc.signals
</p>


<h3>Value</h3>

<p>a list of three <code>data.tables</code>s:
</p>

<ul>
<li>
<p> summary is a vector giving the number of SNPs analysed, and the posterior probabilities of H0 (no causal variant), H1 (causal variant for trait 1 only), H2 (causal variant for trait 2 only), H3 (two distinct causal variants) and H4 (one common causal variant)
</p>
</li>
<li>
<p> df is an annotated version of the input data containing log Approximate Bayes Factors and intermediate calculations, and the posterior probability SNP.PP.H4 of the SNP being causal for the shared signal
</p>
</li>
<li>
<p> df3 is the same for all 2 SNP H3 models
</p>
</li>
</ul>
<h3>Author(s)</h3>

<p>Chris Wallace
</p>


<h3>See Also</h3>

<p><code>coloc.process</code>, <code>coloc.abf</code>
</p>


</div>