<div class="container">

<table style="width: 100%;"><tr>
<td>coloc.abf</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Fully Bayesian colocalisation analysis using Bayes Factors</h2>

<h3>Description</h3>

<p>Bayesian colocalisation analysis
</p>


<h3>Usage</h3>

<pre><code class="language-R">coloc.abf(dataset1, dataset2, MAF = NULL, p1 = 1e-04, p2 = 1e-04, p12 = 1e-05)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>dataset1</code></td>
<td>
<p>a list with specifically named elements defining the dataset
to be analysed. See <code>check_dataset</code> for details.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dataset2</code></td>
<td>
<p>as above, for dataset 2</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>MAF</code></td>
<td>
<p>Common minor allele frequency vector to be used for both dataset1 and dataset2, a shorthand for supplying the same vector as parts of both datasets</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p1</code></td>
<td>
<p>prior probability a SNP is associated with trait 1, default 1e-4</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p2</code></td>
<td>
<p>prior probability a SNP is associated with trait 2, default 1e-4</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p12</code></td>
<td>
<p>prior probability a SNP is associated with both traits, default 1e-5</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function calculates posterior probabilities of different
causal variant configurations under the assumption of a single
causal variant for each trait.
</p>
<p>If regression coefficients and variances are available, it
calculates Bayes factors for association at each SNP.  If only p
values are available, it uses an approximation that depends on the
SNP's MAF and ignores any uncertainty in imputation.  Regression
coefficients should be used if available.
</p>


<h3>Value</h3>

<p>a list of two <code>data.frame</code>s:
</p>

<ul>
<li>
<p> summary is a vector giving the number of SNPs analysed, and the posterior probabilities of H0 (no causal variant), H1 (causal variant for trait 1 only), H2 (causal variant for trait 2 only), H3 (two distinct causal variants) and H4 (one common causal variant)
</p>
</li>
<li>
<p> results is an annotated version of the input data containing log Approximate Bayes Factors and intermediate calculations, and the posterior probability SNP.PP.H4 of the SNP being causal for the shared signal <em>if</em> H4 is true. This is only relevant if the posterior support for H4 in summary is convincing.
</p>
</li>
</ul>
<h3>Author(s)</h3>

<p>Claudia Giambartolomei, Chris Wallace
</p>


</div>