<div class="container">

<table style="width: 100%;"><tr>
<td>COMBAT</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
A Combined Gene-based Association Test
</h2>

<h3>Description</h3>

<p>This function implements a combined gene-based association test using SNP-level P values and reference genotype data.
</p>


<h3>Usage</h3>

<pre><code class="language-R">COMBAT(x, snp.ref, vegas.pct = c(0.1,0.2,0.3,0.4,1), pca_cut_perc = 0.995,
   nperm = 100, seed=12345, ncores=1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>a vector of SNP-level P values.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>snp.ref</code></td>
<td>
<p>a matrix of SNP genotypes (coded as allele counts) from reference samples, with samples in rows and SNPs in columns.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>vegas.pct</code></td>
<td>
<p>a numeric vector, fraction of the top SNPs to be used in the VEGAS method.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pca_cut_perc</code></td>
<td>
<p>numeric, cutoff for percentage of sum of eigen values in the simpleM approach.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nperm</code></td>
<td>
<p>number of permutations for computing the correlation between P values of different tests.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>seed</code></td>
<td>
<p>random seed to derive consistent outcome.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ncores</code></td>
<td>
<p>number of CPU cores for parallel computing.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>COMBAT uses simulation and the extended Simes procedure (<code>ext_simes</code>) to combine multiple gene-based association test statistics (currently including <code>gates</code>, <code>vegas</code>, and <code>simpleM</code>) to perform a more powerful association analysis. This method does not require raw genotype or phenotype data, but needs only SNP-level P-values and correlations between SNPs from ancestry-matched samples. The technical details about the method is described in Wang et al (2017) &lt;doi:10.1534/genetics.117.300257&gt;.
</p>


<h3>Value</h3>

<p>A vector of p-values from COMAT and each individual gene-based test.
</p>


<h3>References</h3>

<p>Minghui Wang, Jianfei Huang, Yiyuan Liu, Li Ma, James B. Potash, Shizhong Han. COMBAT: A Combined Association Test for Genes using Summary Statistics. Genetics 2017, 207(3): 883-891. https://doi.org/10.1534/genetics.117.300257.
</p>


<h3>See Also</h3>

<p><code>ext_simes</code>, <code>gates</code>, <code>vegas</code>, <code>simpleM</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R"># read SNP P values
file1 &lt;- paste(path.package("COMBAT"),"extdata","SNP_info.txt.gz",sep="/")
snp.info  &lt;- read.table(file1, header = TRUE, as.is=TRUE)
snp.pvals &lt;- as.matrix(snp.info[,2])

# read reference genotype
file2 &lt;- paste(path.package("COMBAT"),"extdata","SNP_ref.txt.gz",sep="/")
snp.ref   &lt;- read.table(file2, header = TRUE)
snp.ref   &lt;- as.matrix(snp.ref)
#call COMBAT

COMBAT(snp.pvals, snp.ref, nperm=100, ncores=2)


</code></pre>


</div>