<div class="container">

<table style="width: 100%;"><tr>
<td>gates</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Calling Gene-based Association Tests
</h2>

<h3>Description</h3>

<p>Several gene-based association tests methods are implemented.
</p>


<h3>Usage</h3>

<pre><code class="language-R">gates(x, cor_G)
vegas(x, cor_G, vegas.pct=c(0.1,0.2,0.3,0.4,1), max.simulation=1e6)
simpleM(x, cor_G, pca_cut_perc=0.995)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>a vector of SNP-level P values.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cor_G</code></td>
<td>
<p>SNP-SNP correlation matrix.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>vegas.pct</code></td>
<td>
<p>a numeric vector, specifying the fraction of the top SNPs to be used in the VEGAS method.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>max.simulation</code></td>
<td>
<p>maximum number of simulations to be performed. Must be at least 1e6.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pca_cut_perc</code></td>
<td>
<p>cutoff for percentage of sum of eigen values.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Function <code>gates</code> implements the GATES method (Li et al 2011, American Journal of Human Genetics 88:283-293), <code>vegas</code> implements VEGAS with different proportion tests (Liu et al 2010, American Journal of Human Genetics 87:139-145), and <code>simpleM</code> is the simpleM method (Gao et al 2008, Genetic Epidemiology 32:361-369).
</p>


<h3>Value</h3>

<p>P value(s).
</p>


<h3>See Also</h3>

<p><code>COMBAT</code>, <code>ld.Rsquare</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R"># read SNP P values
file1 &lt;- paste(path.package("COMBAT"),"extdata","SNP_info.txt.gz",sep="/")
snp.info  &lt;- read.table(file1, header = TRUE, as.is=TRUE)
snp.pvals &lt;- as.matrix(snp.info[,2])

# read reference genotype
file2 &lt;- paste(path.package("COMBAT"),"extdata","SNP_ref.txt.gz",sep="/")
snp.ref   &lt;- read.table(file2, header = TRUE)
snp.ref   &lt;- as.matrix(snp.ref)

#compute correlation among SNPs
cor_G &lt;- ld.Rsquare(snp.ref)

#call gates
(pval_gates &lt;- gates(x=snp.pvals, cor_G=cor_G))

#call vegas
(pval_vegas &lt;- vegas(x=snp.pvals, cor_G=cor_G))

#call simpleM
(pval_simpleM &lt;- simpleM(x=snp.pvals, cor_G=cor_G))
</code></pre>


</div>