<div class="container">

<table style="width: 100%;"><tr>
<td>seurat_to_combiroc</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Load CombiROC data.</h2>

<h3>Description</h3>

<p>A function to extract a combiroc data (both labelled and unlabelled) from a SeuratObject.
</p>


<h3>Usage</h3>

<pre><code class="language-R">seurat_to_combiroc(
  SeuratObject,
  gene_list,
  assay = "RNA",
  labelled_data = F,
  case_class = NA,
  case_label = "case",
  control_label = "control"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>SeuratObject</code></td>
<td>
<p>Defines S4 classes for single-cell genomic data and associated information.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gene_list</code></td>
<td>
<p>a list of gene names.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>assay</code></td>
<td>
<p>a character that specifies the assay of interest.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>labelled_data</code></td>
<td>
<p>a boolean that specifies whether the combiroc data to be extracted must be labelled (with 'Class' column) or not.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>case_class</code></td>
<td>
<p>a character or a character vector specifying the category/ies to be considered as 'case class'. Required if labelled_data is TRUE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>case_label</code></td>
<td>
<p>a character that will be assigned to the cells belonging to 'case class' category.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>control_label</code></td>
<td>
<p>a character that will be assigned to the cells belonging to 'control class' category.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>By specifying a gene list (if the genes are in rownames of SeuratObject assay matrix, it subsets the gene expression matrix (@data) and it retreives their expression values. If a combiroc training dataset (labelled_data=T) is required, one or more categories of Idents(SeuratObject) must be selected as case_class, the others will be merged into control class.
</p>


<h3>Value</h3>

<p>a combiroc data.
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
demo_seurat# A subset of PBMC3K dataset from Satijia et al. 2015

# list of markers of interest
gene_list&lt;-c('RBP7','CYP1B1','CD14','FCN1','NKG7', 'GNLY')

# to extract the combircoc data from data from pbmc3k.final (case class: Monocytes)
data &lt;- seurat_to_combiroc(demo_seurat, gene_list = gene_list, labelled_data = TRUE,
                           case_class = c('CD14+ Mono','FCGR3A+ Mono'),
                           case_label = 'Monocyte', control_label='Other')
head(data)

## End(Not run)
</code></pre>


</div>