<div class="container">

<table style="width: 100%;"><tr>
<td>vocal</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Variation in Cell Abundance Loci</h2>

<h3>Description</h3>

<p>Probing immune system genetics via gene expression.
VoCAL is a deconvolution-based method that utilizes transcriptome data
to infer the quantities of immune-cell types, and then uses these quantitative
traits to uncover the underlying DNA loci (iQTLs) assuming homozygosity
(such as in the case of recombinent inbred strains).
</p>


<h3>Usage</h3>

<pre><code class="language-R">vocal(...,reference_data,expression_data,genotyping_data,normalize_data,
 T.i=5,T.e=10,eqtl_association_scores=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>one or more data frames of one column, each one represents a
preselected marker set that likely discriminate well between the immune-cell
types given in the reference data. The number of data frames defines the
number of association scores that would be combined to generate the final
iQTL association score.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>reference_data</code></td>
<td>
<p>a data frame representing immune cell expression profiles.
Each row represents an expression of a gene, and each column represents a
different immune cell type. <code>colnames</code> contains the name of each immune cell
type and the <code>rownames</code> includes the genes' symbol. The names of each immune
cell type and the symbol of each gene should be unique. Any gene with
missing expression values must be excluded.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>expression_data</code></td>
<td>
<p>a data frame representing RNA-seq or microarray
gene-expression profiles of a given complex tissue across a population of
genetically distinct (genotyped) individuals. Each row represents an
expression of a gene, and each column represents a genetically distinct
individual. <code>colnames</code> contain the name of each individual, as written in the
<code>genotyping_data</code>, and <code>rownames</code> includes the genes' symbol.
The name of each individual sample and the symbol of each gene should be unique.
Any gene with missing expression values should be excluded.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genotyping_data</code></td>
<td>
<p>a data frame where each row represents a different
locus, and each column represents a genetically distinct individual.
The genotype should be taken from homozygous individuals only.
Where the genotype is unknown <code>NA</code> should be used.
The first six columns contain the following information: (1) The sequential
identifier of the locus; (2) The name of each locus Chr; (3) Chromosome
position; (4) Start genome position; (5) End genome position;
(6) position in cM.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>normalize_data</code></td>
<td>
<p>normalization type. The data will be normalized by either:
(1) "All" - subtraction of the mean expression of all strains;
(2) "None" - data is already normalized, do nothing;
(3) name of individual included in <code>colnames</code> of <code>expression_data</code>;</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>T.i</code></td>
<td>
<p>numerical. significant iQTL association score <code>(-log10(Pvalue))</code>
cutoff for the refinement step of the VoCAL algorithm.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>T.e</code></td>
<td>
<p>numerical. significant eQTL association score <code>(-log10(Pvalue))</code>
cutoff for the refinement step of the VoCAL algorithm.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>eqtl_association_scores</code></td>
<td>
<p>(optional) a data frame where each entry
represents an association score for a gene given the genotype of all the
individuals that appear in the expression_data data frame, in a specific locus.
This eQTL analysis should be peformed over the normalized expression_data.
<code>colnames</code> contain the UID (as written in the genotyping_data) and
<code>rownames</code> includes the genes' symbol (as written in the expression_data).
The symbol of each gene should be unique. These scores should be in -log10(P value).
Default is NULL, meaning that eQTL analysis will be performed.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>a list of two martices
</p>
<table>
<tr style="vertical-align: top;">
<td><code>final_association_score</code></td>
<td>
<p>a matrix that contains the output iQTL association
score after applying the iterative filteration procedure. Each row represents the genome
wide-association result for a specific immune trait over a range of DNA loci.
<code>rownames</code> provides the identifier of the locus and <code>colnames</code> contains the
immune-cell type names. Each entry provides the <code>-log10(P value)</code> of an iQTL
association score.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>marker_info</code></td>
<td>
<p>the names of all the markers removed from the different
marker sets provided</p>
</td>
</tr>
</table>
<h3>References</h3>

<p>Steuerman Y and Gat-Viks I.
Exploiting Gene-Expression Deconvolution to Probe the Genetics of the Immune System (2015), Submitted.
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(commons)
data(vocalEx)
## Not run: 
 results &lt;- vocal(DCQ_mar, reference_data=immgen_dat, expression_data=lung_dat,
 genotyping_data=gBXD, normalize_data="B6", eqtl_association_scores=eQTL_res)

## End(Not run)

</code></pre>


</div>