<div class="container">

<table style="width: 100%;"><tr>
<td>dcq</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>DCQ - Digital Cell Quantifier</h2>

<h3>Description</h3>

<p>DCQ combines genome-wide gene expression data with an immune cell-type
reference data to infer changes in the quantities immune cell subpopulations.
</p>


<h3>Usage</h3>

<pre><code class="language-R">dcq(reference_data, mix_data, marker_set, alpha_used=0.05,
 lambda_min=0.2, number_of_repeats=3, precent_of_data=1.0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>reference_data</code></td>
<td>
<p>a data frame representing immune cell expression profiles.
Each row represents an expression of a gene, and each column represents a
different immune cell type. <code>colnames</code> contains the name of each immune cell
type and the <code>rownames</code> includes the genes' symbol. The names of each immune
cell type and the symbol of each gene should be unique. Any gene with
missing expression values must be excluded.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mix_data</code></td>
<td>
<p>a data frame representing RNA-seq or microarray
gene-expression profiles of a given complex tissue. Each row represents an
expression of a gene, and each column represents a different experimental sample.
<code>colnames</code> contain the name of each sample and <code>rownames</code> includes the genes' symbol.
The name of each individual sample and the symbol of each gene should be unique.
Any gene with missing expression values should be excluded.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>marker_set</code></td>
<td>
<p>data frames of one column, that includes a
preselected list of genes that likely discriminate well
between the immune-cell types given in the reference data.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alpha_used, lambda_min</code></td>
<td>
<p>parameters of the L1 and L2 regularization. It is generally
recommended to leave the default value. For more information about this
parameter, see the glmnet package.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>number_of_repeats</code></td>
<td>
<p>using one repeat will generate only one output model.
Using many repeats, DCQ calculates a collection of models, and outputs the
average and standard deviation for each predicted relative cell quantity.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>precent_of_data</code></td>
<td>
<p>in order to run the analysis over all the cell types use 1.0.
For bootstrap purposes, you can use part of the data (e.g, 0.5).</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>a list that contains two matrices
</p>
<table>
<tr style="vertical-align: top;">
<td><code>average</code></td>
<td>
<p>a matrix that contains the average relative quantities for each cell
type in everytest sample.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>stdev</code></td>
<td>
<p>a matrix that contains the standard deviations over all repeats for
each cell types in each test sample.</p>
</td>
</tr>
</table>
<h3>References</h3>

<p>Altboum Z, Steuerman Y, David E, Barnett-Itzhaki Z, Valadarsky L, Keren-Shaul H, et al.
Digital cell quantification identifies global immune cell dynamics during influenza
infection. Mol Syst Biol. 2014;10: 720. doi:10.1002/msb.134947
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(commons)
data(dcqEx)
results &lt;- dcq(reference_data=immgen_dat, mix_data=lung_time_series_dat, marker_set=DCQ_mar)

</code></pre>


</div>