<div class="container">

<table style="width: 100%;"><tr>
<td>p2app4conos</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Utility function to generate a pagoda2 app from a conos object</h2>

<h3>Description</h3>

<p>Utility function to generate a pagoda2 app from a conos object
</p>


<h3>Usage</h3>

<pre><code class="language-R">p2app4conos(
  conos,
  cdl = NULL,
  metadata = NULL,
  filename = "conos_app.bin",
  save = TRUE,
  n.cores = 1,
  n.odgenes = 3000,
  nPcs = 100,
  k = 30,
  perplexity = 50,
  log.scale = TRUE,
  trim = 10,
  keep.genes = NULL,
  min.cells.per.gene = 0,
  min.transcripts.per.cell = 100,
  get.largevis = TRUE,
  get.tsne = TRUE,
  make.geneknn = TRUE,
  go.env = NULL,
  cell.subset = NULL,
  max.cells = Inf,
  additional.embeddings = NULL,
  test.pathway.overdispersion = FALSE,
  organism = NULL,
  return.details = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>conos</code></td>
<td>
<p>Conos object</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cdl</code></td>
<td>
<p>list Optional list of raw matrices (so that gene merging doesn't have to be redone) (default=NULL)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>metadata</code></td>
<td>
<p>list Optional list of (named) metadata factors (default=NULL)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>filename</code></td>
<td>
<p>string Name of the *.bin file to seralize for the pagoda2 application if save=TRUE (default='conos_app.bin')</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>save</code></td>
<td>
<p>boolean Save serialized *bin file specified in filename (default=TRUE)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.cores</code></td>
<td>
<p>integer Number of cores (default=1)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.odgenes</code></td>
<td>
<p>numeric Number of top overdispersed genes to use (dfault=3e3). From pagoda2::basicP2proc().</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nPcs</code></td>
<td>
<p>numeric Number of PCs to use (default=100). From pagoda2::basicP2proc().</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>k</code></td>
<td>
<p>numeric Default number of neighbors to use in kNN graph (default=30). From pagoda2::basicP2proc().</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>perplexity</code></td>
<td>
<p>numeric Perplexity to use in generating tSNE and largeVis embeddings (default=50). From pagoda2::basicP2proc().</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>log.scale</code></td>
<td>
<p>boolean Whether to use log scale normalization (default=TRUE). From pagoda2::basicP2proc().</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>trim</code></td>
<td>
<p>numeric Number of cells to trim in winsorization (default=10). From pagoda2::basicP2proc().</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>keep.genes</code></td>
<td>
<p>optional set of genes to keep from being filtered out (even at low counts) (default=NULL). From pagoda2::basicP2proc().</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min.cells.per.gene</code></td>
<td>
<p>numeric Minimal number of cells required for gene to be kept (unless listed in keep.genes) (default=0). From pagoda2::basicP2proc().</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min.transcripts.per.cell</code></td>
<td>
<p>numeric Minimumal number of molecules/reads for a cell to be admitted (default=100). From pagoda2::basicP2proc().</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>get.largevis</code></td>
<td>
<p>boolean Whether to caluclate largeVis embedding (default=TRUE). From pagoda2::basicP2proc().</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>get.tsne</code></td>
<td>
<p>boolean Whether to calculate tSNE embedding (default=TRUE). From pagoda2::basicP2proc().</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>make.geneknn</code></td>
<td>
<p>boolean Whether pre-calculate gene kNN (for gene search) (default=TRUE). From pagoda2::basicP2proc().</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>go.env</code></td>
<td>
<p>GO environment for the organism of interest (default=NULL)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cell.subset</code></td>
<td>
<p>string Cells to subset with the conos embedding conos$embedding. If NULL, uses all cells via rownames(conos$embedding) (default=NULL)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>max.cells</code></td>
<td>
<p>numeric Limit to the cells that are included in the conos. If Inf, there is no limit (default=Inf)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>additional.embeddings</code></td>
<td>
<p>list Additional embeddings to add to conos for the pagoda2 app (default=NULL)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>test.pathway.overdispersion</code></td>
<td>
<p>boolean Find all IDs using GO category against either org.Hs.eg.db ('hs') or org.Mm.eg.db ('mm') (default=FALSE</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>organism</code></td>
<td>
<p>string Organism of interest, either 'hs' (Homo sapiens) or 'mm' (Mus musculus, i.e. mouse) (default=NULL). Only used if test.pathway.overdispersion is TRUE. If NULL and test.pathway.overdispersion=TRUE, then 'hs' is used.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>return.details</code></td>
<td>
<p>boolean If TRUE, return list of p2 application, pagoda2 object, list of raw matrices, and cell names. If FALSE, simply return pagoda2 app object. (default=FALSE)</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>pagoda2 app object
</p>


</div>