<div class="container">

<table style="width: 100%;"><tr>
<td>labelstonumbers</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Converts taxon names to tip/edge numbers</h2>

<h3>Description</h3>

<p>Converts taxon names to corresponding tip/edge numbers in the phylogeny.
</p>


<h3>Usage</h3>

<pre><code class="language-R">labelstonumbers(phyl, tips)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>phyl</code></td>
<td>
<p>The phylogeny of interest in phylo format</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tips</code></td>
<td>
<p>The names of the tips in question</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Simply reads in taxon names, determines which tip or edge number they correspond to, and returns those values
</p>


<h3>Value</h3>

<p>The numbers of all of the tips of interest.
</p>


<h3>References</h3>

<p>Paradis, E., J. Claude, and K. Strimmer (2004) APE: Analyses of phylogenetics
and evolution in R langauge. Bioinformatics, 20, 289-290.
Paradis, E. (2012) Analysis of Phylogenetics and Evolution with R (Second Edition). New York: Springer.
</p>


<h3>Examples</h3>

<pre><code class="language-R">
phyl&lt;-rtree(10)
nums&lt;-labelstonumbers(phyl,c("t1","t2","t3"))
</code></pre>


</div>