<div class="container">

<table style="width: 100%;"><tr>
<td>multianc</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Reconstructs ancestral states for multiple characters</h2>

<h3>Description</h3>

<p>Uses fastAnc to reconstruct ancestral states for multiple phenotypic characters
</p>


<h3>Usage</h3>

<pre><code class="language-R">multianc(phy, phendata)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>phy</code></td>
<td>
<p>The phylogeny of interest in phylo format</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>phendata</code></td>
<td>
<p>Phenotypic data for all tips</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>None
</p>


<h3>Value</h3>

<p>A matrix with the tips data in the first n rows and the ancestral data in the remaining n-1 rows.
</p>


<h3>References</h3>

<p>Paradis, E., J. Claude, and K. Strimmer (2004) APE: Analyses of phylogenetics
and evolution in R langauge. Bioinformatics, 20, 289-290.
</p>
<p>Revell, L. J. (2012) phytools: An R package for phylogenetic comparative
biology (and other things). Methods Ecol. Evol. 3 217-223.
</p>


<h3>Examples</h3>

<pre><code class="language-R">
phy&lt;-rtree(10)

phendata&lt;-fastBM(phy,nsim=2)

ancs&lt;-multianc(phy,phendata)
</code></pre>


</div>