<div class="container">

<table style="width: 100%;"><tr>
<td>cophe.single.lbf</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>cophe.single.lbf</h2>

<h3>Description</h3>

<p>Calculate log bayes factors for each hypothesis (Single causal variant assumption)
</p>


<h3>Usage</h3>

<pre><code class="language-R">cophe.single.lbf(dataset, querysnpid, querytrait, MAF = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>dataset</code></td>
<td>
<p>a list with specifically named elements defining the query trait dataset
to be analysed.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>querysnpid</code></td>
<td>
<p>Id of the query variant, (id in dataset$snp)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>querytrait</code></td>
<td>
<p>Query trait name</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>MAF</code></td>
<td>
<p>Minor allele frequency vector</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>data frame with log bayes factors for Hn and Ha hypotheses
</p>


<h3>Author(s)</h3>

<p>Ichcha Manipur
</p>


<h3>See Also</h3>

<p><code>cophe.single</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">library(cophescan)
data(cophe_multi_trait_data)
query_trait_1 &lt;- cophe_multi_trait_data$summ_stat[['Trait_1']]
querysnpid &lt;- cophe_multi_trait_data$querysnpid
res.single.lbf &lt;- cophe.single.lbf(query_trait_1, querysnpid = querysnpid, querytrait='Trait_1')
res.single.lbf
</code></pre>


</div>