<div class="container">

<table style="width: 100%;"><tr>
<td>sep_assem</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>sep_assem</h2>

<h3>Description</h3>

<p>Separates two or more gene copies from short-read Next-Generation Sequencing data into a small number of overlapping DNA sequences and assemble them into their respective gene copies.
</p>


<h3>Usage</h3>

<pre><code class="language-R">sep_assem(
  copy_number,
  read_length,
  overlap = 225,
  rare_read = 10,
  core_number = 1,
  verbose = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>copy_number</code></td>
<td>
<p>An integer (e.g. 2,3, or 4) giving the anticipated number of gene copies in the input file.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>read_length</code></td>
<td>
<p>An integer (e.g. 250, or 300) giving the read length of your Next-generation Sequencing data. This method is designed for read length &gt;=250bp.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>overlap</code></td>
<td>
<p>An integer describing number of base pairs of overlap between adjacent subsets. More overlap means more subsets. Default 225.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rare_read</code></td>
<td>
<p>A positive integer. During clustering analyses, clusters with less than this number of reads will be ignored. Default 10.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>core_number</code></td>
<td>
<p>An integer describing number of cores to use.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>Turn on (verbose=1; default) or turn off (verbose=0) the output.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A fasta alignment of the anticipated number of full-length gene copies.
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
sep_assem(2,300,225,10,1,1) # all input fasta files in the working directory will be processed

## End(Not run)

</code></pre>


</div>