<div class="container">

<table style="width: 100%;"><tr>
<td>subset_downsize</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>subset_downsize</h2>

<h3>Description</h3>

<p>Subdivides the imported read alignment into subsets and then downsizes each subset by deleting those sequences that have too many gaps or missing data.
</p>


<h3>Usage</h3>

<pre><code class="language-R">subset_downsize(filename, read_length, overlap, verbose = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>filename</code></td>
<td>
<p>A fasta file contains thousands of short reads that have been mapped to a reference. The reference and reads that are not directly mapped to the reference need to be removed after mapping.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>read_length</code></td>
<td>
<p>An integer (e.g. 250, or 300) giving the read length of your Next-generation Sequencing data. This method is designed for read length &gt;=250bp.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>overlap</code></td>
<td>
<p>An integer describing number of base pairs of overlap between adjacent subsets. More overlap means more subsets.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>Turn on (verbose=1; default) or turn off (verbose=0) the output.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A number of overlapping subsets (before and after downsizing) of the input alignment.
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
subset_downsize("inst/extdata/toydata.fasta",300,225,1)

## End(Not run)

</code></pre>


</div>