<div class="container">

<table style="width: 100%;"><tr>
<td>get_diff_ratio_net</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Construct differential expression ratio network</h2>

<h3>Description</h3>

<p>Construct the differential expression ratio network for a single sample.
</p>


<h3>Usage</h3>

<pre><code class="language-R">get_diff_ratio_net(
  ref.ratio.dist,
  expr.val,
  log.expr = FALSE,
  scale.degree = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>ref.ratio.dist</code></td>
<td>
<p>The expression ratio distribution profile returned by <code>get_ratio_distribution</code> or <code>get_ratio_distribution2</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>expr.val</code></td>
<td>
<p>Numeric vector of gene expression values in the sample.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>log.expr</code></td>
<td>
<p>Logical variable indicating whether the input expression vector is in logarithmic scale.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>scale.degree</code></td>
<td>
<p>Logical variable indicating whether the degree values are scaled according to the dropout rate.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>This function will return a list with the following components:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>net</code></td>
<td>
<p>The binary adjacent matrix of differential expression ratio network.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>diff</code></td>
<td>
<p>A numeric vector containing the adjusted degree differences of all genes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>degree</code></td>
<td>
<p>A list containing the raw degree differences and sums of all genes.</p>
</td>
</tr>
</table>
</div>