<div class="container">

<table style="width: 100%;"><tr>
<td>simulate_dropout2</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Simulate dropout expression data</h2>

<h3>Description</h3>

<p>Generate the expression data with desired dropout rate range
</p>


<h3>Usage</h3>

<pre><code class="language-R">simulate_dropout2(counts, min.rate = 0, max.rate = 0.8)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>counts</code></td>
<td>
<p>expression matrix where each row is a gene and each column is a sample.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min.rate</code></td>
<td>
<p>the minimum dropout rate of all samples.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>max.rate</code></td>
<td>
<p>the maximum dropout rate of all samples.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The dropout event is modelled by a logistic distribution such that the low expression genes have 
higher probability of dropout. The expression value of genes in a sample are randomly set to zero
with probabilities associated with their true expression values until the desired dropout rate
for that sample is meet.
</p>


<h3>Value</h3>

<p>This function will return a list with the following components:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>counts</code></td>
<td>
<p>The modified expression matrix with the same dimension as input <code>counts</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>original.counts</code></td>
<td>
<p>The original input expression matrix.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dropout</code></td>
<td>
<p>The binary matrix indicating where the dropout events happen.</p>
</td>
</tr>
</table>
<h3>References</h3>

<p>Peter V. Kharchenko, Lev Silberstein, and David T. Scadden.
Bayesian approach to single-cell differential expression analysis.
Nature Methods, 11(7):740â€“742, 2014.
</p>


</div>