<div class="container">

<table style="width: 100%;"><tr>
<td>get_adjusted_deg_diff</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Calculate adjusted degree differences for given network</h2>

<h3>Description</h3>

<p>Calculate the adjusted degree differences for all genes in the given network.
</p>


<h3>Usage</h3>

<pre><code class="language-R">get_adjusted_deg_diff(net, log.expr.val, scale.degree = FALSE, p = 0.5)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>net</code></td>
<td>
<p>The binary adjacent matrix of differential expression ratio network.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>log.expr.val</code></td>
<td>
<p>Numeric vector containing the logarithmic scale gene expression values.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>scale.degree</code></td>
<td>
<p>Logical variable indicating whether the degree values are scaled according to the dropout rate.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p</code></td>
<td>
<p>The parameter for calculating the adjusted degree differences.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>This function will return a list with the following components:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>diff</code></td>
<td>
<p>A numeric vector containing the adjusted degree differences of all genes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>degree</code></td>
<td>
<p>A list containing the raw degree differences and sums of all genes.</p>
</td>
</tr>
</table>
</div>