<div class="container">

<table style="width: 100%;"><tr>
<td>make_DEG_data</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Simulate differentially expressed gene data (Gaussian)</h2>

<h3>Description</h3>

<p>Generate differentially expressed gene (DEG) data from Gaussian distribution.
</p>


<h3>Usage</h3>

<pre><code class="language-R">make_DEG_data(
  n.genes,
  n.samples.A,
  n.samples.B,
  exp.mean = 8,
  exp.sd = 2,
  alpha = 0.2,
  size.factor.sd = 0.1,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>n.genes</code></td>
<td>
<p>The total number of genes in the simulated data.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.samples.A</code></td>
<td>
<p>The number of samples in the group A.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.samples.B</code></td>
<td>
<p>The number of samples in the group B.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>exp.mean</code></td>
<td>
<p>The mean of log-normal distribution that determines gene-specific expression mean.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>exp.sd</code></td>
<td>
<p>The standard deviation of log-normal distribution that determines gene-specific expression means.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alpha</code></td>
<td>
<p>The dispersion ratio of gene-specific expression standard deviation to mean.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>size.factor.sd</code></td>
<td>
<p>The standard deviation of size factors for samples.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>The parameters passed to function <code>make_DEG_pattern</code>.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The expression values of each gene are assumed following a Gaussian distribution with 
gene-specific mean, which follows a log-normal distribution. The size factor for each
sample follows a Gaussian distribution with zero mean and specific standard deviation.
The heterogeneity of gene expression data is simulated by using the function <code>make_DEG_pattern</code>.
</p>


<h3>Value</h3>

<p>This function will return a list with the following components:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>DEG</code></td>
<td>
<p>The matrix of simulated DEG pattern, which is generated by <code>make_DEG_pattern</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>countsA</code></td>
<td>
<p>The expression matrix of group A. Each row represents a gene and each column represents a sample.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>countsB</code></td>
<td>
<p>The expression matrix of group B. Each row represents a gene and each column represents a sample.</p>
</td>
</tr>
</table>
</div>