<div class="container">

<table style="width: 100%;"><tr>
<td>testInputCORE</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
A table of DNA copy number gain events observed in 100 individual tumor cells

</h2>

<h3>Description</h3>

<p>Each observation in the table corresponds to a DNA copy number gain event in
one of 100 individual breast cancer cells. The entire length of the genome was
divided into 50009 bins. An event is an interval in chromosome <code>chrom</code>
whose start and end bin numbers are given by <code>start</code> and <code>end</code>.

</p>


<h3>Usage</h3>

<pre><code class="language-R">data(testInputCORE)</code></pre>


<h3>Format</h3>

<p>A data frame with 2490 observations on the following 3 variables.
</p>

<dl>
<dt><code>chrom</code></dt>
<dd>
<p>a numeric vector</p>
</dd>
<dt><code>start</code></dt>
<dd>
<p>a numeric vector</p>
</dd>
<dt><code>end</code></dt>
<dd>
<p>a numeric vector</p>
</dd>
</dl>
<h3>References</h3>

<p>Navin N, Kendall J, Troge J, Andrews P, Rodgers L, McIndoo J, Cook K, 
Stepansky A, Levy D, Esposito D et al. 2011. 
Tumour evolution inferred by single-cell sequencing. Nature 472(7341): 90-U119.

</p>


<h3>Examples</h3>

<pre><code class="language-R">data(testInputCORE)
## maybe str(testInputCORE) ; plot(testInputCORE) ...
</code></pre>


</div>