<div class="container">

<table style="width: 100%;"><tr>
<td>testInputBoundaries</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
A table of chromosome boundary positions for DNA copy number analysis

</h2>

<h3>Description</h3>

<p>The entire length of the genome was divided into 50009 bins, with chromosomes
laid out in the usual genomic order: 1,...,22,X,Y. Each observation in the
table provides the <code>start</code> and <code>end</code> bin numbers of each chromosome
(<code>chrom</code>). 

</p>


<h3>Usage</h3>

<pre><code class="language-R">data(testInputBoundaries)</code></pre>


<h3>Format</h3>

<p>A data frame with 24 observations on the following 3 variables.
</p>

<dl>
<dt><code>chrom</code></dt>
<dd>
<p>a numeric vector</p>
</dd>
<dt><code>start</code></dt>
<dd>
<p>a numeric vector</p>
</dd>
<dt><code>end</code></dt>
<dd>
<p>a numeric vector</p>
</dd>
</dl>
<h3>References</h3>

<p>Navin N, Kendall J, Troge J, Andrews P, Rodgers L, McIndoo J, Cook K,
Stepansky A, Levy D, Esposito D et al. 2011.
Tumour evolution inferred by single-cell sequencing. Nature 472(7341): 90-U119.

</p>


<h3>Examples</h3>

<pre><code class="language-R">data(testInputBoundaries)
## maybe str(testInputBoundaries) ; plot(testInputBoundaries) ...
</code></pre>


</div>