<div class="container">

<table style="width: 100%;"><tr>
<td>pcor.shrink</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Shrinkage Estimates of Partial Correlation and Partial Variance</h2>

<h3>Description</h3>

<p>The functions <code>pcor.shrink</code> and <code>pvar.shrink</code> compute shrinkage estimates
of partial correlation and partial variance, respectively.  
</p>


<h3>Usage</h3>

<pre><code class="language-R">pcor.shrink(x, lambda, w, verbose=TRUE)
pvar.shrink(x, lambda, lambda.var, w, verbose=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>a data matrix</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>lambda</code></td>
<td>
<p>the correlation shrinkage intensity (range 0-1).
If <code>lambda</code> is not specified (the default) it is estimated
using an analytic formula from Sch\"afer and Strimmer (2005) 
- see <code>cor.shrink</code>.  
For <code>lambda=0</code> the empirical correlations are recovered.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>lambda.var</code></td>
<td>
<p>the variance shrinkage intensity (range 0-1). 
If <code>lambda.var</code> is not specified (the default) it is estimated
using an analytic formula from Opgen-Rhein and Strimmer (2007)
- see details below.  
For <code>lambda.var=0</code> the empirical variances are recovered.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>w</code></td>
<td>
<p>optional: weights for each data point - if not specified uniform weights
are assumed (<code>w = rep(1/n, n)</code> with <code>n = nrow(x)</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>report progress while computing (default: TRUE)</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The partial variance <code class="reqn">var(X_k | rest)</code> is the variance of <code class="reqn">X_k</code> conditioned on the
remaining variables.  It equals the inverse of the corresponding diagonal entry
of the precision matrix (see Whittaker 1990).
</p>
<p>The partial correlations <code class="reqn">corr(X_k, X_l | rest)</code> is the correlation between
<code class="reqn">X_k</code> and <code class="reqn">X_l</code> conditioned on the remaining variables. It equals the sign-reversed
entries of the off-diagonal entries of the precision matrix, standardized
by the the squared root of the associated inverse partial variances.
</p>
<p>Note that using <code>pcor.shrink(x)</code> <em>much</em> faster than 
<code>cor2pcor(cor.shrink(x))</code>.
</p>
<p>For details about the shrinkage procedure consult Sch\"afer and Strimmer (2005),
Opgen-Rhein and Strimmer (2007), and the help page of <code>cov.shrink</code>.
</p>


<h3>Value</h3>

<p><code>pcor.shrink</code> returns the partial correlation matrix.  Attached to this
matrix are the standardized partial variances (i.e. PVAR/VAR) that 
can be retrieved using <code>attr</code> under the attribute "spv".
</p>
<p><code>pvar.shrink</code> returns the partial variances. 
</p>


<h3>Author(s)</h3>

<p>Juliane Sch\"afer  
and Korbinian Strimmer (<a href="https://strimmerlab.github.io">https://strimmerlab.github.io</a>).
</p>


<h3>References</h3>

<p>Opgen-Rhein, R., and K. Strimmer. 2007. Accurate ranking of 
differentially expressed genes by a distribution-free shrinkage 
approach.    Statist. Appl. Genet. Mol. Biol. <b>6</b>:9.
&lt;DOI:10.2202/1544-6115.1252&gt;
</p>
<p>Sch\"afer, J., and K. Strimmer. 2005.  A shrinkage approach to large-scale
covariance estimation and implications for functional genomics. 
Statist. Appl. Genet. Mol. Biol. <b>4</b>:32.
&lt;DOI:10.2202/1544-6115.1175&gt;
</p>
<p>Whittaker J. 1990.  Graphical Models in Applied Multivariate Statistics.
John Wiley, Chichester.
</p>


<h3>See Also</h3>

<p><code>invcov.shrink</code>, <code>cov.shrink</code>, <code>cor2pcor</code></p>


<h3>Examples</h3>

<pre><code class="language-R"># load corpcor library
library("corpcor")

# generate data matrix
p = 50
n = 10
X = matrix(rnorm(n*p), nrow = n, ncol = p)


# partial variance
pv = pvar.shrink(X)
pv

# partial correlations (fast and recommend way)
pcr1 = pcor.shrink(X)

# other possibilities to estimate partial correlations
pcr2 = cor2pcor( cor.shrink(X) )


# all the same
sum((pcr1 - pcr2)^2)

</code></pre>


</div>