<div class="container">

<table style="width: 100%;"><tr>
<td>EstimatorsFDR</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Estimators of Local False Discovery Rate (LFDR)</h2>

<h3>Description</h3>

<p>EstimatorsFDR is an R function that computes the Nonlocal False Discovery Rate (NFDR) and the estimators of local false discovery rate: Corrected False discovery Rate (CFDR) and Re-ranked False Discovery rate (RFDR).
</p>


<h3>Usage</h3>

<pre><code class="language-R">EstimatorsFDR(pvalue)
</code></pre>


<h3>Arguments</h3>

<table><tr style="vertical-align: top;">
<td><code>pvalue</code></td>
<td>
<p>Input numeric vector of pvalues.</p>
</td>
</tr></table>
<h3>Details</h3>

<p>The input is a list of pvalues. The pvalues can be obtained for example by performing Student's t-test between two datasets. The two groups can be data from healthy and disease states. Let <code class="reqn">i=1, 2, ..., N</code>, where <code class="reqn">i</code> represents the <code class="reqn">ith</code> feature (SNP or gene, for example). Then, for each <code class="reqn">i</code>, the hypothesis indicator <code class="reqn">A_i</code> can have two possible values.<br><code class="reqn">A_i=0</code>, if the <code class="reqn">ith</code> null hypothesis is true, or <br><code class="reqn">A_i=1</code>,  if the <code class="reqn">ith</code> null hypothesis is not true, <br>
where the null hypothesis is defined by: the <code class="reqn">ith</code> feature is unaffected by a treatment, unassociated with a disease, etc.
The values for each estimator (NFDR, CFDR, RFDR) indicate the probability that the null hypothesis of the <code class="reqn">ith</code> feature is true (<code class="reqn">A_i=0</code>) given the statistics <code class="reqn">T_i</code>. The alternative hypothesis is true if <code class="reqn">A_i=1</code>.
For example, in gene expression data analysis, if the null hypothesis is true, this would mean that the genes are not differentially expressed.
</p>


<h3>Value</h3>

<p>The output returns three lists. It returns the NFDR, CFDR, and RFDR estimators: </p>

<table>
<tr>
<td style="text-align: left;"> NFDR</td>
<td style="text-align: left;"> nonlocal FDR</td>
</tr>
<tr>
<td style="text-align: left;"> CFDR</td>
<td style="text-align: left;"> corrected FDR</td>
</tr>
<tr>
<td style="text-align: left;"> RFDR</td>
<td style="text-align: left;"> re-ranked FDR</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>Code: Abbas Rahal. <br> Documentation: Anna Akpawu, Justin Chitpin and Abbas Rahal. <br> Maintainer: Abbas Rahal &lt;Abbas.Rahal13@gmail.com&gt;
</p>


<h3>References</h3>

<p>Bickel, D.R., Rahal, A. (2019). Correcting false discovery rates for their bias toward
false positives. Communications in Statistics - Simulation and Computation, <a href="https://tinyurl.com/kkdc9rk8">https://tinyurl.com/kkdc9rk8</a>.
</p>
<p>Bickel, D. R. (2015). Corrigendum to: Simple estimators of false discovery rates given as few as one or two p-values without strong parametric assumptions. Statistical Applications in Genetics and Molecular Biology, 2015, 14, 225.
</p>
<p>Bickel, D. R. (2013). Simple estimators of false discovery rates given as few as one or two p-values without strong parametric assumptions. Statistical Applications in Genetics and Molecular Biology, 2013, 12, 529-543.
</p>


<h3>Examples</h3>

<pre><code class="language-R">#The examples below are from the "ProData" package.
#In order to use the "Prodata" input you would first need to install the ProData package.
#You will also need the function exprs in this package.
#First, make sure that the ProData package is properly installed:
#source("https://bioconductor.org/biocLite.R")
#biocLite("ProData")
#library(ProData)
#data("f45cbmk")
#q1&lt;- quantile(as(exprs(f45cbmk[, pData(f45cbmk)$GROUP == "B"]), "numeric"), probs = 0.25)
#logish&lt;- function(x){log(x + q1)}
#Vectors of proteins for 20 patients ER/PR-positive and Healthy
#Y&lt;- logish(exprs(f45cbmk[, pData(f45cbmk)$GROUP == "B"])) # Control (Healthy)
#X.ER&lt;- logish(exprs(f45cbmk[, pData(f45cbmk)$GROUP == "C"])) # Case ER/PR-positive
#pvalue&lt;- NULL
#for (i in 1:nrow(X.ER))
#{
# t&lt;-t.test(x=X.ER[i,], y=Y[i,], alternative = "two.sided")
# pvalue[i]&lt;- t$p.value
#}

#The pvalues obtained from the t-test:
pvalue&lt;- c(0.1981, 0.3794, 0.000001443, 0.02325, 0.03264, 0.07263, 0.02965, 0.8016, 0.8888,
          0.9133, 0.2971, 0.4573, 0.2815, 0.0007119, 0.5743, 0.927, 0.369, 0.8478, 0.38, 0.9904)
output&lt;- EstimatorsFDR(pvalue)
#Three lists
output$NFDR
output$CFDR
output$RFDR
</code></pre>


</div>