<div class="container">

<table style="width: 100%;"><tr>
<td>signature</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Mutational signatures data from COSMIC, Catalogue Of Somatic Mutations In
Cancer (v3.3 - June 2022)</h2>

<h3>Description</h3>

<p>Mutational signatures data from COSMIC, Catalogue Of Somatic Mutations In
Cancer (v3.3 - June 2022)
</p>


<h3>Usage</h3>

<pre><code class="language-R">signature
</code></pre>


<h3>Format</h3>

<p>A list with the following elements:
</p>

<ul>
<li>
<p> GRCh37: Homo sapiens (human) genome assembly GRCh37.
</p>
</li>
<li>
<p> GRCh38: Homo sapiens (human) genome assembly GRCh38.
</p>
</li>
<li>
<p> mm9: Mus musculus (house mouse) genome assembly mm9.
</p>
</li>
<li>
<p> mm10: Mus musculus (house mouse) genome assembly mm10.
</p>
</li>
<li>
<p> rn6: Rattus norvegicus (Norway rat) genome assembly rn6.
</p>
</li>
</ul>
<p>Each element contains the sub elements:
</p>

<ul>
<li>
<p> SBS96: Strand-agnostic single-base substitutions in trinucleotide context.
</p>
</li>
<li>
<p> DBS78: Strand-agnostic doublet-base substitutions.
</p>
</li>
</ul>
<p>Element GRCh37 contains the additional sub elements:
</p>

<ul><li>
<p> ID: Strand-agnostic indels (short insertions and deletions).
</p>
</li></ul>
<h3>Remark</h3>

<p>The signatures are all genome signatures. <br> See CatalogRowOrder
in package ICAMS for the classification of mutation types.
</p>


<h3>Source</h3>

<p><a href="https://cancer.sanger.ac.uk/signatures/">https://cancer.sanger.ac.uk/signatures/</a>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">sbs96_sig &lt;- signature$GRCh37$SBS96

</code></pre>


</div>