<div class="container">

<table style="width: 100%;"><tr>
<td>lTransformer</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Log-based transformation
</h2>

<h3>Description</h3>

<p>Log-based transformation.
</p>


<h3>Usage</h3>

<pre><code class="language-R">lTransformer(mat, low = 1e-04, upp = 100)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>mat</code></td>
<td>

<p>G x n data matrix, where G is the number of genes and n is the number of subjects
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>low</code></td>
<td>

<p>lower bound for the model parameter
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>upp</code></td>
<td>

<p>upper bound for the model parameter
</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Denote <code class="reqn">x_{gi}</code> as the expression level of the <code class="reqn">g</code>-th gene for
the <code class="reqn">i</code>-th subject.
We perform the log transformation
</p>
<p style="text-align: center;"><code class="reqn">y_{gi}=\log_2\left(x_{gi} + \frac{1}{\delta}\right)</code>
</p>
<p>. The optimal value for the parameter <code class="reqn">\delta</code> is to minimize the squared 
difference between the sample mean and the sample median of the pooled data <code class="reqn">y_{gi}</code>, <code class="reqn">g=1, \ldots, G</code>, <code class="reqn">i=1, \ldots, n</code>,
where <code class="reqn">G</code> is the number of genes and <code class="reqn">n</code> is the number of subjects.
</p>


<h3>Value</h3>

<p>A list with 3 elements:	   
</p>
<table>
<tr style="vertical-align: top;">
<td><code>res.delta </code></td>
<td>
<p>An object returned by <code>optimize</code> function</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>delta </code></td>
<td>
<p>model parameter</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mat2 </code></td>
<td>
<p>transformed data matrix having the same dimension as <code>mat</code></p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>Zeyu Zhang, Danyang Yu, Minseok Seo, Craig P. Hersh, Scott T. Weiss, Weiliang Qiu
</p>


<h3>References</h3>

<p>Zhang Z, Yu D, Seo M, Hersh CP, Weiss ST, Qiu W. 
Novel Data Transformations for RNA-seq Differential Expression Analysis. 
(2019) 9:4820 https://rdcu.be/brDe5
</p>


<h3>Examples</h3>

<pre><code class="language-R">library(Biobase)

data(es)
print(es)

# expression set
ex = exprs(es)
print(dim(ex))
print(ex[1:3,1:2])

# mean-median before transformation
vec = c(ex)
m = mean(vec)
md = median(vec)
diff = m - md
cat("m=", m, ", md=", md, ", diff=", diff, "\n")

res = lTransformer(mat = ex)

# estimated model parameter
print(res$delta)

# mean-median after transformation
vec2 = c(res$mat2)
m2 = mean(vec2)
md2 = median(vec2)
diff2 = m2 - md2
cat("m2=", m2, ", md2=", md2, ", diff2=", diff2, "\n")
</code></pre>


</div>