<div class="container">

<table style="width: 100%;"><tr>
<td>fpkm</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Convert Counts to Fragments per Kilobase of Transcript per Million (FPKM)
Convert counts to Fragments Per Kilobase of transcript per Million mapped reads (FPKM)
</h2>

<h3>Description</h3>

<p>fpkm() function returns a numeric matrix normalized by library size and feature length.
</p>


<h3>Usage</h3>

<pre><code class="language-R">fpkm (counts, featureLength, meanFragmentLength)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>counts</code></td>
<td>
<p>A numeric matrix of raw feature counts</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>featureLength</code></td>
<td>
<p>A numeric vector with feature lengths which can be obtained using 'biomaRt' package.
The length of items should be as the same of rows in read count matrix.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>meanFragmentLength</code></td>
<td>
<p>A numeric vector with mean fragment lengths, 
which can be calculated using 'CollectInsertSizeMetrics(Picard)' tool.
The length of items should be as the same of columns in read count matrix.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Implements the algorithm described in Trapnell,C. et al. (2010). "Transcript assembly and quantification by RNA-seq reveals unannotated transcripts and isoform switching during cell differentiation". Nat. Biotechnol., 28, 511-515. doi: 10.1038/nbt.1621. This function takes a matrix of read feature counts of RNA-seq, a numeric vector with feature lengths which can be retrieved using the 'biomaRt' package, and a numeric vector with mean fragment length which can be calculated using the 'CollectInsertSizeMetrics(Picard)' tool. It then returns a matrix of FPKM normalised data by library size and feature effective length. Please see the original manuscript for further details.
</p>


<h3>Value</h3>

<p>A data matrix normalized by library size and feature length.
</p>


<h3>References</h3>

<p>Trapnell,C. et al. (2010) Transcript assembly and quantification by RNA-seq reveals unannotated transcripts and isoform switching during cell differentiation. 
Nat. Biotechnol., 28, 511-515. doi: 10.1038/nbt.1621.
</p>
<p>Lior Pachter. Models for transcript quantification from RNA-Seq.
arXiv:1104.3889v2.
</p>


<h3>Examples</h3>

<pre><code class="language-R">library(countToFPKM)

file.readcounts &lt;- system.file("extdata", "RNA-seq.read.counts.csv", package="countToFPKM")
file.annotations &lt;- system.file("extdata", "Biomart.annotations.hg38.txt", package="countToFPKM")
file.sample.metrics &lt;- system.file("extdata", "RNA-seq.samples.metrics.txt", package="countToFPKM")


# Import the read count matrix data into R.
counts &lt;- as.matrix(read.csv(file.readcounts))

# Import feature annotations.
# Assign feature length into a numeric vector.
gene.annotations &lt;- read.table(file.annotations, sep="\t", header=TRUE)
featureLength &lt;- gene.annotations$length

# Import sample metrics.
# Assign mean fragment length into a numeric vector.
samples.metrics &lt;- read.table(file.sample.metrics, sep="\t", header=TRUE)
meanFragmentLength &lt;- samples.metrics$meanFragmentLength

# Return FPKM into a numeric matrix.
fpkm_matrix &lt;- fpkm (counts, featureLength, meanFragmentLength)
</code></pre>


</div>