<div class="container">

<table style="width: 100%;"><tr>
<td>AssayMutRatio</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Calculate the mutation detection rate using different assays</h2>

<h3>Description</h3>

<p>This function is to use the well established assays information to detect mutations in different
SARS-CoV-2 genomic sites. The output will be series of figures presenting the mutation profile using a
specific assay and a figure for comparison between the mutation detection rate in each primers binding region.
</p>


<h3>Usage</h3>

<pre><code class="language-R">AssayMutRatio(
  nucmerr = nucmerr,
  assays = assays,
  totalsample = totalsample,
  plotType = "barplot",
  outdir = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>nucmerr</code></td>
<td>
<p>Mutation information containing group list(derived from "nucmer" object using "nucmerRMD" function).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>assays</code></td>
<td>
<p>Assays dataframe including the detection ranges of mutations.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>totalsample</code></td>
<td>
<p>Total sample number, total cleared GISAID fasta data.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plotType</code></td>
<td>
<p>Figure type for either "barplot" or "logtrans".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>outdir</code></td>
<td>
<p>The output directory.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Plot the selected figure type as output.
</p>


<h3>Examples</h3>

<pre><code class="language-R">data("nucmerr")
data("assays")
Total &lt;- 52 ## Total Cleared GISAID fasta data, sekitseq
#outdir &lt;- tempdir()
#Output the results
AssayMutRatio(nucmerr = nucmerr,
              assays = assays,
              totalsample = Total,
              plotType = "logtrans",
              outdir = NULL)
</code></pre>


</div>