<div class="container">

<table style="width: 100%;"><tr>
<td>globalSNPprofile</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Global single nucleotide polymorphism (SNP) profiling in virus genome</h2>

<h3>Description</h3>

<p>This function is to visualize the global SNP pattern in the
SARS-CoV-2 genome.
</p>


<h3>Usage</h3>

<pre><code class="language-R">globalSNPprofile(
  nucmerr = nucmerr,
  outdir = NULL,
  figure_Type = "heatmap",
  country = "global",
  top = 5
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>nucmerr</code></td>
<td>
<p>Mutation information containing group list(derived from "nucmer" object using "nucmerRMD" function).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>outdir</code></td>
<td>
<p>The output directory.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>figure_Type</code></td>
<td>
<p>Figure type for either "heatmap" or "count".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>country</code></td>
<td>
<p>Choose a country to plot the mutational pattern or
choose "global" to profile mutations across all countries. The
default is "global".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>top</code></td>
<td>
<p>The number of mutational classes to plot.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Plot the selected figure type as output.
</p>


<h3>Examples</h3>

<pre><code class="language-R">data("nucmerr")
outdir &lt;- tempdir()
globalSNPprofile(nucmerr = nucmerr,
                 outdir = outdir,
                 figure_Type = "heatmap",
                 country = "global",
                 top = 5)
</code></pre>


</div>