<div class="container">

<table style="width: 100%;"><tr>
<td>MutByGene</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Plot mutation counts for certain genes</h2>

<h3>Description</h3>

<p>After annotating the mutations, this function is to plot the counts of mutational events for
each gene in the SARS-CoV-2 genome.
</p>


<h3>Usage</h3>

<pre><code class="language-R">MutByGene(nucmerr = nucmerr, gff3 = gff3, figurelist = FALSE, outdir = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>nucmerr</code></td>
<td>
<p>Mutation information containing group list(derived from "nucmer" object using "nucmerRMD" function).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gff3</code></td>
<td>
<p>"GFF3" format gene position data for SARS-Cov-2(the "GFF3" file should include columns named: "Gene", "Start", "Stop").</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>figurelist</code></td>
<td>
<p>Whether to output the integrated plot list for each gene.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>outdir</code></td>
<td>
<p>The output directory, if the figurelist = TRUE, output the figure in the R session.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Plot the mutation counts figure for each gene as output.
</p>


<h3>Examples</h3>

<pre><code class="language-R">data("nucmerr")
data("gene_position")
#outdir &lt;- tempdir()
MutByGene(nucmerr = nucmerr, gff3 = gene_position, figurelist = FALSE, outdir = NULL)
#if figurelist = TRUE, the recommendation for figure display(in pixel)is: width=1650, height=1300
</code></pre>


</div>