<div class="container">

<table style="width: 100%;"><tr>
<td>mergeEvents</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Merge neighboring events of single nucleotide polymorphism (SNP), insertion and deletion.</h2>

<h3>Description</h3>

<p>The first step for handling the nucmer object, then effects of mutations
can be analysed using "indelSNP" function.
</p>


<h3>Usage</h3>

<pre><code class="language-R">mergeEvents(nucmer = nucmer)
</code></pre>


<h3>Arguments</h3>

<table><tr style="vertical-align: top;">
<td><code>nucmer</code></td>
<td>
<p>An object called "nucmer", mutation information derived from
"nucmer.snp" variant file by "seqkit" software and "nucmer SNP-calling"
scripts.</p>
</td>
</tr></table>
<h3>Value</h3>

<p>An updated "nucmer" object.
</p>


<h3>Examples</h3>

<pre><code class="language-R">#The example data:
data("nucmer")
#options(stringsAsFactors = FALSE)

#The input nucmer object can be made by the comment below:
#nucmer&lt;-read.delim("nucmer.snps",as.is=TRUE,skip=4,header=FALSE)
#colnames(nucmer)&lt;-c("rpos","rvar","qvar","qpos","","","","",
#"rlength","qlength","","","rname","qname")
#rownames(nucmer)&lt;-paste0("var",1:nrow(nucmer))

# Fix IUPAC codes
nucmer&lt;-nucmer[!nucmer$qvar%in%c("B","D","H","K","M","N","R","S","V","W","Y"),]
nucmer&lt;- mergeEvents(nucmer = nucmer)## This will update the nucmer object
</code></pre>


</div>