<div class="container">

<table style="width: 100%;"><tr>
<td>LastfiveNrMutation</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Bacth assay analysis for last five Nr of primers</h2>

<h3>Description</h3>

<p>Last five nucleotides of primer mutation count/type for any reverse transcription
polymerase chain reaction (RT-PCR) primer.
</p>


<h3>Usage</h3>

<pre><code class="language-R">LastfiveNrMutation(
  nucmerr = nucmerr,
  assays = assays,
  totalsample = totalsample,
  figurelist = FALSE,
  outdir = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>nucmerr</code></td>
<td>
<p>Mutation information containing group list(derived from "nucmer" object using "nucmerRMD" function).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>assays</code></td>
<td>
<p>Assays dataframe including the detection ranges of mutations.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>totalsample</code></td>
<td>
<p>Total sample number, total cleared GISAID fasta data.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>figurelist</code></td>
<td>
<p>Whether to output the integrated plot list for each assay.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>outdir</code></td>
<td>
<p>The output directory. if the figurelist = TRUE, output the figure in the R session.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Plot the mutation counts(last five nucleotides for each primer) for each assay as output.
</p>


<h3>Examples</h3>

<pre><code class="language-R">data("nucmerr")
data("assays")
totalsample &lt;- 434
#outdir &lt;- tempdir()
LastfiveNrMutation(nucmerr = nucmerr,
                   assays = assays,
                   totalsample = totalsample,
                   figurelist = FALSE,
                   outdir = NULL)
</code></pre>


</div>