<div class="container">

<table style="width: 100%;"><tr>
<td>splsdrcox</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>sPLS-DRCOX</h2>

<h3>Description</h3>

<p>This function performs a sparse partial least squares deviance residual Cox (sPLS-DRCOX)
(based on plsRcox R package). The function returns a Coxmos model with the attribute model as
"sPLS-DRCOX".
</p>


<h3>Usage</h3>

<pre><code class="language-R">splsdrcox(
  X,
  Y,
  n.comp = 4,
  penalty = 0.5,
  x.center = TRUE,
  x.scale = FALSE,
  remove_near_zero_variance = TRUE,
  remove_zero_variance = FALSE,
  toKeep.zv = NULL,
  remove_non_significant = FALSE,
  alpha = 0.05,
  MIN_EPV = 5,
  returnData = TRUE,
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>X</code></td>
<td>
<p>Numeric matrix or data.frame. Explanatory variables. Qualitative variables must be
transform into binary variables.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Y</code></td>
<td>
<p>Numeric matrix or data.frame. Response variables. Object must have two columns named as
"time" and "event". For event column, accepted values are: 0/1 or FALSE/TRUE for censored and event
observations.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.comp</code></td>
<td>
<p>Numeric. Number of latent components to compute for the (s)PLS model (default: 10).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>penalty</code></td>
<td>
<p>Numeric. Penalty for sPLS-DRCOX. If penalty = 0 no penalty is applied, when
penalty = 1 maximum penalty (no variables are selected) based on 'plsRcox' penalty. Equal or greater
than 1 cannot be selected (default: 0.5).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>x.center</code></td>
<td>
<p>Logical. If x.center = TRUE, X matrix is centered to zero means (default: TRUE).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>x.scale</code></td>
<td>
<p>Logical. If x.scale = TRUE, X matrix is scaled to unit variances (default: FALSE).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>remove_near_zero_variance</code></td>
<td>
<p>Logical. If remove_near_zero_variance = TRUE, near zero variance
variables will be removed (default: TRUE).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>remove_zero_variance</code></td>
<td>
<p>Logical. If remove_zero_variance = TRUE, zero variance variables will
be removed (default: TRUE).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>toKeep.zv</code></td>
<td>
<p>Character vector. Name of variables in X to not be deleted by (near) zero
variance filtering (default: NULL).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>remove_non_significant</code></td>
<td>
<p>Logical. If remove_non_significant = TRUE, non-significant
variables/components in final cox model will be removed until all variables are significant by
forward selection (default: FALSE).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alpha</code></td>
<td>
<p>Numeric. Numerical values are regarded as significant if they fall below the
threshold (default: 0.05).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>MIN_EPV</code></td>
<td>
<p>Numeric. Minimum number of Events Per Variable (EPV) you want reach for the final
cox model. Used to restrict the number of variables/components can be computed in final cox models.
If the minimum is not meet, the model cannot be computed (default: 5).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>returnData</code></td>
<td>
<p>Logical. Return original and normalized X and Y matrices (default: TRUE).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>Logical. If verbose = TRUE, extra messages could be displayed (default: FALSE).</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The <code>sPLS-DRCOX</code> function implements the sparse partial least squares deviance residual Cox
(sPLS-DRCOX) model, a specialized approach tailored for survival analysis. This method integrates
the strengths of the sparse partial least squares (sPLS) technique with the Cox proportional hazards
model, leveraging deviance residuals as a bridge.
</p>
<p>The function's core lies in its ability to handle high-dimensional data, often encountered in
genomics or other omics studies. By incorporating the <code>penalty</code> parameter, which governs the sparsity
level, the function offers a fine-grained control over variable selection. This ensures that only
the most informative predictors contribute to the model, enhancing interpretability and reducing
overfitting.
</p>
<p>Data preprocessing is seamlessly integrated, with options to center and scale the predictors, and
to remove variables exhibiting near-zero or zero variance. The function also provides a mechanism
to retain specific variables, regardless of their variance, ensuring that domain-specific knowledge
can be incorporated.
</p>
<p>The output is comprehensive, detailing both the sPLS and Cox model components. It provides insights
into the selected variables, their contributions across latent components, and the overall fit of
the survival model. This rich output aids in understanding the underlying relationships between
predictors and survival outcomes.
</p>
<p>The <code>sPLS-DRCOX</code> function is grounded in established methodologies and is a valuable tool for
researchers aiming to unravel complex survival associations in high-dimensional datasets.
</p>


<h3>Value</h3>

<p>Instance of class "Coxmos" and model "sPLS-DRCOX". The class contains the following
elements:
<code>X</code>: List of normalized X data information.
</p>

<ul>
<li> <p><code>(data)</code>: normalized X matrix
</p>
</li>
<li> <p><code>(weightings)</code>: sPLS weights
</p>
</li>
<li> <p><code>(weightings_norm)</code>: sPLS normalize weights
</p>
</li>
<li> <p><code>(W.star)</code>: sPLS W* vector
</p>
</li>
<li> <p><code>(loadings)</code>: sPLS loadings
</p>
</li>
<li> <p><code>(scores)</code>: sPLS scores/variates
</p>
</li>
<li> <p><code>(E)</code>: error matrices
</p>
</li>
<li> <p><code>(x.mean)</code>: mean values for X matrix
</p>
</li>
<li> <p><code>(x.sd)</code>: standard deviation for X matrix
</p>
</li>
</ul>
<p><code>Y</code>: List of normalized Y data information.
</p>

<ul>
<li> <p><code>(deviance_residuals)</code>: deviance residual vector used as Y matrix in the sPLS.
</p>
</li>
<li> <p><code>(dr.mean)</code>: mean values for deviance residuals Y matrix
</p>
</li>
<li> <p><code>(dr.sd)</code>: standard deviation for deviance residuals Y matrix'
</p>
</li>
<li> <p><code>(data)</code>: normalized X matrix
</p>
</li>
<li> <p><code>(weightings)</code>: sPLS weights
</p>
</li>
<li> <p><code>(loadings)</code>: sPLS loadings
</p>
</li>
<li> <p><code>(scores)</code>: sPLS scores/variates
</p>
</li>
<li> <p><code>(ratio)</code>: r value for the sPLS model (used to perform predictions)
</p>
</li>
<li> <p><code>(y.mean)</code>: mean values for Y matrix
</p>
</li>
<li> <p><code>(y.sd)</code>: standard deviation for Y matrix'
</p>
</li>
</ul>
<p><code>survival_model</code>: List of survival model information.
</p>

<ul>
<li> <p><code>fit</code>: coxph object.
</p>
</li>
<li> <p><code>AIC</code>: AIC of cox model.
</p>
</li>
<li> <p><code>BIC</code>: BIC of cox model.
</p>
</li>
<li> <p><code>lp</code>: linear predictors for train data.
</p>
</li>
<li> <p><code>coef</code>: Coefficients for cox model.
</p>
</li>
<li> <p><code>YChapeau</code>: Y Chapeau residuals.
</p>
</li>
<li> <p><code>Yresidus</code>: Y residuals.
</p>
</li>
</ul>
<p><code>penalty</code>: Penalty value selected.
</p>
<p><code>n.comp</code>: Number of components selected.
</p>
<p><code>var_by_component</code>: Variables selected in each PLS component.
</p>
<p><code>call</code>: call function
</p>
<p><code>X_input</code>: X input matrix
</p>
<p><code>Y_input</code>: Y input matrix
</p>
<p><code>B.hat</code>: sPLS beta matrix
</p>
<p><code>R2</code>: sPLS R2
</p>
<p><code>SCR</code>: sPLS SCR
</p>
<p><code>SCT</code>: sPLS SCT
</p>
<p><code>alpha</code>: alpha value selected
</p>
<p><code>nsv</code>: Variables removed by cox alpha cutoff.
</p>
<p><code>nzv</code>: Variables removed by remove_near_zero_variance or remove_zero_variance.
</p>
<p><code>nz_coeffvar</code>: Variables removed by coefficient variation near zero.
</p>
<p><code>class</code>: Model class.
</p>
<p><code>time</code>: time consumed for running the cox analysis.
</p>


<h3>Author(s)</h3>

<p>Pedro Salguero Garcia. Maintainer: pedsalga@upv.edu.es
</p>


<h3>References</h3>

<p>Bastien P (2008).
“Deviance residuals based PLS regression for censored data in high dimensional setting.”
<em>Chemometrics and Intelligent Laboratory Systems</em>.
<a href="https://doi.org/10.1016/j.chemolab.2007.09.009">doi:10.1016/j.chemolab.2007.09.009</a>, <a href="https://www.sciencedirect.com/science/article/abs/pii/S0169743907001931?via%3Dihub">https://www.sciencedirect.com/science/article/abs/pii/S0169743907001931?via%3Dihub</a>.
Bastien P, Bastien P, Bertrand F, Meyer N, Meyer N, Meyer N, Maumy-Bertrand M (2015).
“Deviance residuals-based sparse PLS and sparse kernel PLS regression for censored data.”
<em>Bioinformatics</em>.
<a href="https://academic.oup.com/bioinformatics/article/31/3/397/2366078">https://academic.oup.com/bioinformatics/article/31/3/397/2366078</a>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">data("X_proteomic")
data("Y_proteomic")
X &lt;- X_proteomic[,1:50]
Y &lt;- Y_proteomic
splsdrcox(X, Y, n.comp = 3, penalty = 0.25, x.center = TRUE, x.scale = TRUE)
</code></pre>


</div>