<div class="container">

<table style="width: 100%;"><tr>
<td>estim.pi0</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Estimation of the proportion of true null hypotheses
</h2>

<h3>Description</h3>

<p>From a proteomics viewpoint, this function estimates the global proportion of proteins (resp. of peptides) that are non differentially abundant from the tested protein list (resp. from the tested peptide list).
This proportion is later used as a correcting factor to compute the adjusted p-values, that are in turn used to tune a threshold according to a desired false discovery rate.
</p>
<p>From a statistical viewpoint, this function allows estimating the proportion of true null hypotheses (pi0) from a vector of raw p-values following eight different estimation methods from the literature.
</p>


<h3>Usage</h3>

<pre><code class="language-R">estim.pi0(p, pi0.method = "ALL", nbins = 20, pz = 0.05)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>p</code></td>
<td>
<p>Numeric vector of raw p-values. Raw p-values are assumed without missing values, and between 0 and 1.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pi0.method</code></td>
<td>
<p>Name of an estimation method for the proportion of true null hypotheses among <code>"st.boot"</code>, <code>"st.spline"</code>, <code>"langaas"</code>, <code>"jiang"</code>, <code>"histo"</code>, <code>"pounds"</code>, <code>"abh"</code> or <code>"slim"</code>. Default is <code>"ALL"</code>: all the eight estimation methods are performed simultaneously.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nbins</code></td>
<td>
<p>Number of bins. Parameter used for the <code>"jiang"</code> and  <code>"histo"</code> methods. Default is 20.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pz</code></td>
<td>
<p>P-value threshold such as p-values below are associated to false null hypotheses. Used for the <code>"slim"</code> method. Wang, Tuominen and Tsai (2011) suggest to take a value between 0.01 and 0.1. Default is 0.05.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function allows to estimate the proportion of true null hypotheses following different estimation methods :
</p>

<table>
<tr>
<td style="text-align: left;">
<code>"abh"</code>
</td>
<td style="text-align: left;"> the least slope method proposed in Benjamini and Hochberg (2000).</td>
</tr>
<tr>
<td style="text-align: left;">
 </td>
<td style="text-align: left;"> </td>
</tr>
<tr>
<td style="text-align: left;">
<code>"st.spline"</code>
</td>
<td style="text-align: left;"> the smoother method described in Storey and Tibshirani (2003). </td>
</tr>
<tr>
<td style="text-align: left;"> 
 </td>
<td style="text-align: left;"> The <code>qvalue</code> function of R package <code>qvalue</code> with default tuning is used (Storey (2015)).</td>
</tr>
<tr>
<td style="text-align: left;">
  </td>
<td style="text-align: left;"> </td>
</tr>
<tr>
<td style="text-align: left;">
<code>"st.boot"</code>
</td>
<td style="text-align: left;"> the bootstrap method described in Storey et al. (2004).</td>
</tr>
<tr>
<td style="text-align: left;">
 </td>
<td style="text-align: left;"> The <code>qvalue</code> function of R package <code>qvalue</code> with default tuning is used (Storey (2015)).</td>
</tr>
<tr>
<td style="text-align: left;">
 </td>
<td style="text-align: left;"> </td>
</tr>
<tr>
<td style="text-align: left;">
<code>"langaas"</code>
</td>
<td style="text-align: left;"> the method described in Langaas, Ferkingstad and Lindqvist (2005) using a convex</td>
</tr>
<tr>
<td style="text-align: left;">
  </td>
<td style="text-align: left;"> decreasing density estimate for p-values. The <code>convest</code> function of R package <code>limma</code>
</td>
</tr>
<tr>
<td style="text-align: left;">
 </td>
<td style="text-align: left;">  with default tuning is used (Ritchie et al. (2015)).</td>
</tr>
<tr>
<td style="text-align: left;">
 </td>
<td style="text-align: left;"> </td>
</tr>
<tr>
<td style="text-align: left;">
<code>"histo"</code>
</td>
<td style="text-align: left;"> the histogram method described in Nettleton, Hwang, Caldo and Wise (2006).</td>
</tr>
<tr>
<td style="text-align: left;">
 </td>
<td style="text-align: left;"> </td>
</tr>
<tr>
<td style="text-align: left;">
<code>"pounds"</code>
</td>
<td style="text-align: left;"> the conservative estimate described in Pounds and Cheng (2006).</td>
</tr>
<tr>
<td style="text-align: left;">
 </td>
<td style="text-align: left;"> </td>
</tr>
<tr>
<td style="text-align: left;">
<code>"jiang"</code>
</td>
<td style="text-align: left;"> the average estimate method described in Jiang and Doerge (2008).</td>
</tr>
<tr>
<td style="text-align: left;">
 </td>
<td style="text-align: left;"> </td>
</tr>
<tr>
<td style="text-align: left;">
<code>"slim"</code>
</td>
<td style="text-align: left;"> the method of Wang, Tuominen and Tsai (2011) using a sliding linear model.</td>
</tr>
<tr>
<td style="text-align: left;">
 </td>
<td style="text-align: left;"> The default tuning suggested by Wang, Tuominen and Tsai (2011) is used.</td>
</tr>
<tr>
<td style="text-align: left;">
 </td>
<td style="text-align: left;"> Using their notations, lambda1 is fixed to 0.1, n to 10 and B to 100.
</td>
</tr>
</table>
<p>To take into account of right censorship on the vector of p-values, each p-value is divided by the maximum p-value present in <code>p</code>. Accordingly, the p-values of the true null hypotheses are assumed uniformly distributed between 0 and this maximum. This kind of censorship happens in proteomics when a first thresholding is performed on the fold-changes. 
</p>
<p>If you want to assume that the p-values are uniformly distributed between 0 and 1, replace <code>p</code> by <code>c(p,1)</code> when using <code>estim.pi0</code>. 
</p>


<h3>Value</h3>

<table><tr style="vertical-align: top;">
<td><code>pi0</code></td>
<td>
<p>Numeric value of the estimated proportion of true null hypotheses from the selected method; Numeric vector if <code>pi0.method="ALL"</code>.</p>
</td>
</tr></table>
<h3>Author(s)</h3>

<p>Quentin Giai Gianetto &lt;quentin2g@yahoo.fr&gt;
</p>


<h3>References</h3>

<p>Y. Benjamini and Y. Hochberg. On the adaptive control of the false discovery rate in multiple testing with independent statistics. Journal of Educational and Behavioral Statistics, 25(1):60-83, 2000.
</p>
<p>H. Jiang and R.W. Doerge. Estimating the proportion of true null hypotheses for multiple comparisons. Cancer informatics, 6:25, 2008.
</p>
<p>M. Langaas, B.H. Lindqvist, and E. Ferkingstad. Estimating the proportion of true null hypotheses, with application to dna microarray data. Journal of the Royal Statistical Society: Series B (Statistical Methodology), 67(4):555-572, 2005.
</p>
<p>D. Nettleton, J.T.G. Hwang, R.A. Caldo, and R.P. Wise. Estimating the number of true null hypotheses from a histogram of p values. Journal of Agricultural, Biological, and Environmental Statistics, 11(3):337-356, 2006.
</p>
<p>S. Pounds and C. Cheng. Robust estimation of the false discovery rate. Bioinformatics, 22(16):1979-1987, 2006.
</p>
<p>M.E. Ritchie, B. Phipson, D. Wu, Y. Hu, C.W. Law, W. Shi and G.K. Smyth. “limma powers differential expression analyses for RNA-sequencing and microarray studies.” Nucleic Acids Research, 43(7), pp.e47. 2015.
</p>
<p>J.D. Storey, J.E. Taylor, and D. Siegmund. Strong control, conservative point estimation and simultaneous conservative consistency of false discovery rates: a unified approach. Journal of the Royal Statistical Society: Series B (Statistical Methodology), 66(1):187-205, 2004.
</p>
<p>J.D. Storey and R. Tibshirani. Statistical significance for genomewide studies. Proceedings of the National Academy of Sciences, 100(16):9440-9445, 2003.
</p>
<p>J.D. Storey. qvalue: Q-value estimation for false discovery rate control. R package version 2.0.0, http://qvalue.princeton.edu/, http://github.com/jdstorey/qvalue. 2015.
</p>
<p>H.-Q. Wang, L.K. Tuominen, and C.-J. Tsai. SLIM: a sliding linear model for estimating the proportion of true null hypotheses in datasets with dependence structures. Bioinformatics, 27(2):225-231, 2011.
</p>


<h3>See Also</h3>

<p><code>calibration.plot</code>, <code>adjust.p</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">#get p-values
data(LFQRatio2)
p=LFQRatio2[,7]

#estimate the proportion of true null hypotheses with different methods
r=estim.pi0(p)
r$pi0

#estimate the proportion of true null hypotheses with the "abh" method
r=estim.pi0(p, pi0.method="abh")
r$pi0

#compare with one minus the proportion of human proteins 
prop_human=sum(LFQRatio2$Organism=="human")/length(LFQRatio2$Organism)
pi0_true=1-prop_human
pi0_true

</code></pre>


</div>