<div class="container">

<table style="width: 100%;"><tr>
<td>crosshap-package</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>crosshap: Local Haplotype Clustering and Visualization</h2>

<h3>Description</h3>

<p>A local haplotyping visualization toolbox to capture major patterns of co-inheritance between clusters of linked variants, whilst connecting findings to phenotypic and demographic traits across individuals. 'crosshap' enables users to explore and understand genomic variation across a trait-associated region. For an example of successful local haplotype analysis, see Marsh et al. (2022) <a href="https://doi.org/10.1007/s00122-022-04045-8">doi:10.1007/s00122-022-04045-8</a>.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Jacob Marsh <a href="mailto:jake.marsh@live.com.au">jake.marsh@live.com.au</a> (<a href="https://orcid.org/0000-0003-3734-2023">ORCID</a>)
</p>
<p>Authors:
</p>

<ul>
<li>
<p> Brady Johnston <a href="mailto:brady.johnston@me.com">brady.johnston@me.com</a> (<a href="https://orcid.org/0000-0001-6301-2269">ORCID</a>)
</p>
</li>
<li>
<p> Jakob Petereit <a href="mailto:petinho86@gmail.com">petinho86@gmail.com</a> (<a href="https://orcid.org/0000-0003-2159-0380">ORCID</a>)
</p>
</li>
</ul>
<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://jacobimarsh.github.io/crosshap/">https://jacobimarsh.github.io/crosshap/</a>
</p>
</li>
<li>
<p> Report bugs at <a href="https://github.com/jacobimarsh/crosshap/issues">https://github.com/jacobimarsh/crosshap/issues</a>
</p>
</li>
</ul>
</div>