<div class="container">

<table style="width: 100%;"><tr>
<td>tagphenos</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Calculate SNP phenotypic associations</h2>

<h3>Description</h3>

<p>tagphenos() reports the frequency of allele types for each SNP and calculates
phenotype associations for the different alleles, before returning this
information in a $Varfile in a HapObject. This is an internal function that
is not intended for external use.
</p>


<h3>Usage</h3>

<pre><code class="language-R">tagphenos(MGfile, bin_vcf, pheno, het_phenos = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>MGfile</code></td>
<td>
<p>SNP marker groups clustered using DBscan.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bin_vcf</code></td>
<td>
<p>Binary VCF for region of interest reformatted by
run_haplotyping().</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pheno</code></td>
<td>
<p>Input numeric phenotype data for each individual.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>het_phenos</code></td>
<td>
<p>When FALSE, phenotype associations for SNPs are calculated
from reference and alternate allele individuals only, when TRUE, heterozygous
individuals are included assuming additive allele effects.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Returns intermediate of haplotype object.
</p>


</div>