<div class="container">

<table style="width: 100%;"><tr>
<td>pseudo_haps</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Identify haplotypes from clustered SNPs</h2>

<h3>Description</h3>

<p>pseudo_haps() calls the most common allelic states for each SNP marker group
across individuals, before building dummy SNPs for each marker group that
mimic the binary vcf format. This is the step which determines the haplotype
combinations, and therefore enables several summaries to be returned - as
contained in the $Hapfile and preliminary $Indfile and finalised $MGfile,
following marker group smoothing. This is an internal function not intended
for external use.
</p>


<h3>Usage</h3>

<pre><code class="language-R">pseudo_haps(preMGfile, bin_vcf, minHap, LD, keep_outliers)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>preMGfile</code></td>
<td>
<p>SNP clusters from DBscan.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bin_vcf</code></td>
<td>
<p>Binary VCF for region of interest reformatted by
run_haplotyping().</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>minHap</code></td>
<td>
<p>Minimum size (nIndividuals) to keep haplotype combinations</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>LD</code></td>
<td>
<p>LD matrix input.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>keep_outliers</code></td>
<td>
<p>When FALSE, marker group smoothing is performed to
remove outliers.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Returns intermediate of haplotype object
</p>


</div>