<div class="container">

<table style="width: 100%;"><tr>
<td>build_right_clusterplot</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Right intra-cluster linkage plot</h2>

<h3>Description</h3>

<p>build_right_jitterplot() builds a horizontal plot displaying the mean
pairwise R^2 linkage between each SNP and all other SNPs in its marker group,
grouped by marker group, coloured by alternate allele frequency. Makes use of
the $Varfile information from haplotyping object. It is an internal function
called by crosshap_viz(), though can be called separately to build a
stand-alone plot.
</p>


<h3>Usage</h3>

<pre><code class="language-R">build_right_clusterplot(HapObject, epsilon, hide_labels = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>HapObject</code></td>
<td>
<p>Haplotype object created by run_haplotyping().</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>epsilon</code></td>
<td>
<p>Epsilon to visualize haplotyping results for.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>hide_labels</code></td>
<td>
<p>If TRUE, legend is hidden.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A ggplot2 object.
</p>


<h3>Examples</h3>

<pre><code class="language-R">build_right_clusterplot(HapObject, epsilon = 0.6, hide_labels = FALSE)

</code></pre>


</div>