<div class="container">

<table style="width: 100%;"><tr>
<td>run_haplotyping</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Cluster SNPs and identify haplotypes</h2>

<h3>Description</h3>

<p>run_haplotyping() performs density-based clustering of SNPs in region of
interest to identify Marker Groups. Individuals are classified by haplotype
combination based on shared combinations of Marker Group alleles. Returns a
haplotyping object (HapObject), which can be used as input to build clustering
tree for epsilon optimization using clustree_viz(), and can be visualized with
reference to phenotype and metadata using crosshap_viz().
</p>


<h3>Usage</h3>

<pre><code class="language-R">run_haplotyping(
  vcf,
  LD,
  pheno,
  metadata = NULL,
  epsilon = c(0.2, 0.4, 0.6, 0.8, 1),
  MGmin = 30,
  minHap = 9,
  hetmiss_as = "allele",
  het_phenos = FALSE,
  keep_outliers = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>vcf</code></td>
<td>
<p>Input VCF for region of interest.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>LD</code></td>
<td>
<p>Pairwise correlation matrix of SNPs in region (e.g. from PLINK).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pheno</code></td>
<td>
<p>Input numeric phenotype data for each individual.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>metadata</code></td>
<td>
<p>Metadata input (optional).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>epsilon</code></td>
<td>
<p>Epsilon values for clustering SNPs with DBscan.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>MGmin</code></td>
<td>
<p>Minimum SNPs in marker groups, MinPts parameter for DBscan.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>minHap</code></td>
<td>
<p>Minimum nIndividuals in a haplotype combination.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>hetmiss_as</code></td>
<td>
<p>If hetmiss_as = "allele", heterozygous-missing SNPs './N'
are recoded as 'N/N', if hetmiss_as = "miss", the site is recoded as missing.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>het_phenos</code></td>
<td>
<p>When FALSE, phenotype associations for SNPs are calculated
from reference and alternate allele individuals only, when TRUE, heterozygous
individuals are included assuming additive allele effects.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>keep_outliers</code></td>
<td>
<p>When FALSE, marker group smoothing is performed to
remove outliers.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A comprehensive haplotyping S3 object (HapObject) for each provided
epsilon value, needed for clustree_viz() and crosshap_viz().
</p>


</div>