<div class="container">

<table style="width: 100%;"><tr>
<td>prepare_hap_umap</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>UMAP haplotype visualization helper</h2>

<h3>Description</h3>

<p>prepare_hap_umap() builds a large composite ggplot2 object ready for faceting
and animation (see vignette) for visualizing SNP alleles (coloured by Marker
Group) possessed by individuals with each haplotype. UMAP
coordinates for each SNP can be generated using umap::umap(), with the LD
matrix generated for run_haplotyping() as input. When fully rendered and
faceted, the resultant GIF intuitively visualizes the shared loci within each
Marker Group that are constant within each haplotype combination.
</p>


<h3>Usage</h3>

<pre><code class="language-R">prepare_hap_umap(
  umap_in,
  hetmiss_as = "allele",
  HapObject,
  epsilon,
  vcf,
  nsamples = 25
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>umap_in</code></td>
<td>
<p>UMAP results produced for a haplotype object at a given epsilon.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>hetmiss_as</code></td>
<td>
<p>If hetmiss_as = "allele", heterozygous-missing SNPs './N'
are recoded as 'N/N', if hetmiss_as = "miss", the site is recoded as missing.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>HapObject</code></td>
<td>
<p>Haplotype object created by run_haplotyping().</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>epsilon</code></td>
<td>
<p>Epsilon matching the haplotype object used for umap_in.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>vcf</code></td>
<td>
<p>Input vcf.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nsamples</code></td>
<td>
<p>Number of times to sample each haplotype group, will directly
translate to the number of frames in animation. Should be the same as the
nframes passed to gganimate::animate().</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A large ggplot2 object.
</p>


</div>