<div class="container">

<table style="width: 100%;"><tr>
<td>ctl-package</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
CTL - CTL mapping in experimental crosses
</h2>

<h3>Description</h3>

<p>Analysis of experimental crosses to identify genetic markers associated with correlation 
changes in quantitative traits (CTL). The additional correlation information obtained 
can be combined with QTL information to perform de novo reconstruction of interaction 
networks.
</p>
<p>For more background information about the method we refer to the methodology article 
published in XX (201X).
</p>
<p>The R package is a basic iomplementation and it includes the following core functionality:
</p>

<ul>
<li> <p><code>CTLscan</code> - Main function to scan for CTL.
</p>
</li>
<li> <p><code>CTLsignificant</code> - Significant interactions from a <code>CTLscan</code>.
</p>
</li>
<li> <p><code>CTLnetwork</code> - Create a CTL network from a <code>CTLscan</code>.
</p>
</li>
<li> <p><code>image.CTLobject</code> - Heatmap overview of a CTLscan.
</p>
</li>
<li> <p><code>plot.CTLscan</code> - Plot the CTL curve for a single trait.
</p>
</li>
<li> <p><code>ctl.circle</code> - Circle plot CTLs on single and multiple traits.
</p>
</li>
<li> <p><code>ctl.lineplot</code> - Line plot CTLs on single and multiple traits.
</p>
</li>
<li> <p><code>CTLprofiles</code> - Extract CTL interaction profiles.
</p>
</li>
</ul>
<p>For all these functions we also provide examples and demonstrations on real genetical 
genomics data. We thank all contributors for publishing their data online and will 
accept submissions of intrestion datasets, currently ctl provides:
</p>

<ul>
<li> <p><code>ath.metabolites</code> - Metabolite expression data from Arabidopsis Thaliana
</p>
</li>
<li> <p><code>ath.churchill</code> - Metabolite expression data from Arabidopsis Thaliana
</p>
</li>
<li> <p><code>yeast.brem</code> - Gene expression data from Saccharomyces cerevisiae
</p>
</li>
</ul>
<h3>Details</h3>

<p>More detailed information and/or examples are given per function as needed.
Some additional functionality:
</p>

<ul>
<li> <p><code>basic.qc</code> - Some basic quality checks for phenotype and genotype data
</p>
</li>
<li> <p><code>CTLscan.cross</code> - Use an R/qtl cross object with <code>CTLscan</code>
</p>
</li>
</ul>
<h3>Author(s)</h3>

 
<p>Danny Arends <a href="mailto:Danny.Arends@gmail.com">Danny.Arends@gmail.com</a><br>
Maintainer: Danny Arends <a href="mailto:Danny.Arends@gmail.com">Danny.Arends@gmail.com</a><br>
Contributions from: Bruno Tesson, Pjotr Prins and Ritsert C. Jansen
</p>


<h3>References</h3>


<ul><li>
<p> TODO
</p>
</li></ul>
<h3>See Also</h3>


<ul>
<li> <p><code>CTLscan</code> - Scan for CTL
</p>
</li>
<li> <p><code>CTLscan.cross</code> - Use an R/qtl cross object with <code>CTLscan</code>
</p>
</li>
</ul>
</div>