<div class="container">

<table style="width: 100%;"><tr>
<td>plot.CTLscan</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Plot CTL results as bar, line or GWAS plot. </h2>

<h3>Description</h3>

<p>Plot the CTL results for a genome scan (the output of <code>CTLscan</code>) as a barplot, curved line 
or GWAS plot.
</p>


<h3>Usage</h3>

<pre><code class="language-R">## S3 method for class 'CTLscan'
plot(x, mapinfo = NULL, type = c("barplot","gwas","line"), 
onlySignificant = TRUE, significance = 0.05, gap = 25, plot.cutoff = FALSE, 
do.legend=TRUE, legend.pos = "topleft", cex.legend=1.0, ydim=NULL, 
ylab="-log10(P-value)", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>                 An object of class <code>"CTLscan"</code>, as output by <code>CTLscan</code>. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mapinfo</code></td>
<td>
<p>           The mapinfo matrix with 3 columns: "Chr" - the chromosome number, "cM" - the 
location of the marker in centiMorgans and the 3rd column "Mbp" - The location 
of the marker in Mega basepairs. If supplied the marker names (rownames) 
should match those in the CTLobject. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>type</code></td>
<td>
<p>              Type of plot: Summed barplot, GWAS style plot or a basic line plot. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>onlySignificant</code></td>
<td>
<p>   Plot only the significant contributions to the CTL profile. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>significance</code></td>
<td>
<p>      Significance threshold for setting a genomewide FDR. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gap</code></td>
<td>
<p>               Gap in Cm between chromosomes. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot.cutoff</code></td>
<td>
<p>       Adds a line at -log10(significance) and adds a legend showing the 
significance level. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>do.legend</code></td>
<td>
<p>         Adds a legend showing which phenotypes contribute to the CTL profile. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>legend.pos</code></td>
<td>
<p>        Position of the legend in the plot window. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cex.legend</code></td>
<td>
<p>        Maginification of the text in the legend. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ydim</code></td>
<td>
<p>              Dimension of the y-axis, if NULL then it will be calculated. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ylab</code></td>
<td>
<p>              Label for the y-axis. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>             Passed to the function <code>plot</code> when it is called. </p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>None.
</p>


<h3>Value</h3>

<p>None.
</p>


<h3>Author(s)</h3>

 
<p>Danny Arends <a href="mailto:Danny.Arends@gmail.com">Danny.Arends@gmail.com</a><br>
Maintainer: Danny Arends <a href="mailto:Danny.Arends@gmail.com">Danny.Arends@gmail.com</a> 
</p>


<h3>See Also</h3>


<ul>
<li> <p><code>CTLscan</code> - Scan for CTL
</p>
</li>
<li> <p><code>print.CTLscan</code> - Print a summary of a CTLscan
</p>
</li>
<li> <p><code>par</code> - Plot parameters
</p>
</li>
<li> <p><code>colors</code> - Colors used in plotting
</p>
</li>
</ul>
<h3>Examples</h3>

<pre><code class="language-R">  library(ctl)
  data(ath.result)       # Arabidopsis thaliana results
  data(ath.metabolites)  # Arabidopsis thaliana data (phenotypes, genotypes and mapinfo

  plot(ath.result[[3]])
  plot(ath.result[[2]], mapinfo = ath.metab[[3]])
  plot(ath.result[[1]], mapinfo = ath.metab[[3]])
  plot(ath.result[[3]], mapinfo = ath.metab[[3]])
  plot(ath.result[[3]], mapinfo = ath.metab[[3]], type="gwas")
  plot(ath.result[[3]], mapinfo = ath.metab[[3]], type="line")
</code></pre>


</div>