<div class="container">

<table style="width: 100%;"><tr>
<td>CTLscan</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>CTLscan - Scan for Correlated Trait Locus (CTL) </h2>

<h3>Description</h3>

<p>Scan for Correlated Trait Locus (CTL) in populations
</p>


<h3>Usage</h3>

<pre><code class="language-R">CTLscan(genotypes, phenotypes, phenocol, nperm=100, nthreads = 1, 
strategy = c("Exact", "Full", "Pairwise"),
parametric = FALSE, adjust=TRUE, qtl = TRUE, verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>genotypes</code></td>
<td>
<p>   Matrix of genotypes. (individuals x markers) </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>phenotypes</code></td>
<td>
<p>  Matrix of phenotypes. (individuals x phenotypes) </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>phenocol</code></td>
<td>
<p>    Which phenotype column(s) should we analyse. Default: Analyse all phenotypes. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nperm</code></td>
<td>
<p>       Number of permutations to perform. This parameter is not used when method="Exact". </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nthreads</code></td>
<td>
<p>    Number of CPU cores to use during the analysis. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>strategy</code></td>
<td>
<p>    The permutation strategy to use, either 
</p>

<ul>
<li>
<p> Exact - Uses exact calculations to calculate the likelihood of a difference 
in correlation: Cor(AA) - Cor(BB). Using a Bonferroni correction.
</p>
</li>
<li>
<p> Full - Most powerful analysis method - Compensate for marker and trait 
correlation structure (Breitling et al.).
</p>
</li>
<li>
<p> Pairwise - Suitable when we have a lot of markers and only a few 
traits (&lt; 50) (human GWAS)- Compensates only for marker correlation structure.
</p>
</li>
</ul>
<p>Note: Exact is the default and fastest option it uses a normal distribution for 
estimating p-values and uses bonferoni correction. It has however the least power to 
detect CTLs, the two other methods (Full and Pairwise) perform permutations to
assign significance. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>parametric</code></td>
<td>
<p>  Use non-parametric testing (Spearman) or parametric testing (Pearson). The DEFAULT is to 
use non-parametric tests which are less sensitive to outliers in the phenotype data. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>adjust</code></td>
<td>
<p>      Adjust p-values for multiple testing (only used when strategy = Exact). </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>qtl</code></td>
<td>
<p>         Use the internal slow QTL mapping method to map QTLs. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>     Be verbose. </p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>By default the algorithm will not do QTL mapping, the qtl component of the output is an vector of 0 scores 
for LOD. This is to remove some computational burden, please use the have.qtls parameter to provide QTL data. 
Some computational bottleneck of the algorithm are:
</p>

<ul>
<li>
<p> RAM available to the system with large number of markers (100K+) and/or  phenotypes (100K+).
</p>
</li>
<li>
<p> Computational time with large sample sizes (5000+) and/or huge amount of phenotype data (100K+).
</p>
</li>
<li>
<p> Very very huge amounts of genotype markers (1M+)
</p>
</li>
</ul>
<p>Some way of avoiding these problems are: CTL mapping using only a single chromosome at a time and / or
selecting a smaller subsets of phenotype data for analysis.
</p>


<h3>Value</h3>

<p>CTLobject, a list with at each index (i) an CTLscan object:
</p>

<ul>
<li>
<p> $dcor  - Matrix of Z scores (method=Exact), or Power/Adjacency Z scores or for each trait at each
marker (n.markers x n.phenotypes)
</p>
</li>
<li>
<p> $perms - Vector of maximum scores obtained during permutations  (n.perms)
</p>
</li>
<li>
<p> $ctl   - Matrix of LOD scores for CTL likelihood of phenotype i (n.markers x n.phenotypes)
</p>
</li>
<li>
<p> $qtl   - Vector of LOD scores for QTL likelihood of phenotype i (n.markers)
</p>
</li>
</ul>
<h3>Author(s)</h3>

 
<p>Danny Arends <a href="mailto:Danny.Arends@gmail.com">Danny.Arends@gmail.com</a><br>
Maintainer: Danny Arends <a href="mailto:Danny.Arends@gmail.com">Danny.Arends@gmail.com</a> 
</p>


<h3>References</h3>

<p>TODO
</p>


<h3>See Also</h3>


<ul>
<li> <p><code>CTLscan.cross</code> - Use an R/qtl cross object with <code>CTLscan</code>
</p>
</li>
<li> <p><code>CTLregions</code> - Regions with significant CTLs from a <code>CTLscan</code>
</p>
</li>
<li> <p><code>CTLsignificant</code> - Significant interactions from a <code>CTLscan</code>
</p>
</li>
<li> <p><code>CTLnetwork</code> - Create a CTL network from a <code>CTLscan</code>
</p>
</li>
<li> <p><code>image.CTLobject</code> - Heatmap overview of a <code>CTLscan</code>
</p>
</li>
<li> <p><code>plot.CTLscan</code> - Plot the CTL curve for a single trait
</p>
</li>
</ul>
<h3>Examples</h3>

<pre><code class="language-R">  library(ctl)
  data(ath.metabolites)                 # Arabidopsis Thaliana data set

  ctlscan &lt;- CTLscan(ath.metab$genotypes, ath.metab$phenotypes, phenocol=1:4)
  ctlscan

  # Genetic regions with significant CTLs found for the first phenotype
  CTLregions(ctlscan, ath.metab$map, phenocol = 1)
  
  summary &lt;- CTLsignificant(ctlscan)    # Matrix of Trait, Marker, Trait interactions 
  summary                               # Get a list of significant CTLs

  nodes &lt;- ctl.lineplot(ctlscan, ath.metab$map)  # Line plot the phenotypes
  nodes
</code></pre>


</div>