<div class="container">

<table style="width: 100%;"><tr>
<td>codon_diff</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Differential codon usage analysis</h2>

<h3>Description</h3>

<p><code>codon_diff</code> takes two set of coding sequences and
perform differential codon usage analysis.
</p>


<h3>Usage</h3>

<pre><code class="language-R">codon_diff(seqs1, seqs2, codon_table = get_codon_table())
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>seqs1</code></td>
<td>
<p>DNAStringSet, or an object that can be coerced to a DNAStringSet</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>seqs2</code></td>
<td>
<p>DNAStringSet, or an object that can be coerced to a DNAStringSet</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>codon_table</code></td>
<td>
<p>a table of genetic code derived from 'get_codon_table' or 'create_codon_table'.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>a data.table of the differential codon usage analysis. Global tests examine wthether a codon
is used differently relative to all the other codons. Family tests examine whether a codon is used
differently relative to other codons that encode the same amino acid. Subfamily tests examine whether
a codon is used differently relative to other synonymous codons that share the same first two nucleotides.
Odds ratio &gt; 1 suggests a codon is used at higher frequency in <code>seqs1</code> than in <code>seqs2</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">yeast_exp_sorted &lt;- yeast_exp[order(yeast_exp$fpkm),]
seqs1 &lt;- yeast_cds[names(yeast_cds) %in% head(yeast_exp_sorted$gene_id, 1000)]
seqs2 &lt;- yeast_cds[names(yeast_cds) %in% tail(yeast_exp_sorted$gene_id, 1000)]
cudiff &lt;- codon_diff(seqs1, seqs2)

</code></pre>


</div>