<div class="container">

<table style="width: 100%;"><tr>
<td>est_rscu</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Estimate RSCU</h2>

<h3>Description</h3>

<p><code>est_rscu</code> returns the RSCU value of codons
</p>


<h3>Usage</h3>

<pre><code class="language-R">est_rscu(cf, weight = 1, pseudo_cnt = 1, codon_table = get_codon_table())
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>cf</code></td>
<td>
<p>matrix of codon frequencies as calculated by 'count_codons()'.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>weight</code></td>
<td>
<p>a vector of the same length as 'seqs' that gives different weights
to CDSs when count codons. for example, it could be gene expression levels.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pseudo_cnt</code></td>
<td>
<p>pseudo count to avoid dividing by zero. This may occur when
only a few sequences are available for RSCU calculation.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>codon_table</code></td>
<td>
<p>a table of genetic code derived from 'get_codon_table' or
'create_codon_table'.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>a data.table of codon info. RSCU values are reported in the last column.
</p>


<h3>References</h3>

<p>Sharp PM, Tuohy TM, Mosurski KR. 1986. Codon usage in yeast: cluster analysis clearly differentiates highly and lowly expressed genes. Nucleic Acids Res 14:5125-5143.
</p>


<h3>Examples</h3>

<pre><code class="language-R"># compute RSCU of all yeast genes
cf_all &lt;- count_codons(yeast_cds)
est_rscu(cf_all)

# compute RSCU of highly expressed (top 500) yeast genes
heg &lt;- head(yeast_exp[order(-yeast_exp$fpkm), ], n = 500)
cf_heg &lt;- count_codons(yeast_cds[heg$gene_id])
est_rscu(cf_heg)

</code></pre>


</div>