<div class="container">

<table style="width: 100%;"><tr>
<td>get_tai</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Calculate TAI</h2>

<h3>Description</h3>

<p><code>get_tai</code> calculates tRNA Adaptation Index (TAI) of each CDS
</p>


<h3>Usage</h3>

<pre><code class="language-R">get_tai(cf, trna_w)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>cf</code></td>
<td>
<p>matrix of codon frequencies as calculated by 'count_codons()'.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>trna_w</code></td>
<td>
<p>tRNA weight for each codon, can be generated with 'est_trna_weight()'.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>a named vector of TAI values
</p>


<h3>References</h3>

<p>dos Reis M, Savva R, Wernisch L. 2004. Solving the riddle of codon usage
preferences: a test for translational selection. Nucleic Acids Res 32:5036-5044.
</p>


<h3>Examples</h3>

<pre><code class="language-R"># calculate TAI of yeast genes based on genomic tRNA copy numbers
w &lt;- est_trna_weight(yeast_trna_gcn)
cf_all &lt;- count_codons(yeast_cds)
tai &lt;- get_tai(cf_all, w)
head(tai)
hist(tai)

</code></pre>


</div>