<div class="container">

<table style="width: 100%;"><tr>
<td>Simulation Tool</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Simulate ORFs and Expression Data</h2>

<h3>Description</h3>

<p>These utility functions generate data for simulation studies
including fake ORFs and expression values.
</p>


<h3>Usage</h3>

<pre><code class="language-R">  simu.orf(n, b.Init, phi.Obs = NULL, AA.prob = NULL, orf.length = NULL,
           orf.names = NULL, model = .CF.CT$model)
  simu.phi.Obs(Phi, sigmaW.lim = 1, bias.Phi = 0)
  simu.mixnormerr(n, param)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>n</code></td>
<td>
<p>number of ORFs or sequences.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>b.Init</code></td>
<td>
<p>parameters of mutation and selection of format
<code>b</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>phi.Obs</code></td>
<td>
<p>an object of format <code>phi.Obs</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>AA.prob</code></td>
<td>
<p>proportion of amino acids.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>orf.length</code></td>
<td>
<p>lengths of ORFs.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>orf.names</code></td>
<td>
<p>names of ORFs.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>model</code></td>
<td>
<p>model to be simulated.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Phi</code></td>
<td>
<p>expression values (potentially true expression).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sigmaW.lim</code></td>
<td>
<p>std of measurement errors (between Phi and phi.Obs).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bias.Phi</code></td>
<td>
<p>bias (in log scale) for observed phi.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>param</code></td>
<td>
<p>as in <code>dmixnormerr()</code></p>
</td>
</tr>
</table>
<h3>Details</h3>

<p><code>simu.orf()</code> generates ORFs or sequences based on the <code>b.Init</code>
and <code>phi.Obs</code>.
</p>
<p>If <code>phi.Obs</code> is omitted, then standard log normal random variables
are instead).
</p>
<p>If <code>AA.prob</code> is omitted, then uniform proportion is assigned.
</p>
<p>If <code>orf.length</code> is omitted, then 10 to 20 codons are randomly
assigned.
</p>
<p>If <code>orf.names</code> is omitted, then "ORF1" to "ORFn" are assigned.
</p>
<p><code>simu.phi.Obs()</code> generates <code>phi.Obs</code> by adding normal random
errors to <code>Phi</code>, and errors have mean 0 and standard deviation
<code>sigmaW.lim</code>.
</p>
<p><code>simu.mixnormerr()</code> generates <code>Phi</code> according to the <code>param</code>,
and adds normal random errors to <code>Phi</code>.
</p>


<h3>Value</h3>

<p><code>simu.orf()</code> returns a list of format <code>seq.data</code>.
</p>
<p><code>simu.phi.Obs()</code> returns a vector of format <code>phi.Obs</code>.
</p>
<p><code>simu.mixnormerr()</code> returns a list contains three vectors of length
<code>n</code>: one for expected gene expression <code>Phi</code>, one for observed
gene expression <code>phi.Obs</code>, and one for the component id <code>id.K</code>.
</p>


<h3>Author(s)</h3>

<p>Wei-Chen Chen <a href="mailto:wccsnow@gmail.com">wccsnow@gmail.com</a>.
</p>


<h3>References</h3>

<p><a href="https://github.com/snoweye/cubfits/">https://github.com/snoweye/cubfits/</a>
</p>


<h3>See Also</h3>

<p><code>read.seq()</code>, <code>read.phi.df()</code>,
<code>write.seq()</code>, <code>write.phi.df()</code>, and
<code>mixnormerr.optim()</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
suppressMessages(library(cubfits, quietly = TRUE))
set.seed(1234)

# Generate sequences.
da.roc &lt;- simu.orf(length(ex.train$phi.Obs), b.Init$roc,
                   phi.Obs = ex.train$phi.Obs, model = "roc")
names(da.roc) &lt;- names(ex.train$phi.Obs)
write.fasta(da.roc, names(da.roc), "toy_roc.fasta")

## End(Not run)
</code></pre>


</div>