<div class="container">

<table style="width: 100%;"><tr>
<td>SCUO Index</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Function for Synonymous Codon Usage Order (SCUO) Index</h2>

<h3>Description</h3>

<p>Calculate the Synonymous Codon Usage Order (SCUO) index for each gene.
Used as a substitute for expression in cases of without expression
measurements.
</p>


<h3>Usage</h3>

<pre><code class="language-R">  calc_scuo_values(codon.counts)
</code></pre>


<h3>Arguments</h3>

<table><tr style="vertical-align: top;">
<td><code>codon.counts</code></td>
<td>
<p>an object of format <code>scuo</code>.</p>
</td>
</tr></table>
<h3>Details</h3>

<p>This function computes SCUO index for each gene. Typically, this method
is completely based on entropy and information theory to estimate
expression values of sequences according to their codon information.
</p>


<h3>Value</h3>

<p><code>SCUO</code> indices are returned.
</p>


<h3>Author(s)</h3>

<p>Drew Schmidt.
</p>


<h3>References</h3>

<p><a href="https://www.tandfonline.com/doi/abs/10.1080/03081070500502967">https://www.tandfonline.com/doi/abs/10.1080/03081070500502967</a>
</p>
<p>Wan X.-F., Zhou J., Xu D.
“CodonO: a new informatics method for measuring synonymous codon usage
bias within and across genomes”
International Journal of General Systems Vol. 35, Iss. 1, 2006.
</p>


<h3>See Also</h3>

<p><code>scuo.random()</code>, <code>calc_cai_values()</code>,
<code>calc_scu_values()</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
suppressMessages(library(cubfits, quietly = TRUE))

y.scuo &lt;- convert.y.to.scuo(ex.train$y)
SCUO &lt;- calc_scuo_values(y.scuo)$SCUO
plotprxy(ex.train$phi.Obs, SCUO, ylab = "SCUO (log10)")

## End(Not run)
</code></pre>


</div>