<div class="container">

<table style="width: 100%;"><tr>
<td>Data Formats</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Data Formats</h2>

<h3>Description</h3>

<p>Data formats used in <span class="pkg">cubfits</span>.
</p>


<h3>Format</h3>

<p>All are in simple formats as S3 default lists or data frames.
</p>


<h3>Details</h3>


<ul>
<li>
<p>Format <code>b</code>: <br>
A named list <code>A</code> contains amino acids.
Each element of the list <code>A[[i]]</code> is a list of elements
<code>coefficients</code> (coefficients of log(mu) and Delta.t),
<code>coef.mat</code> (matrix format of <code>coefficients</code>), and
<code>R</code> (covariance matrix of <code>coefficients</code>).
Note that <code>coefficients</code> and <code>R</code> are typically as in the output
of <code>vglm()</code> of <span class="pkg">VGAM</span> package.
Also, <code>coef.mat</code> and <code>R</code> may miss in some cases.
<br>
e.g. <code>A[[i]]$coef.mat</code> is the regression beta matrix of <code>i</code>-th
amino acid.
</p>
</li>
<li>
<p>Format <code>bVec</code>: <br>
A vector simply contains all coefficients of a <code>b</code> object <code>A</code>.
Note that this is probably only used inside MCMC or the output of
<code>vglm()</code> of <span class="pkg">VGAM</span> package.
<br>
e.g. <code>do.call("c", lapply(A, function(x) x$coefficients))</code>.
</p>
</li>
<li>
<p>Format <code>n</code>: <br>
A named list <code>A</code> contains amino acids.
Each element of the list <code>A[[i]]</code> is a vector containing total
codon counts.
<br>
e.g. <code>A[[i]][j]</code> is for <code>j</code>-th ORF of <code>i</code>-th amino acid
<code>names(A)[i]</code>.
</p>
</li>
<li>
<p>Format <code>n.list</code>: <br>
A named list <code>A</code> contains ORFs.
Each element of the list <code>A[[i]]</code> is a named list of amino acid
containing total count.
<br>
e.g. <code>A[[i]][[j]]</code> contains total count of
<code>j</code>-th amino acid in <code>i</code>-th ORF.
</p>
</li>
<li>
<p>Format <code>phi.df</code>: <br>
A data frame <code>A</code> contains two columns <code>ORF</code> and <code>phi.value</code>.
<br>
e.g. <code>A[i,]</code> is for <code>i</code>-th ORF.
</p>
</li>
<li>
<p>Format <code>reu13.df</code>: <br>
A named list <code>A</code> contains amino acids.
Each element is a data frame summarizing ORF and expression.
The data frame has four to five columns including
<code>ORF</code>, <code>phi</code> (expression), <code>Pos</code> (amino acid position),
<code>Codon</code> (synonymous codon), and
<code>Codon.id</code> (synonymous codon id, for computing only).
Note that <code>Codon.id</code> may miss in some cases.
<br>
e.g. <code>A[[i]][17,]</code> is the 17-th recode of <code>i</code>-th amino acid.
</p>
</li>
<li>
<p>Format <code>reu13.list</code>: <br>
A named list <code>A</code> contains ORFs.
Each element is a named list <code>A[[i]]</code> contains amino acids.
Each element of nested list <code>A[[i]][[j]]</code> is a position vector
of synonymous codon.
<br>
e.g. <code>A[[i]][[j]][k]</code> is the <code>k</code>-th synonymous codon position of
<code>j</code>-th amino acid in the <code>i</code>-th ORF.
</p>
</li>
<li>
<p>Format <code>scuo</code>: <br>
A data frame of 8 named columns includes
<code>AA</code> (amino acid), <code>ORF</code>, <code>C1</code>, ..., <code>C6</code>
where <code>C*</code>'s are for codon counts.
</p>
</li>
<li>
<p>Format <code>seq.string</code>: <br>
Default outputs of <code>read.fasta()</code> of <span class="pkg">seqinr</span> package.
A named list <code>A</code> contains ORFs.
Each element of the list is a long string of a ORF.
<br>
e.g. <code>A[[i]][1]</code> or <code>A[[i]]</code> is the sequence of
<code>i</code>-th ORF.
</p>
</li>
<li>
<p>Format <code>seq.data</code>: <br>
Converted from <code>seq.string</code> format.
A named list <code>A</code> contains ORFs.
Each element of the list <code>A[[i]]</code> is a string vector.
Each element of the vector is a codon string.
<br>
e.g. <code>A[[i]][j]</code> is <code>i</code>-th ORF and <code>j</code>-th codon.
</p>
</li>
<li>
<p>Format <code>phi.Obs</code>: <br>
A named vector <code>A</code> of observed expression values and possibly
with measurement errors.
<br>
e.g. <code>A[i]</code> is the observed phi value of <code>i</code>-th ORF.
</p>
</li>
<li>
<p>Format <code>y</code>: <br>
A named list <code>A</code> contains amino acids.
Each element of the list <code>A[[i]]</code> is a matrix
where ORFs are in row and synonymous codons are in column.
The element of the matrix contains codon counts.
<br>
e.g. <code>A[[i]][j, k]</code> is the count for <code>i</code>-th amino acid,
<code>j</code>-th ORF, and <code>k</code>-th synonymous codon.
</p>
</li>
<li>
<p>Format <code>y.list</code>: <br>
A named list <code>A</code> contains ORFs.
Each element of the list <code>A[[i]]</code> is a named list <code>A[[i]][[j]]</code>
contains amino acids.
The element of amino acids list is a codon count vector.
<br>
e.g. <code>A[[i]][[j]][k]</code> is the count for <code>i</code>-th ORF,
<code>j</code>-th amino acid, and <code>k</code>-th synonymous codon.
</p>
</li>
</ul>
<h3>Author(s)</h3>

<p>Wei-Chen Chen <a href="mailto:wccsnow@gmail.com">wccsnow@gmail.com</a>.
</p>


<h3>References</h3>

<p><a href="https://github.com/snoweye/cubfits/">https://github.com/snoweye/cubfits/</a>
</p>


</div>