<div class="container">

<table style="width: 100%;"><tr>
<td>Generating Utility</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Generating Data Structure</h2>

<h3>Description</h3>

<p>These utility functions generate and summarize sequence strings into several
useful formats such as <code>reu13.df</code>, <code>y</code>, and
<code>n</code>, etc.
</p>


<h3>Usage</h3>

<pre><code class="language-R">  gen.reu13.df(seq.string, phi.df = NULL, aa.names = .CF.GV$amino.acid,
               split.S = TRUE, drop.X = TRUE, drop.MW = TRUE,
               drop.1st.codon = TRUE)
  gen.y(seq.string, aa.names = .CF.GV$amino.acid,
        split.S = TRUE, drop.X = TRUE, drop.MW = TRUE)
  gen.n(seq.string, aa.names = .CF.GV$amino.acid,
        split.S = TRUE, drop.X = TRUE, drop.MW = TRUE)

  gen.reu13.list(seq.string, aa.names = .CF.GV$amino.acid,
                 split.S = TRUE, drop.X = TRUE, drop.MW = TRUE,
                 drop.1st.codon = TRUE)
  gen.phi.Obs(phi.df)
  gen.scuo(seq.string, aa.names = .CF.GV$amino.acid,
           split.S = TRUE, drop.X = TRUE, drop.MW = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>seq.string</code></td>
<td>
<p>a list of sequence strings.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>phi.df</code></td>
<td>
<p>a <code>phi.df</code> object returned from <code>read.phi.df()</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>aa.names</code></td>
<td>
<p>a vector contains amino acid names for analysis.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>split.S</code></td>
<td>
<p>split amino acid 'S' if any.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>drop.X</code></td>
<td>
<p>drop amino acid 'X' if any.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>drop.MW</code></td>
<td>
<p>drop amino acid 'M' and 'W' if any.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>drop.1st.codon</code></td>
<td>
<p>if drop the first codon.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>These functions mainly take inputs of sequence strings
<code>seq.string</code> or <code>phi.df</code> and turn them
into corresponding format.
</p>


<h3>Value</h3>

<p>The outputs are data structure in corresponding formats.
See AllDataFormats for details.
</p>


<h3>Author(s)</h3>

<p>Wei-Chen Chen <a href="mailto:wccsnow@gmail.com">wccsnow@gmail.com</a>.
</p>


<h3>References</h3>

<p><a href="https://github.com/snoweye/cubfits/">https://github.com/snoweye/cubfits/</a>
</p>


<h3>See Also</h3>

<p>AllDataFormats,
<code>read.seq()</code>, <code>read.phi.df()</code>, and
<code>convert.seq.data.to.string()</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
suppressMessages(library(cubfits, quietly = TRUE))

seq.data &lt;- read.seq(get.expath("seq_200.fasta"))
phi.df &lt;- read.phi.df(get.expath("phi_200.tsv"))
aa.names &lt;- c("A", "C", "D")

# Read in from FASTA file.
seq.string &lt;- convert.seq.data.to.string(seq.data)
reu13.df &lt;- gen.reu13.df(seq.string, phi.df, aa.names)
reu13.list.new &lt;- gen.reu13.list(seq.string, aa.names)
y &lt;- gen.y(seq.string, aa.names)
n &lt;- gen.n(seq.string, aa.names)
scuo &lt;- gen.scuo(seq.string, aa.names)

# Convert to list format.
reu13.list &lt;- convert.reu13.df.to.list(reu13.df)
y.list &lt;- convert.y.to.list(y)
n.list &lt;- convert.n.to.list(n)

## End(Not run)
</code></pre>


</div>