<div class="container">

<table style="width: 100%;"><tr>
<td>Coverting Utility</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Convert Data Frame to Other Formats</h2>

<h3>Description</h3>

<p>These utility functions convert data of format divided by amino acids into
list of format divided by ORFs, or convert data to other formats.
</p>


<h3>Usage</h3>

<pre><code class="language-R">  convert.reu13.df.to.list(reu13.df)
  convert.y.to.list(y)
  convert.n.to.list(n)

  convert.y.to.scuo(y)
  convert.seq.data.to.string(seq.data)

  codon.low2up(x)
  codon.up2low(x)

  dna.low2up(x)
  dna.up2low(x)

  convert.b.to.bVec(b)
  convert.bVec.to.b(bVec, aa.names, model = .CF.CT$model[1])
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>reu13.df</code></td>
<td>
<p>a list of <code>reu13.df</code> data frames
divided by amino acids.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>y</code></td>
<td>
<p>a list of <code>y</code> data frames divided by amino acids.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n</code></td>
<td>
<p>a list of <code>n</code> vectors divided by amino acids.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>seq.data</code></td>
<td>
<p>a vector of <code>seq.data</code> format.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>a codon or dna string, such "ACG", "acg", or "A", "a".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>b</code></td>
<td>
<p>a <code>b</code> object.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bVec</code></td>
<td>
<p>a <code>bVec</code> object.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>aa.names</code></td>
<td>
<p>a vector contains amino acid names for analysis.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>model</code></td>
<td>
<p>model fitted.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p><code>convert.reu13.df.to.list()</code>, <code>convert.y.to.list()</code>, and
<code>convert.n.to.list()</code>:
these utility functions take the inputs divided by amino acids
and return the outputs divided by ORFs.
</p>
<p><code>convert.y.scuo()</code> converts <code>y</code> into <code>scuo</code>
format.
</p>
<p><code>convert.seq.data.to.string()</code> converts <code>seq.data</code> into
<code>seq.string</code> format.
</p>
<p><code>codon.low2up()</code> and <code>codon.up2low()</code> convert codon strings
between lower or upper cases.
</p>
<p><code>convert.bVec.to.b()</code> and <code>convert.b.to.bVec()</code> convert
objects <code>b</code> and <code>bVec</code>.
</p>


<h3>Value</h3>

<p>All functions return the corresponding formats.
</p>


<h3>Author(s)</h3>

<p>Wei-Chen Chen <a href="mailto:wccsnow@gmail.com">wccsnow@gmail.com</a>.
</p>


<h3>References</h3>

<p><a href="https://github.com/snoweye/cubfits/">https://github.com/snoweye/cubfits/</a>
</p>


<h3>See Also</h3>

<p>AllDataFormats,
<code>rearrange.n()</code>,
<code>rearrange.reu13.df()</code>,
<code>rearrange.y()</code>, and
<code>read.seq()</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
suppressMessages(library(cubfits, quietly = TRUE))

reu13.list &lt;- convert.reu13.df.to.list(ex.train$reu13.df)
y.list &lt;- convert.y.to.list(ex.train$y)
n.list &lt;- convert.n.to.list(ex.train$n)

scuo &lt;- convert.y.to.scuo(ex.train$y)

seq.data &lt;- read.seq(get.expath("seq_200.fasta"))
seq.string &lt;- convert.seq.data.to.string(seq.data)

codon.low2up("acg")
codon.up2low("ACG")

dna.low2up(c("a", "c", "g"))
dna.up2low(c("A", "C", "G"))

## End(Not run)
</code></pre>


</div>